---
title: "STIM1 cEFhand DMS analysis"
author: "Nisha D. Kamath, Kenneth A. Matreyek"
date: "2024-12-31"
output: html_document
---

```{r}
knitr::opts_chunk$set(include = FALSE)
library(tidyverse)
library(reshape)
library(ggrepel)
library(gridExtra)
library(scales)
library(stringr)
library(ggpubr)
library(GGally)
# install.packages("pak")
# pak::pkg_install("ggpmisc")
library(ggpmisc)
library(ggbeeswarm)
library(patchwork)
library(rstatix)
library(ggtext)
library(googlesheets4)
library(rstatix)

theme_set(theme_bw())
theme_update(panel.grid.minor = element_blank())
set.seed(96)
```

#Preliminary experiments
Testing the High and Low STIM1 overexpressor cells, using mCherry as the proxy to see how that population changes over time
```{r High vs low kozak STIM1 overexpression survival}
kozak = read.csv("data/fitness_kozak_stim1.csv")
stim_fit_ace2_r1 = kozak %>% filter(plasmid_name == "G758" & replicate == "1") 
stim_fit_ace2_r1 = stim_fit_ace2_r1 %>% mutate(frac_mcherry = pct_mcherry/stim_fit_ace2_r1[1,5])
stim_fit_consensus_r1 = kozak %>% filter(plasmid_name == "pNK001B" & replicate == "1")
stim_fit_consensus_r1 = stim_fit_consensus_r1 %>% mutate(frac_mcherry = pct_mcherry/stim_fit_consensus_r1[1,5])
stim_fit_subopt_r1 = kozak %>% filter(plasmid_name == "pNK005C" & replicate == "1") 
stim_fit_subopt_r1 = stim_fit_subopt_r1 %>% mutate(frac_mcherry = pct_mcherry/stim_fit_subopt_r1[1,5])

stim_fit_r1 = bind_rows(stim_fit_ace2_r1, stim_fit_consensus_r1, stim_fit_subopt_r1)

stim_fit_ace2_r2 = kozak %>% filter(plasmid_name == "G758" & replicate == "2") 
stim_fit_ace2_r2 = stim_fit_ace2_r2 %>% mutate(frac_mcherry = pct_mcherry/stim_fit_ace2_r2[1,5])
stim_fit_consensus_r2 = kozak %>% filter(plasmid_name == "pNK001B" & replicate == "2") 
stim_fit_consensus_r2 = stim_fit_consensus_r2 %>% mutate(frac_mcherry = pct_mcherry/stim_fit_consensus_r2[1,5])
stim_fit_subopt_r2 = kozak %>% filter(plasmid_name == "pNK005C" & replicate == "2") 
stim_fit_subopt_r2 = stim_fit_subopt_r2 %>% mutate(frac_mcherry = pct_mcherry/stim_fit_subopt_r2[1,5])

fit_merge = bind_rows(stim_fit_ace2_r1, stim_fit_ace2_r2, stim_fit_consensus_r1, stim_fit_consensus_r2, stim_fit_subopt_r1, stim_fit_subopt_r2)
fit_merge_mean = fit_merge %>% group_by(plasmid_name, time_pt) %>% mutate(mean_frac = mean(frac_mcherry), sd = sd(frac_mcherry))

fit_merge_mean$Sample <- factor(fit_merge_mean$Sample, levels = c("dEctoACE2","Consensus Kozak STIM1","Sub-optimal Kozak STIM1"))

#Fitness effect of kozak sequence
High_low_kozak_mCherry <- ggplot() +
  scale_y_log10(limits = c(0.1,1.1), expand = c(0,0.1)) + scale_x_continuous(breaks = c(0,5,10))+ 
  geom_line(data = fit_merge_mean, aes(x = time_pt, y = mean_frac, color = Sample), linewidth = 2, alpha = 0.4)+ 
  geom_point(data = fit_merge_mean, aes(x = time_pt, y = frac_mcherry, color = Sample), size = 2, alpha = 0.6) + 
  labs(x = "Days after initial time point" , y = "Fraction of initial fluorescent population") + 
  theme(legend.position="bottom", panel.grid.minor = element_blank()) +
  scale_color_discrete(labels=c("None", "High", "Low")) +
  NULL; High_low_kozak_mCherry
ggsave("Plots/supplement/supp1a_High_low_kozak_mCherry.pdf", High_low_kozak_mCherry, width = 2, height = 2)
```

Now looking at how low overexpressor STIM1 cells deplete if it's a variant rather than the WT being overexpressed
```{r Non-multiplex STIM1 variant depletion}
mch_dep = read.csv("data/stimvar_fit_final.csv")
mch_dep$per_day <- (1 - mch_dep$norm_pct)/7

mch_dep_r1 <- mch_dep %>% filter(replicate == 1)
mch_dep_r1$wt_norm <- mch_dep_r1$pct_mcherry / mch_dep_r1[mch_dep_r1$Sample == "WT STIM1" & mch_dep_r1$time_pt == 7, "pct_mcherry"]
mch_dep_r2 <- mch_dep %>% filter(replicate == 2)
mch_dep_r2$wt_norm <- mch_dep_r2$pct_mcherry / mch_dep_r2[mch_dep_r2$Sample == "WT STIM1" & mch_dep_r2$time_pt == 7, "pct_mcherry"]

mch_dep_combined_d7 <- rbind(mch_dep_r1, mch_dep_r2) #%>% filter(time_pt == 7)
mch_dep_combined_d14 <- rbind(mch_dep_r1, mch_dep_r2) #%>% filter(time_pt == 14)

mch_dep_combined <- rbind(mch_dep_combined_d7, mch_dep_combined_d14)

mch_dep_combined[mch_dep_combined$Sample == "dEctoACE2","Sample"] <- "Control"
mch_dep_combined$Sample <- factor(mch_dep_combined$Sample, levels = c("Control", "WT STIM1", "[R304W]STIM1", "[R429C]STIM1"))
mch_dep_combined$replicate <- factor(mch_dep_combined$replicate, levels = c(1,2))
mch_dep_combined$time_pt <- factor(mch_dep_combined$time_pt, levels = c(7,14))
mch_dep_combined[mch_dep_combined$Sample == "[R304W]STIM1" & mch_dep_combined$time_pt == 14 & mch_dep_combined$replicate == 2,"pct_mcherry"] <- 0.01

#Non-multiplex cell survival - Figure 2C
Low_STIM1_variant_mCherry_plot <- ggplot() + theme(panel.grid.major.x = element_blank(), axis.text.x = element_text(angle = 45, hjust = 0.5, vjust = 0.5)) +
  labs(x = NULL, y = "Percent mCherry+ cells") +
  scale_y_log10(limits = c(0.01, 15), expand = c(0,0.01)) + 
  geom_point(data = mch_dep_combined, aes(x = Sample, y = pct_mcherry, color = time_pt, shape = replicate), position = position_dodge(width = 0.5)) +
  NULL; Low_STIM1_variant_mCherry_plot
ggsave(file = "Plots/main/fig2c_Low_STIM1_variant_mCherry_plot.pdf", Low_STIM1_variant_mCherry_plot, height = 2, width = 3)

```

Similar to above but variants are multiplexed
```{r Multiplex STIM1 variant depletion}
#Multiplex cell survival - Figure 2D
comb_dbc_score = read.csv("data/comb_dbc_score.csv")
dbc_validation_plot = ggplot(data = comb_dbc_score, aes(x= endo_dbc_score, y= sko_dbc_score, color = variant))+ 
  scale_y_continuous(limits = c(-0.5, 0.5)) + scale_x_continuous(limits = c(-1,0.5))+
  scale_color_manual(values = c("WT" = "black", "R304W" = "blue", "R429C" = "red")) +
  theme(panel.grid.minor = element_blank())+
  geom_point(alpha = 0.5, size = 3) +
  geom_text_repel(data = comb_dbc_score, aes(x= endo_dbc_score, y= sko_dbc_score, color = variant, label = variant)) +
  theme(legend.position = "none") +
  labs(x= "Standard score", y= "Knockout score") +
  NULL; dbc_validation_plot
ggsave("plots/main/fig2d_dbc_validation_plot.pdf", dbc_validation_plot, height = 1.75, width = 2.25)
```

Deep mutational scan of the STIM1 cEFhand
We first bring in all the DMS data and create a large dataframe for analysis.
##Plasmid library
```{r counts}
raw_plasmid = read.csv("data/lib_raw.csv")
raw_plasmid = raw_plasmid[-1]
variant_name = read.csv("data/variant_name.csv", header = TRUE) %>% select(c("variant"))
variant_name = variant_name %>% mutate(across(where(is.character), str_trim))
raw_plasmid = full_join(variant_name, raw_plasmid)

## Figure out expected frequency limits of detection based on total read counts
plasmid_readcounts <- colSums(raw_plasmid[,c("lib1","lib2","lib3","lib4","lib5")], na.rm = T)

lod_frame <- read.csv(file = "data/Sampling_limit_of_detection.csv", header = T, stringsAsFactors = F)

## figure out read counts that should service as limits of detection for each sample
plasmid_lod_readcounts <- c()
plasmid_lod_freqs <- c()
for(re in plasmid_readcounts){
  lod_frame$reads - re
  temp_subset <- (lod_frame %>% mutate(abs_diff = abs(reads - re)) %>% arrange(abs_diff))[1,]
  plasmid_lod_readcounts <- c(plasmid_lod_readcounts, temp_subset$count)
  plasmid_lod_freqs <- c(plasmid_lod_freqs, temp_subset$freq)
}

lod_dataframe <- data.frame("reads" = plasmid_readcounts, "count" = plasmid_lod_readcounts, "freq" = plasmid_lod_freqs)

## Now figure out which data points were below the limit of detection
raw_plasmid[is.na(raw_plasmid)] <- 0

raw_plasmid2 <- data.frame("variant" = raw_plasmid$variant,
                           "lib1" = raw_plasmid$lib1 < plasmid_lod_readcounts[1],
                           "lib2" = raw_plasmid$lib2 < plasmid_lod_readcounts[2],
                           "lib3" = raw_plasmid$lib3 < plasmid_lod_readcounts[3],
                           "lib4" = raw_plasmid$lib4 < plasmid_lod_readcounts[4],
                           "lib5" = raw_plasmid$lib5 < plasmid_lod_readcounts[5])

## These were the total number of variants below the limit of detection in each plasmid sequencing run:
colSums(raw_plasmid2[,c("lib1","lib2","lib3","lib4","lib5")])

## Now calculate plasmid frequencies, and for plasmids below the limit of detection in the original plasmid prep, call them NA
plasmid_freq = raw_plasmid %>% mutate(r1 = lib1/plasmid_readcounts[1], 
                                      r2 = lib2/plasmid_readcounts[2], 
                                      r3 = lib3/plasmid_readcounts[3],
                                      r4 = lib4/plasmid_readcounts[4],
                                      r5 = lib5/plasmid_readcounts[5])

plasmid_freq[raw_plasmid2$lib3 == T,"r3"] <- plasmid_lod_freqs[3]
plasmid_freq[raw_plasmid2$lib4 == T,"r4"] <- plasmid_lod_freqs[4]
plasmid_freq[raw_plasmid2$lib5 == T,"r5"] <- plasmid_lod_freqs[5]

plasmid_freq[raw_plasmid2$lib1 == T,c("r1","r2","r3","r4","r5")] <- NA
plasmid_freq[raw_plasmid2$lib2 == T,c("r1","r2","r3","r4","r5")] <- NA

plasmid_freq2 = raw_plasmid %>% mutate(r1_freq = lib1/sum(lib1, na.rm= T), r2_freq = lib2/sum(lib2, na.rm= T), r3_freq = lib3/sum(lib3, na.rm= T), r4_freq = lib4/sum(lib4, na.rm= T), r5_freq = lib5/sum(lib5, na.rm= T))
plasmid_freq2$freq_endo = rowMeans(plasmid_freq2[,c("r1_freq","r2_freq")], na.rm = T)
plasmid_freq2$freq_sko = rowMeans(plasmid_freq2[,c("r3_freq", "r4_freq", "r5_freq")], na.rm= T)
plasmid_freq2$zscore_endo = (plasmid_freq2$freq_endo - mean(plasmid_freq2$freq_endo, na.rm= TRUE))/sd(plasmid_freq2$freq_endo, na.rm= TRUE)
plasmid_freq2$zscore_sko = (plasmid_freq2$freq_sko - mean(plasmid_freq2$freq_sko, na.rm= TRUE))/sd(plasmid_freq2$freq_sko, na.rm= TRUE)

#plasmid library density plot - figure supplement 2A
plasmid_fitted_distribution_plot <- ggplot() + labs(x= "Frequency of variant", y= "Density") + 
  scale_x_log10(labels = scales::scientific) +
  geom_density(data = plasmid_freq2, aes(x = freq_endo), color = "red") +
  geom_density(data = plasmid_freq2, aes(x = freq_sko), color = "green") +
  scale_y_continuous(limits = c(0,1.2))+
  scale_color_manual(values = c("Library 1", "Library 2"))
plasmid_fitted_distribution_plot

ggsave("plots/supplement/supp2a_plasmid_fitted_distribution_plot.pdf", plasmid_fitted_distribution_plot, height = 2, width = 3)
# #distribution of missense, nonsense and synonymous variants in plasmid librar - figure supplement 2B
plasmid_freq2 = plasmid_freq2 %>% mutate(start = substr(variant, 1,1),
                                         position = substr(variant, 2,3),
                                         end = substr(variant, 4,4))
plasmid_freq2 = plasmid_freq2 %>% mutate(type = case_when(start == end ~ "synonymous",
                                                          start != end & end != "_" ~ "missense",
                                                          end == "_" ~ "nonsense"))
plasmid_endo_hist = ggplot(data = plasmid_freq2, aes(x= freq_endo, fill = type))+
  geom_histogram(bins = 30)+ scale_x_log10()+ theme(legend.position = "none")+
  facet_grid(rows = vars(type), scale = "free")+NULL; plasmid_endo_hist

plasmid_sko_hist = ggplot(data = plasmid_freq2, aes(x= freq_sko, fill = type))+
  geom_histogram(bins = 30)+ scale_x_log10()+ theme(legend.position = "none")+
  facet_grid(rows = vars(type), scale = "free")+NULL; plasmid_sko_hist

ggsave("plots/supplement/supp2b_plasmid_endo_hist.pdf", plasmid_endo_hist, height = 2, width = 2)
ggsave("plots/supplement/supp2b_plasmid_sko_hist.pdf", plasmid_sko_hist, height = 2, width = 2)


#plasmid library replicate correlation plot - figure supplement 2C
plasmid_cor = log10(plasmid_freq[,c("r1","r2","r3", "r4", "r5")])
colnames(plasmid_cor) = c("R1", "R2", "R3", "R4", "R5")

#library A
plasmid_cor_plot_liba = plasmid_cor %>% select(R1, R2) %>% ggpairs(aes(alpha = 0.15))

#library B
plasmid_cor_plot_libb = plasmid_cor %>% select(R3, R4, R5) %>% ggpairs(aes(alpha = 0.15))

#all 5 replicates
plasmid_cor_plot = ggpairs(plasmid_cor, aes(alpha = 0.15))


ggsave("plots/supplement/supp2c_plasmid_cor_plot.pdf", plasmid_cor_plot, height = 5, width = 5)
ggsave("Revision_1/plots_r1/supplement_r1/supp2c_plasmid_cor_plot_liba.pdf", plasmid_cor_plot_liba, height = 2.5, width = 2.5)
ggsave("Revision_1/plots_r1/supplement_r1/supp2c_plasmid_cor_plot_libb.pdf", plasmid_cor_plot_libb, height = 2.5, width = 2.5)
```

```{r simulation to determine cells required for full coverage of library}
singlemut_uniq_number <- length(unique(plasmid_freq2$variant))
elib_sampling_frame <- data.frame("cells" = c(1e4,3e4,1e5,3e5,1e6,3e6), "fraction_observed" = 0)
slib_sampling_frame <- data.frame("cells" = c(1e4,3e4,1e5,3e5,1e6,3e6), "fraction_observed" = 0)
plasmid_freq3 <- lapply(plasmid_freq2, function(x) replace(x, is.nan(x) | is.na(x), 0))

for(x in 1:nrow(elib_sampling_frame)){
  elib_sampling <- sample(plasmid_freq3$variant, elib_sampling_frame$cells[x], prob = plasmid_freq3$freq_endo, replace = T)
  elib_sampling_df <- data.frame(table(elib_sampling))
  elib_sampling_df2 <- subset(elib_sampling_df, Freq >= 10) # Whatever cutoff you want to use
  elib_sampling_frame$fraction_observed[x] <- nrow(elib_sampling_df2) / singlemut_uniq_number
}

endo_summary_frame <- data.frame("cells" = elib_sampling_frame$cells, "endo" = elib_sampling_frame$fraction_observed)


for(x in 1:nrow(slib_sampling_frame)){
  slib_sampling <- sample(plasmid_freq3$variant, slib_sampling_frame$cells[x], prob = plasmid_freq3$freq_sko, replace = T)
  slib_sampling_df <- data.frame(table(slib_sampling))
  slib_sampling_df2 <- subset(slib_sampling_df, Freq >= 10) # Whatever cutoff you want to use
  slib_sampling_frame$fraction_observed[x] <- nrow(slib_sampling_df2) / singlemut_uniq_number
}

sko_summary_frame <- data.frame("cells" = slib_sampling_frame$cells, "sko" = slib_sampling_frame$fraction_observed)

real_summary_frame = left_join(endo_summary_frame, sko_summary_frame)

#simulation of how many cells need to be recombined to see a variant at least 10 times - figure supplement 2D
lib_simulation = ggplot() + 
  labs(y = "Fraction of possible variants\nrecombined at least 10 times", x = "Number of cells recombined") +
  scale_x_log10(labels = scales::scientific, breaks= c(1e4,3e4,1e5,3e5,1e6, 3e6)) + scale_y_continuous(limits = c(0.2,1.0), expand = c(0,0.01))+
  geom_line(data = real_summary_frame, aes(x = cells, y = endo), color = "red") + geom_point(data = real_summary_frame, aes(cells, endo), color = "red") +
  geom_line(data = real_summary_frame, aes(x = cells, y = sko), color = "green") + geom_point(data = real_summary_frame, aes(cells, sko), color = "green")+NULL; lib_simulation 

ggsave("plots/supplement/supp2d_lib_simulation.pdf", lib_simulation, height = 2, width = 3)
```
Key: endo = standard LP cells, sko = knockout LP cells
## Looking at standard LP cell DMS data and turning variant counts into frequencies
```{r counts}
## Import raw standard LP DMS counts
endo_p = plasmid_freq2 %>% dplyr::select(c("variant", "lib1", "lib2"))
endo_raw_counts = read.csv("data/endo_raw_cells.csv", header = TRUE)
endo_raw_counts = subset(endo_raw_counts, select = -X )
endo_raw_counts = left_join(endo_p, endo_raw_counts)

## Figure out the total number of reads per sample
endo_readcounts <- data.frame("total_count" = colSums(endo_raw_counts[,2:ncol(endo_raw_counts)], na.rm = T))

## Plot the number of total reads we have per sample
ggplot() + scale_x_log10(limits = c(1e5,1e7)) + 
  geom_histogram(data = endo_readcounts, aes(x = total_count), bins = 20, color = "black", fill = "grey80")
## Libraries have lower total reads (presumably because of Amp-EZ); genomic DNA sequencing had generally comparable total reads

## figure out read counts that should service as limits of detection for each sample
endo_lod_readcounts <- c()
endo_lod_freqs <- c()
for(re in endo_readcounts$total_count){
  temp_subset <- (lod_frame %>% mutate(abs_diff = abs(reads - re)) %>% arrange(abs_diff))[1,]
  endo_lod_readcounts <- c(endo_lod_readcounts, temp_subset$count)
  endo_lod_freqs <- c(endo_lod_freqs, temp_subset$freq)
}
endo_lod_dataframe <- data.frame("reads" = endo_readcounts, "count" = endo_lod_readcounts, "freq" = endo_lod_freqs)

## Now figure out which data points were below the limit of detection
endo_raw_counts[is.na(endo_raw_counts)] <- 0

endo_raw_counts2 <- data.frame("variant" = endo_raw_counts$variant,
                           "lib1" = endo_raw_counts$lib1 < endo_lod_readcounts[1],
                           "lib2" = endo_raw_counts$lib2 < endo_lod_readcounts[2],
                           "r1tp1" = endo_raw_counts$r1tp1 < endo_lod_readcounts[3],
                           "r1tp2" = endo_raw_counts$r1tp2 < endo_lod_readcounts[4],
                           "r1tp3" = endo_raw_counts$r1tp3 < endo_lod_readcounts[5],
                           "r2tp1" = endo_raw_counts$r2tp1 < endo_lod_readcounts[6],
                           "r2tp2" = endo_raw_counts$r2tp2 < endo_lod_readcounts[7],
                           "r2tp3" = endo_raw_counts$r2tp3 < endo_lod_readcounts[8],
                           "r4tp1" = endo_raw_counts$r4tp1 < endo_lod_readcounts[9],
                           "r4tp2" = endo_raw_counts$r4tp2 < endo_lod_readcounts[10],
                           "r4tp3" = endo_raw_counts$r4tp3 < endo_lod_readcounts[11],
                           "r5tp1" = endo_raw_counts$r5tp1 < endo_lod_readcounts[12],
                           "r5tp2" = endo_raw_counts$r5tp2 < endo_lod_readcounts[13],
                           "r5tp3" = endo_raw_counts$r5tp3 < endo_lod_readcounts[14],
                           "r6tp1" = endo_raw_counts$r6tp1 < endo_lod_readcounts[15],
                           "r6tp2" = endo_raw_counts$r6tp2 < endo_lod_readcounts[16],
                           "r6tp3" = endo_raw_counts$r6tp3 < endo_lod_readcounts[17]
                           )

## These were the total number of variants below the limit of detection in each endo sequencing run:
colSums(endo_raw_counts2[,c("r1tp1","r1tp2","r1tp3","r2tp1","r2tp2","r2tp3","r4tp1","r4tp2","r4tp3","r5tp1","r5tp2","r5tp3","r6tp1","r6tp2","r6tp3")])

## Now calculate endo frequencies, and for endos below the limit of detection, give them the limit of detection frequency value
endo_freq = endo_raw_counts %>% mutate(f_lib1 = lib1/endo_readcounts$total_count[1], 
                                      f_lib2 = lib2/endo_readcounts$total_count[2], 
                                      f_r1tp1 = r1tp1/endo_readcounts$total_count[3],
                                      f_r1tp2 = r1tp2/endo_readcounts$total_count[4],
                                      f_r1tp3 = r1tp3/endo_readcounts$total_count[5],
                                      f_r2tp1 = r2tp1/endo_readcounts$total_count[6],
                                      f_r2tp2 = r2tp2/endo_readcounts$total_count[7],
                                      f_r2tp3 = r2tp3/endo_readcounts$total_count[8],
                                      f_r4tp1 = r4tp1/endo_readcounts$total_count[9],
                                      f_r4tp2 = r4tp2/endo_readcounts$total_count[10],
                                      f_r4tp3 = r4tp3/endo_readcounts$total_count[11],
                                      f_r5tp1 = r5tp1/endo_readcounts$total_count[12],
                                      f_r5tp2 = r5tp2/endo_readcounts$total_count[13],
                                      f_r5tp3 = r5tp3/endo_readcounts$total_count[14],
                                      f_r6tp1 = r6tp1/endo_readcounts$total_count[15],
                                      f_r6tp2 = r6tp2/endo_readcounts$total_count[16],
                                      f_r6tp3 = r6tp3/endo_readcounts$total_count[17]
                                      )

endo_freq[endo_raw_counts2$r1tp1 == T,"f_r1tp1"] <- endo_lod_freqs[3]
endo_freq[endo_raw_counts2$r1tp2 == T,"f_r1tp2"] <- endo_lod_freqs[4]
endo_freq[endo_raw_counts2$r1tp3 == T,"f_r1tp3"] <- endo_lod_freqs[5]
endo_freq[endo_raw_counts2$r2tp1 == T,"f_r2tp1"] <- endo_lod_freqs[6]
endo_freq[endo_raw_counts2$r2tp2 == T,"f_r2tp2"] <- endo_lod_freqs[7]
endo_freq[endo_raw_counts2$r2tp3 == T,"f_r2tp3"] <- endo_lod_freqs[8]
endo_freq[endo_raw_counts2$r4tp1 == T,"f_r4tp1"] <- endo_lod_freqs[9]
endo_freq[endo_raw_counts2$r4tp2 == T,"f_r4tp2"] <- endo_lod_freqs[10]
endo_freq[endo_raw_counts2$r4tp3 == T,"f_r4tp3"] <- endo_lod_freqs[11]
endo_freq[endo_raw_counts2$r5tp1 == T,"f_r5tp1"] <- endo_lod_freqs[12]
endo_freq[endo_raw_counts2$r5tp2 == T,"f_r5tp2"] <- endo_lod_freqs[13]
endo_freq[endo_raw_counts2$r5tp3 == T,"f_r5tp3"] <- endo_lod_freqs[14]
endo_freq[endo_raw_counts2$r6tp1 == T,"f_r6tp1"] <- endo_lod_freqs[15]
endo_freq[endo_raw_counts2$r6tp2 == T,"f_r6tp2"] <- endo_lod_freqs[16]
endo_freq[endo_raw_counts2$r6tp3 == T,"f_r6tp3"] <- endo_lod_freqs[17]

endo_freq[endo_raw_counts2$lib1 == T,c("f_lib1","f_lib2","f_r1tp1","f_r1tp2","f_r1tp3","f_r2tp1","f_r2tp2","f_r2tp3","f_r4tp1","f_r4tp2","f_r4tp3","f_r5tp1","f_r5tp2","f_r5tp3","f_r6tp1","f_r6tp2","f_r6tp3")] <- NA
endo_freq[endo_raw_counts2$lib2 == T,c("f_lib1","f_lib2","f_r1tp1","f_r1tp2","f_r1tp3","f_r2tp1","f_r2tp2","f_r2tp3","f_r4tp1","f_r4tp2","f_r4tp3","f_r5tp1","f_r5tp2","f_r5tp3","f_r6tp1","f_r6tp2","f_r6tp3")] <- NA

endo_freq$endo_frac_lod <- rowSums(endo_raw_counts2[,c("r1tp1","r1tp2","r1tp3","r2tp1","r2tp2","r2tp3","r4tp1","r4tp2","r4tp3","r5tp1","r5tp2","r5tp3","r6tp1","r6tp2","r6tp3")])/length(c("r1tp1","r1tp2","r1tp3","r2tp1","r2tp2","r2tp3","r4tp1","r4tp2","r4tp3","r5tp1","r5tp2","r5tp3","r6tp1","r6tp2","r6tp3"))


## Take geometric means of the variant frequencies for the two preps
endo_freq$lib_freq <- 10^rowMeans(log10(endo_freq[,c("f_lib1","f_lib2")]))

endo_freq$r1tp1_ratio <- endo_freq$f_r1tp1 / endo_freq$lib_freq
endo_freq$r1tp2_ratio <- endo_freq$f_r1tp2 / endo_freq$lib_freq
endo_freq$r1tp3_ratio <- endo_freq$f_r1tp3 / endo_freq$lib_freq
endo_freq$r1_ratio <- 10^rowMeans(log10(endo_freq[,c("r1tp1_ratio","r1tp2_ratio","r1tp3_ratio")]))
endo_freq$r2tp1_ratio <- endo_freq$f_r1tp1 / endo_freq$lib_freq
endo_freq$r2tp2_ratio <- endo_freq$f_r2tp2 / endo_freq$lib_freq
endo_freq$r2tp3_ratio <- endo_freq$f_r2tp3 / endo_freq$lib_freq
endo_freq$r2_ratio <- 10^rowMeans(log10(endo_freq[,c("r2tp1_ratio","r2tp2_ratio","r2tp3_ratio")]))
endo_freq$r4tp1_ratio <- endo_freq$f_r4tp1 / endo_freq$lib_freq
endo_freq$r4tp2_ratio <- endo_freq$f_r4tp2 / endo_freq$lib_freq
endo_freq$r4tp3_ratio <- endo_freq$f_r4tp3 / endo_freq$lib_freq
endo_freq$r4_ratio <- 10^rowMeans(log10(endo_freq[,c("r4tp1_ratio","r4tp2_ratio","r4tp3_ratio")]))
endo_freq$r5tp1_ratio <- endo_freq$f_r5tp1 / endo_freq$lib_freq
endo_freq$r5tp2_ratio <- endo_freq$f_r5tp2 / endo_freq$lib_freq
endo_freq$r5tp3_ratio <- endo_freq$f_r5tp3 / endo_freq$lib_freq
endo_freq$r5_ratio <- 10^rowMeans(log10(endo_freq[,c("r5tp1_ratio","r5tp2_ratio","r5tp3_ratio")]))
endo_freq$r6tp1_ratio <- endo_freq$f_r6tp1 / endo_freq$lib_freq
endo_freq$r6tp2_ratio <- endo_freq$f_r6tp2 / endo_freq$lib_freq
endo_freq$r6tp3_ratio <- endo_freq$f_r6tp3 / endo_freq$lib_freq
endo_freq$r6_ratio <- 10^rowMeans(log10(endo_freq[,c("r6tp1_ratio","r6tp2_ratio","r6tp3_ratio")]))

endo_freq$geo_ratio <- 10^rowMeans(log10(endo_freq[,c("r1_ratio","r2_ratio","r4_ratio","r5_ratio","r6_ratio")]))

endo_freq$position <- substr(endo_freq$variant,2,3)
endo_freq$start <- substr(endo_freq$variant,1,1)
endo_freq$end <- substr(endo_freq$variant,4,4)
```
```{r standard LP cell survival scores}
endo_wt = endo_freq %>% filter(variant == "Z0Z")
endo_raw_freq_ratio = endo_freq %>% mutate(start = substr(variant, 1,1), end = substr(variant, 4,4), position = substr(variant, 2,3))
endo_score = endo_raw_freq_ratio %>% mutate(type = case_when(end == start ~ "syn",
                                                                      end == "_" ~ "non", 
                                                                      end != start & end != "_" ~ "mis")) %>% dplyr::select("variant", "r1_ratio", "r2_ratio", "r4_ratio", "r5_ratio", "r6_ratio", "start", "end", "position", "type", "endo_frac_lod")

endo_score$mean_ratio <- 10^rowMeans(log10(endo_score[,c("r1_ratio","r2_ratio","r4_ratio","r5_ratio","r6_ratio")]), na.rm = T)
endo_score$rep_count <- 5 - rowSums(is.na(endo_score[,c("r1_ratio","r2_ratio","r4_ratio","r5_ratio","r6_ratio")]))

endo_score_3plusreps <- endo_score %>% filter(rep_count >= 3)

endo_score_3plusreps_log = endo_score_3plusreps %>% mutate(r1_log = log10(r1_ratio),
                                                         r2_log = log10(r2_ratio),
                                                         r4_log = log10(r4_ratio),
                                                         r5_log = log10(r5_ratio),
                                                         r6_log = log10(r6_ratio))

endo_cor = endo_score_3plusreps_log %>% select(r1_log, r2_log, r4_log, r5_log, r6_log)
colnames(endo_cor) = c("R1", "R2", "R3", "R4", "R5")
endo_cor_plot = endo_cor %>% ggpairs(aes(alpha = 0.10))
ggsave("plots/supplement/supp3a_endo_cor_plot.pdf", endo_cor_plot, height = 4, width = 5 )

```

## Knockout LP cell DMS data
```{r counts}
## Import raw knockout LP DMS counts
sko_p = plasmid_freq2 %>% dplyr::select(c("variant", "lib3", "lib4", "lib5"))
sko_raw_counts <- read.csv(file = "data/raw_counts_sko.csv", header = T, stringsAsFactors = F)
sko_raw_counts <-  subset(sko_raw_counts, select = -X )
sko_raw_counts = left_join(sko_p, sko_raw_counts)

## Figure out the total number of reads per sample
sko_readcounts <- data.frame("total_count" = colSums(sko_raw_counts[,2:ncol(sko_raw_counts)], na.rm = T))

## Plot the number of total reads we have per sample
ggplot() + scale_x_log10(limits = c(1e5,1e7)) + 
  geom_histogram(data = sko_readcounts, aes(x = total_count), bins = 20, color = "black", fill = "grey80")
## Libraries have lower total reads (presumably because of Amp-EZ); genomic DNA sequencing had generally comparable total reads

## figure out read counts that should service as limits of detection for each sample
sko_lod_readcounts <- c()
sko_lod_freqs <- c()
for(re in sko_readcounts$total_count){
  temp_subset <- (lod_frame %>% mutate(abs_diff = abs(reads - re)) %>% arrange(abs_diff))[1,]
  sko_lod_readcounts <- c(sko_lod_readcounts, temp_subset$count)
  sko_lod_freqs <- c(sko_lod_freqs, temp_subset$freq)
}
sko_lod_dataframe <- data.frame("reads" = sko_readcounts, "count" = sko_lod_readcounts, "freq" = sko_lod_freqs)

## Now figure out which data points were below the limit of detection
sko_raw_counts[is.na(sko_raw_counts)] <- 0

sko_raw_counts2 <- data.frame("variant" = sko_raw_counts$variant,
                           "lib3" = sko_raw_counts$lib3 < sko_lod_readcounts[1],
                           "lib4" = sko_raw_counts$lib4 < sko_lod_readcounts[2],
                           "lib5" = sko_raw_counts$lib5 < sko_lod_readcounts[3],
                           "r1tp1" = sko_raw_counts$r1tp1 < sko_lod_readcounts[4],
                           "r1tp2" = sko_raw_counts$r1tp2 < sko_lod_readcounts[5],
                           "r1tp3" = sko_raw_counts$r1tp3 < sko_lod_readcounts[6],
                           "r2tp1" = sko_raw_counts$r2tp1 < sko_lod_readcounts[7],
                           "r2tp2" = sko_raw_counts$r2tp2 < sko_lod_readcounts[8],
                           "r2tp3" = sko_raw_counts$r2tp3 < sko_lod_readcounts[9],
                           "r3tp1" = sko_raw_counts$r3tp1 < sko_lod_readcounts[10],
                           "r3tp3" = sko_raw_counts$r3tp3 < sko_lod_readcounts[11],
                           "r4tp1" = sko_raw_counts$r4tp1 < sko_lod_readcounts[12],
                           "r4tp2" = sko_raw_counts$r4tp2 < sko_lod_readcounts[13],
                           "r5tp1" = sko_raw_counts$r5tp1 < sko_lod_readcounts[14],
                           "r5tp2" = sko_raw_counts$r5tp2 < sko_lod_readcounts[15],
                           "r5tp3" = sko_raw_counts$r5tp3 < sko_lod_readcounts[16],
                           "r6tp1" = sko_raw_counts$r6tp1 < sko_lod_readcounts[17],
                           "r6tp2" = sko_raw_counts$r6tp2 < sko_lod_readcounts[18],
                           "r6tp3" = sko_raw_counts$r6tp3 < sko_lod_readcounts[19]
                           )

## These were the total number of variants below the limit of detection in each sko sequencing run:
colSums(sko_raw_counts2[,c("r1tp1","r1tp2","r1tp3","r2tp1","r2tp2","r2tp3","r3tp1","r3tp3","r4tp1","r4tp2","r5tp1","r5tp2","r5tp3","r6tp1","r6tp2","r6tp3")])

## Now calculate sko frequencies, and for skos below the limit of detection, give them the limit of detection frequency value
sko_freq = sko_raw_counts %>% mutate(f_lib3 = lib3/sko_readcounts$total_count[1], 
                                     f_lib4 = lib4/sko_readcounts$total_count[2], 
                                     f_lib5 = lib5/sko_readcounts$total_count[3], 
                                     f_r1tp1 = r1tp1/sko_readcounts$total_count[4],
                                     f_r1tp2 = r1tp2/sko_readcounts$total_count[5],
                                     f_r1tp3 = r1tp3/sko_readcounts$total_count[6],
                                     f_r2tp1 = r2tp1/sko_readcounts$total_count[7],
                                     f_r2tp2 = r2tp2/sko_readcounts$total_count[8],
                                     f_r2tp3 = r2tp3/sko_readcounts$total_count[9],
                                     f_r3tp1 = r3tp1/sko_readcounts$total_count[10],
                                     f_r3tp3 = r3tp3/sko_readcounts$total_count[11],
                                     f_r4tp1 = r4tp1/sko_readcounts$total_count[12],
                                     f_r4tp2 = r4tp2/sko_readcounts$total_count[13],
                                     f_r5tp1 = r5tp1/sko_readcounts$total_count[14],
                                     f_r5tp2 = r5tp2/sko_readcounts$total_count[15],
                                     f_r5tp3 = r5tp3/sko_readcounts$total_count[16],
                                     f_r6tp1 = r6tp1/sko_readcounts$total_count[17],
                                     f_r6tp2 = r6tp2/sko_readcounts$total_count[18],
                                     f_r6tp3 = r6tp3/sko_readcounts$total_count[19]
                                     )

sko_freq[sko_raw_counts2$r1tp1 == T,"f_r1tp1"] <- sko_lod_freqs[4]
sko_freq[sko_raw_counts2$r1tp2 == T,"f_r1tp2"] <- sko_lod_freqs[5]
sko_freq[sko_raw_counts2$r1tp3 == T,"f_r1tp3"] <- sko_lod_freqs[6]
sko_freq[sko_raw_counts2$r2tp1 == T,"f_r2tp1"] <- sko_lod_freqs[7]
sko_freq[sko_raw_counts2$r2tp2 == T,"f_r2tp2"] <- sko_lod_freqs[8]
sko_freq[sko_raw_counts2$r2tp3 == T,"f_r2tp3"] <- sko_lod_freqs[9]
sko_freq[sko_raw_counts2$r3tp1 == T,"f_r3tp1"] <- sko_lod_freqs[10]
sko_freq[sko_raw_counts2$r3tp3 == T,"f_r3tp3"] <- sko_lod_freqs[11]
sko_freq[sko_raw_counts2$r4tp1 == T,"f_r4tp1"] <- sko_lod_freqs[12]
sko_freq[sko_raw_counts2$r4tp2 == T,"f_r4tp2"] <- sko_lod_freqs[13]
sko_freq[sko_raw_counts2$r5tp1 == T,"f_r5tp1"] <- sko_lod_freqs[14]
sko_freq[sko_raw_counts2$r5tp2 == T,"f_r5tp2"] <- sko_lod_freqs[15]
sko_freq[sko_raw_counts2$r5tp3 == T,"f_r5tp3"] <- sko_lod_freqs[16]
sko_freq[sko_raw_counts2$r6tp1 == T,"f_r6tp1"] <- sko_lod_freqs[17]
sko_freq[sko_raw_counts2$r6tp2 == T,"f_r6tp2"] <- sko_lod_freqs[18]
sko_freq[sko_raw_counts2$r6tp3 == T,"f_r6tp3"] <- sko_lod_freqs[19]

sko_freq[sko_raw_counts2$lib3 == T,c("f_lib3","f_lib4","f_lib5","f_r1tp1","f_r1tp2","f_r1tp3","f_r2tp1","f_r2tp2","f_r2tp3","f_r3tp1","f_r3tp3","f_r4tp1","f_r4tp2","f_r5tp1","f_r5tp2","f_r5tp3","f_r6tp1","f_r6tp2","f_r6tp3")] <- NA
sko_freq[sko_raw_counts2$lib4 == T,c("f_lib3","f_lib4","f_lib5","f_r1tp1","f_r1tp2","f_r1tp3","f_r2tp1","f_r2tp2","f_r2tp3","f_r3tp1","f_r3tp3","f_r4tp1","f_r4tp2","f_r5tp1","f_r5tp2","f_r5tp3","f_r6tp1","f_r6tp2","f_r6tp3")] <- NA
sko_freq[sko_raw_counts2$lib5 == T,c("f_lib3","f_lib4","f_lib5","f_r1tp1","f_r1tp2","f_r1tp3","f_r2tp1","f_r2tp2","f_r2tp3","f_r3tp1","f_r3tp3","f_r4tp1","f_r4tp2","f_r5tp1","f_r5tp2","f_r5tp3","f_r6tp1","f_r6tp2","f_r6tp3")] <- NA

sko_freq$sko_frac_lod <- rowSums(sko_raw_counts2[,c("r1tp1","r1tp2","r1tp3","r2tp1","r2tp2","r2tp3","r3tp1","r3tp3","r4tp1","r4tp2","r5tp1","r5tp2","r5tp3","r6tp1","r6tp2","r6tp3")])/length(c("r1tp1","r1tp2","r1tp3","r2tp1","r2tp2","r2tp3","r3tp1","r3tp3","r4tp1","r4tp2","r5tp1","r5tp2","r5tp3","r6tp1","r6tp2","r6tp3"))

## Take geometric means of the variant frequencies for the two preps
sko_freq$lib_freq <- 10^rowMeans(log10(sko_freq[,c("f_lib3","f_lib4","f_lib5")]))

sko_freq$r1tp1_ratio <- sko_freq$f_r1tp1 / sko_freq$lib_freq
sko_freq$r1tp2_ratio <- sko_freq$f_r1tp2 / sko_freq$lib_freq
sko_freq$r1tp3_ratio <- sko_freq$f_r1tp3 / sko_freq$lib_freq
sko_freq$r1_ratio <- 10^rowMeans(log10(sko_freq[,c("r1tp1_ratio","r1tp2_ratio","r1tp3_ratio")]))
sko_freq$r2tp1_ratio <- sko_freq$f_r1tp1 / sko_freq$lib_freq
sko_freq$r2tp2_ratio <- sko_freq$f_r2tp2 / sko_freq$lib_freq
sko_freq$r2tp3_ratio <- sko_freq$f_r2tp3 / sko_freq$lib_freq
sko_freq$r2_ratio <- 10^rowMeans(log10(sko_freq[,c("r2tp1_ratio","r2tp2_ratio","r2tp3_ratio")]))
sko_freq$r3tp1_ratio <- sko_freq$f_r3tp1 / sko_freq$lib_freq
sko_freq$r3tp3_ratio <- sko_freq$f_r3tp3 / sko_freq$lib_freq
sko_freq$r3_ratio <- 10^rowMeans(log10(sko_freq[,c("r3tp1_ratio","r3tp3_ratio")]))
sko_freq$r4tp1_ratio <- sko_freq$f_r4tp1 / sko_freq$lib_freq
sko_freq$r4tp2_ratio <- sko_freq$f_r4tp2 / sko_freq$lib_freq
sko_freq$r4_ratio <- 10^rowMeans(log10(sko_freq[,c("r4tp1_ratio","r4tp2_ratio")]))
sko_freq$r5tp1_ratio <- sko_freq$f_r5tp1 / sko_freq$lib_freq
sko_freq$r5tp2_ratio <- sko_freq$f_r5tp2 / sko_freq$lib_freq
sko_freq$r5tp3_ratio <- sko_freq$f_r5tp3 / sko_freq$lib_freq
sko_freq$r5_ratio <- 10^rowMeans(log10(sko_freq[,c("r5tp1_ratio","r5tp2_ratio","r5tp3_ratio")]))
sko_freq$r6tp1_ratio <- sko_freq$f_r6tp1 / sko_freq$lib_freq
sko_freq$r6tp2_ratio <- sko_freq$f_r6tp2 / sko_freq$lib_freq
sko_freq$r6tp3_ratio <- sko_freq$f_r6tp3 / sko_freq$lib_freq
sko_freq$r6_ratio <- 10^rowMeans(log10(sko_freq[,c("r6tp1_ratio","r6tp2_ratio","r6tp3_ratio")]))

sko_freq$geo_ratio <- 10^rowMeans(log10(sko_freq[,c("r1_ratio","r2_ratio","r3_ratio","r4_ratio","r5_ratio","r6_ratio")]))

sko_freq$position <- substr(sko_freq$variant,2,3)
sko_freq$start <- substr(sko_freq$variant,1,1)
sko_freq$end <- substr(sko_freq$variant,4,4)

```

```{r knockout LP cell survival scores}
sko_wt = sko_freq %>% filter(variant == "Z0Z")
sko_raw_freq_ratio = sko_freq %>% mutate(start = substr(variant, 1,1), end = substr(variant, 4,4), position = substr(variant, 2,3))
sko_score = sko_raw_freq_ratio %>% mutate(type = case_when(end == start ~ "syn",
                                                                      end == "_" ~ "non", 
                                                                      end != start & end != "_" ~ "mis")) %>% dplyr::select("variant", "r1_ratio", "r2_ratio", "r3_ratio" ,"r4_ratio", "r5_ratio", "r6_ratio", "start", "end", "position", "type","sko_frac_lod")

sko_score$mean_ratio <- 10^rowMeans(log10(sko_score[,c("r1_ratio","r2_ratio","r3_ratio", "r4_ratio","r5_ratio","r6_ratio")]), na.rm = T)
sko_score$rep_count <- 5 - rowSums(is.na(sko_score[,c("r1_ratio","r2_ratio", "r3_ratio",
                                                      "r4_ratio","r5_ratio","r6_ratio")]))

sko_score_3plusreps <- sko_score %>% filter(rep_count >= 3)

sko_score_3plusreps_log = sko_score_3plusreps %>% mutate(r1_log = log10(r1_ratio),
                                                         r2_log = log10(r2_ratio),
                                                         r3_log = log10(r3_ratio),
                                                         r4_log = log10(r4_ratio),
                                                         r5_log = log10(r5_ratio),
                                                         r6_log = log10(r6_ratio))

sko_cor = sko_score_3plusreps_log %>% select(r1_log, r2_log, r3_log, r4_log, r5_log, r6_log) 
colnames(sko_cor) = c("R1", "R2", "R3", "R4", "R5", "r6")
sko_cor_plot = sko_cor %>% ggpairs(aes(alpha = 0.10))
ggsave("plots/supplement/supp3b_sko_cor_plot.pdf", sko_cor_plot, height = 4, width = 6)
```

## Great, so now we have enrichment scores for experiments done in standard and knockout LP cells
```{r Combining all existing DMS scores}

sum(!is.na(endo_score$mean_ratio))
sum(!is.na(sko_score$mean_ratio))

var_endo = na.omit(endo_score) %>% select("variant") %>% mutate(endo = "yes")
var_sko = na.omit(sko_score) %>% select("variant") %>% mutate(sko = "yes")

var_both = left_join(variant_name, var_endo)
var_both = left_join(var_both, var_sko)
var_both = na.omit(var_both)

## Combining all existing data
combined <- merge(endo_score[,c("variant","position","start","end","type","mean_ratio","endo_frac_lod")], sko_score[,c("variant","mean_ratio","sko_frac_lod")], by = "variant", all = T) %>% filter(!is.na(type))
colnames(combined)[colnames(combined) == "mean_ratio.x"] <- "endo_ratio"
colnames(combined)[colnames(combined) == "mean_ratio.y"] <- "sko_ratio"

combined = replace(combined, combined == 0, NA)

```

## Bring in the validation data next
```{r SSV Validation counts and analysis - standard LP}
## Import raw standard LP SSV counts
endo_ssv <- read.csv(file = "data/endo_ssv_raw.csv", header = T, stringsAsFactors = F)
endo_ssv_freq <- data.frame(apply(endo_ssv[,2:ncol(endo_ssv)],2,function(x){x/sum(x)}))
endo_ssv_freq$variant <- endo_ssv$variant
endo_ssv_freq$lib = (((endo_ssv_freq$lib_1) + (endo_ssv_freq$lib_2))/2)
endo_ssv_freq = endo_ssv_freq %>% dplyr::select(c("variant", "lib", "r1tp1", "r1tp2", "r1tp3", "r2tp1", "r2tp2", "r2tp3", "r3tp1", "r3tp2", "r3tp3"))

## Figure out the total number of reads per sample
endo_ssv_colsums <- data.frame("total_count" = colSums(endo_ssv[,2:ncol(endo_ssv)]))

## Plot the number of total reads we have per sample
ggplot() + scale_x_log10(limits = c(1e5,1e7)) + 
  geom_histogram(data = endo_ssv_colsums, aes(x = total_count), bins = 20, color = "black", fill = "grey80")
## Libraries have lower total reads (presumably because of Amp-EZ); genomic DNA sequencing had generally comparable total reads

## Confirm that frequencies were adequately calculated
colSums(endo_ssv_freq[,2:ncol(endo_ssv_freq)])

## Now we can add the column labels back
endo_ssv_freq2 <- endo_ssv_freq

endo_ssv_freq3 <- data.frame("variant" = endo_ssv_freq2$variant)

endo_ssv_freq3$plasmid <- endo_ssv_freq2$lib
endo_ssv_freq3$r1 <- rowMeans(endo_ssv_freq2[,c("r1tp1","r1tp2","r1tp3")], na.rm = F)
endo_ssv_freq3$r2 <- rowMeans(endo_ssv_freq2[,c("r2tp1","r2tp2","r2tp3")], na.rm= F)
endo_ssv_freq3$r3 <- rowMeans(endo_ssv_freq2[,c("r3tp1", "r3tp3")], na.rm= F)

endo_ssv_freq_ratio <- data.frame("variant" = endo_ssv_freq2$variant)

endo_ssv_freq_ratio$r1_ratio <- endo_ssv_freq3$r1 / endo_ssv_freq3$plasmid
endo_ssv_freq_ratio$r2_ratio <- endo_ssv_freq3$r2 / endo_ssv_freq3$plasmid
endo_ssv_freq_ratio$r3_ratio <- endo_ssv_freq3$r3 / endo_ssv_freq3$plasmid

endo_ssv_freq_ratio$r1_ratio_log10 <- log10(endo_ssv_freq_ratio$r1_ratio)
endo_ssv_freq_ratio$r2_ratio_log10 <- log10(endo_ssv_freq_ratio$r2_ratio)
endo_ssv_freq_ratio$r3_ratio_log10 <- log10(endo_ssv_freq_ratio$r3_ratio)


endo_ssv_cor = endo_ssv_freq_ratio %>% dplyr::select(c("r1_ratio_log10", "r2_ratio_log10", "r3_ratio_log10"))
endo_ssv_cor = endo_ssv_cor[!is.infinite(rowSums(endo_ssv_cor)),]
colnames(endo_ssv_cor) = c("R1", "R2", "R3")

endo_ssv_cor_plot = endo_ssv_cor %>% ggpairs()
endo_ssv_cor_plot

endo_ssv_freq_ratio = endo_ssv_freq_ratio %>% mutate(start = substr(variant, 1,1), end = substr(variant, 4,4), position = substr(variant, 2,3)) #%>% filter(variant != "Z0Z")
endo_ssv_score = endo_ssv_freq_ratio %>% mutate(type = case_when(end == start ~ "syn",
                                                                      end == "_" ~ "non", 
                                                                      end != start & end != "_" ~ "mis")) %>% dplyr::select("variant", "r1_ratio", "r2_ratio", "r3_ratio", "start", "end", "position", "type")

endo_ssv_score$mean_ratio <- 10^rowMeans(log10(endo_ssv_score[,c("r1_ratio","r2_ratio", "r3_ratio")])) #skipped R3 bc of no tp2
endo_ssv_score$endo_ssv_score = log10(endo_ssv_score$mean_ratio)
```

```{r SSV Validation counts and analysis - knockout LP}
## Import raw knockout LP SSV counts
sko_ssv <- read.csv(file = "data/sko_ssv_raw.csv", header = T, stringsAsFactors = F)
sko_ssv_freq <- data.frame(apply(sko_ssv[,2:ncol(sko_ssv)],2,function(x){x/sum(x)}))
sko_ssv_freq$variant <- sko_ssv$variant
sko_ssv_freq$lib = (((sko_ssv_freq$lib_1) + (sko_ssv_freq$lib_2))/2)

sko_ssv_freq = sko_ssv_freq %>% dplyr::select(c("variant","lib_1","lib_2","lib", "r1tp1", "r1tp2", "r1tp3", "r2tp1", "r2tp2", "r2tp3", "r3tp1", "r3tp2", "r3tp3"))

## Confirm that frequencies were adequately calculated
colSums(sko_ssv_freq[,2:ncol(sko_ssv_freq)])

## Now we can add the column labels back
sko_ssv_freq2 <-sko_ssv_freq

sko_ssv_freq3 <- data.frame("variant" = sko_ssv_freq2$variant)

sko_ssv_freq3$plasmid <- log10(sko_ssv_freq2$lib)
sko_ssv_freq3$r1 <- rowMeans(log10(sko_ssv_freq2[,c("r1tp1","r1tp2","r1tp3")]), na.rm = F)
sko_ssv_freq3$r2 <- rowMeans(log10(sko_ssv_freq2[,c("r2tp1","r2tp2","r2tp3")]), na.rm= F)
sko_ssv_freq3$r3 <- rowMeans(log10(sko_ssv_freq2[,c("r3tp1","r3tp2", "r3tp3")]), na.rm= F)

sko_ssv_freq_ratio <- data.frame("variant" = sko_ssv_freq2$variant)

sko_ssv_freq_ratio$r1_ratio <- 10^(sko_ssv_freq3$r1 - sko_ssv_freq3$plasmid)
sko_ssv_freq_ratio$r2_ratio <- 10^(sko_ssv_freq3$r2 - sko_ssv_freq3$plasmid)
sko_ssv_freq_ratio$r3_ratio <- 10^(sko_ssv_freq3$r3 - sko_ssv_freq3$plasmid)

sko_ssv_freq_ratio$r1_ratio_log10 <- log10(sko_ssv_freq_ratio$r1_ratio)
sko_ssv_freq_ratio$r2_ratio_log10 <- log10(sko_ssv_freq_ratio$r2_ratio)
sko_ssv_freq_ratio$r3_ratio_log10 <- log10(sko_ssv_freq_ratio$r3_ratio)


sko_ssv_cor = sko_ssv_freq_ratio %>% dplyr::select(c("r1_ratio_log10", "r2_ratio_log10", "r3_ratio_log10"))
sko_ssvcor = sko_ssv_cor[!is.infinite(rowSums(sko_ssv_cor)),]
colnames(sko_ssv_cor) = c("R1", "R2", "R3")

sko_ssv_cor_plot = sko_ssv_cor %>% ggpairs()
sko_ssv_cor_plot

sko_ssv_wt = sko_ssv_freq_ratio %>% filter(variant == "Z0Z")
sko_ssv_freq_ratio = sko_ssv_freq_ratio %>% mutate(start = substr(variant, 1,1), end = substr(variant, 4,4), position = substr(variant, 2,3))# %>% filter(variant != "Z0Z")
sko_ssv_score = sko_ssv_freq_ratio %>% mutate(type = case_when(end == start ~ "syn",
                                                                      end == "_" ~ "non", 
                                                                      end != start & end != "_" ~ "mis")) %>% dplyr::select("variant", "r1_ratio", "r2_ratio", "r3_ratio", "start", "end", "position", "type")

sko_ssv_score$mean_ratio <- 10^rowMeans(log10(sko_ssv_score[,c("r1_ratio","r2_ratio", "r3_ratio")]))
sko_ssv_score$sko_ssv_score = log10(sko_ssv_score$mean_ratio)
```

## Add the validation data to the combined dataframe
```{r combined table with dms and validation data}
## All DMS and single variant validation values included so far
combined2 <- merge(merge(combined, sko_ssv_score[,c("variant","mean_ratio")], by = "variant", all = T), endo_ssv_score[,c("variant","mean_ratio")], by = "variant", all = T)
colnames(combined2)[colnames(combined2) %in% c("mean_ratio.x", "mean_ratio.y")] <- c("endo_ssv_ratio","sko_ssv_ratio")

combined2[combined2$variant == "Z0Z","type"] <- "WT"

## Add plasmid frequencies as well
combined2 <- merge(combined2,endo_freq[,c("variant","lib_freq")], by = "variant", all.x = T)
colnames(combined2)[colnames(combined2) == "lib_freq"] = "endo_plasmid"

combined2 <- merge(combined2,sko_freq[,c("variant","lib_freq")], by = "variant", all.x = T)
colnames(combined2)[colnames(combined2) == "lib_freq"] = "sko_plasmid"

#adding depletion in bacteria
## Take geometric means of the variant frequencies for the two preps
plasmid_freq$lib_prep1 <- 10^rowMeans(log10(plasmid_freq[,c("r1","r2")]))
plasmid_freq$lib_prep2 <- 10^rowMeans(log10(plasmid_freq[,c("r3","r4","r5")]))

plasmid_freq$lib_prep_ratio <- plasmid_freq$lib_prep2 / plasmid_freq$lib_prep1

plasmid_freq$position <- substr(plasmid_freq$variant,2,3)
plasmid_freq$start <- substr(plasmid_freq$variant,1,1)
plasmid_freq$end <- substr(plasmid_freq$variant,4,4)
combined2 <- merge(combined2, plasmid_freq[,c("variant","lib_prep_ratio")], all.x = T)
colnames(combined2)[colnames(combined2) == "lib_prep_ratio"] = "bac_ratio"


## Making positional data
combined_positional <- combined2 %>% filter(type == "mis") %>% group_by(position) %>% summarize(endo_positional = 10^mean(log10(endo_ratio), na.rm = T), sko_positional = 10^mean(log10(sko_ratio), na.rm = T), bac_positional = 10^mean(log10(bac_ratio), na.rm = T))
positional_scores = combined2 %>% filter(type == "mis") %>% group_by(position) %>% summarize(endo_avg = 10^mean(log10(endo_ratio), na.rm = T), sko_avg = 10^mean(log10(sko_ratio), na.rm = T), bac_avg = 10^mean(log10(bac_ratio), na.rm = T))
## Add positional data to my variant dataframe
combined2 <- merge(combined2,combined_positional, by = "position")

combined2 = combined2 %>% group_by(position) %>% mutate(endo_lib_positional = mean(endo_plasmid, na.rm=T),
                                                        sko_lib_positional = mean(sko_plasmid, na.rm = T))

#Data frame of survival scores for standard LP, knockout LP and bacteria
write.csv(combined2, "data/dms_scores.csv")
```

```{r Looking at whether SKO scores can be affected by the fact that they were done with the retransformed plasmid library}
combined2$hypo_lowest <- NA
for(x in 1:nrow(combined2)){
  combined2$hypo_lowest[x]
}

ggplot() + scale_x_log10() + scale_y_log10() +
  geom_point(data = combined2, aes(x = sko_plasmid, y = sko_ratio)) +
  geom_point(data = combined2 %>% filter(sko_frac_lod > 2/3), aes(x = sko_plasmid, y = sko_ratio), color = "red") +
  geom_point(data = combined2 %>% filter(position == 70), aes(x = sko_plasmid, y = sko_ratio), color = "green")
             
ggplot() + scale_x_log10() + scale_y_log10() +
  geom_point(data = combined2, aes(x = endo_plasmid, y = endo_ratio)) +
  geom_point(data = combined2 %>% filter(endo_frac_lod > 2/3), aes(x = endo_plasmid, y = endo_ratio), color = "red") +
  geom_point(data = combined2 %>% filter(position == 93), aes(x = endo_plasmid, y = endo_ratio), color = "green")

#marking variant scores that will be inaccurate due to low counts
combined2$endo_inac_score <- "no"
combined2$sko_inac_score <- "no"
for(x in 1:nrow(combined2)){
  if(!is.na(combined2$endo_frac_lod[x])){
    if(combined2$endo_frac_lod[x] > 2/3){
      combined2$endo_inac_score[x] <- "yes"
    }
  }
  if(!is.na(combined2$sko_frac_lod[x])){
    if(combined2$sko_frac_lod[x] > 2/3){
      combined2$sko_inac_score[x] <- "yes"
    }
  }
}

ggplot() + scale_x_log10() + scale_y_log10() +
  geom_point(data = combined2, aes(x = sko_plasmid, y = sko_ratio)) +
  geom_point(data = combined2 %>% filter(sko_inac_score == "yes"), aes(x = sko_plasmid, y = sko_ratio), color = "red") +
  geom_point(data = combined2 %>% filter(position == 93), aes(x = sko_plasmid, y = sko_ratio), color = "green")
             
ggplot() + scale_x_log10() + scale_y_log10() +
  geom_point(data = combined2, aes(x = endo_plasmid, y = endo_ratio)) +
  geom_point(data = combined2 %>% filter(endo_inac_score == "yes"), aes(x = endo_plasmid, y = endo_ratio), color = "red") +
  geom_point(data = combined2 %>% filter(position == 93), aes(x = endo_plasmid, y = endo_ratio), color = "green")
```
## Make some histograms separated by mutation type
```{r Score histogram}
histogram_data = combined2 %>% filter(type %in% c("mis", "syn", "non"))
histogram_data$type[histogram_data$type == "mis"] = "Mis"
histogram_data$type[histogram_data$type == "non"] = "Non"
histogram_data$type[histogram_data$type == "syn"] = "Syn"

histogram_data$type <- factor(histogram_data$type, rev(c("Mis","Non","Syn")))


endo_histogram_plot2 = ggplot() + 
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank()) +
  geom_histogram(data = histogram_data, aes(x = log10(endo_ratio)), alpha = 0.75, color = "black", bins = 10) + 
  labs(x= "Enrichment Score", y = "Count", fill = "Sample", title = "Standard scores") +
  facet_grid(rows = vars(type), scales = "free") +
  theme(legend.position = "none")+
  scale_x_continuous(breaks = pretty_breaks(n = 3)) +  
  scale_y_continuous(breaks = pretty_breaks(n = 3)) +
  NULL; endo_histogram_plot2
ggsave("plots/main/fig3b_endo_histogram_plot2.pdf", endo_histogram_plot2, height = 1.75, width = 1.5)

sko_histogram_plot2 = ggplot() + 
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank()) +
  geom_histogram(data = histogram_data, 
                 aes(x = log10(sko_ratio)), 
                 alpha = 0.75, 
                 color = "black", 
                 bins = 10) + 
  labs(x = "Enrichment Score", 
       y = "Count", 
       title = "Knockout scores") +
  facet_grid(rows = vars(type), scales = "free") + 
  scale_x_continuous(breaks = pretty_breaks(n = 3)) +  
  scale_y_continuous(breaks = pretty_breaks(n = 3)) +
  theme(legend.position = "none")+
  NULL; sko_histogram_plot2
ggsave("plots/main/fig3c_sko_histogram_plot2.pdf", sko_histogram_plot2, height = 1.75, width = 1.5)

bac_histogram_plot2 = ggplot() + 
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(), 
        strip.text = element_blank()) +
  geom_histogram(data = histogram_data, aes(x = log10(bac_ratio)), alpha = 0.75, color = "black", bins = 10) + 
  labs(x= "Score", y = "Count", fill = "Sample", title = NULL) +
  facet_grid(rows = vars(type), scale = "free") +
  scale_x_continuous(breaks = pretty_breaks(n = 3)) +  
  scale_y_continuous(breaks = pretty_breaks(n = 2)) +
  theme(legend.position = "none")+
  NULL; bac_histogram_plot2
ggsave("plots/main/fig4b_bac_histogram_plot2.pdf", bac_histogram_plot2, height = 1.5, width = 1.5)
```

Heatmaps of DMS scores in standard LP, knockout LP and bacteria
```{r Make a dataframe we can use for annotating the heatmaps}

wt_seq <- "LSFEAVRNIHKLMDDDANGDVDVEESDEFLREDLNY"

annotation_df <- data.frame("position" = seq(63,98))
annotation_df$start <- NA
for(x in 1:nrow(annotation_df)){annotation_df$start[x] <- substr(wt_seq,x,x)}

annotation_df2 <- annotation_df

main_chain_coordination_res <- c(82)
side_chain_coordination_res <- c(76, 78, 80, 87)

annotation_df3 <- merge(annotation_df2, data.frame("position" = side_chain_coordination_res, "coord" = 1), all.x = T)
annotation_df3[annotation_df3$position %in% main_chain_coordination_res,"coord"] <- 1.1

helix <- c(seq(64,73),seq(89,97))
annotation_df4 <- merge(annotation_df3, data.frame("position" = helix, "helix" = 1), all.x = T)
annotation_df4[is.na(annotation_df4)] <- 0
annotation_df4$position = as.character(annotation_df4$position)

## Make another dataframe of all posssible variants for helping with data frames of missing values

all_res_list <- c("_","P","C","M","T","S","G","A","V","I","L","F","Y","W","H","K","R","N","Q","D","E")
all_variant_list <- c()

for(x in 1:nrow(annotation_df4)){
  for(y in all_res_list){
    all_variant_list <- c(all_variant_list,temp_start_pos <- paste0(annotation_df4$start[x], annotation_df4$position[x], y))
  }
}

all_variant_df <- data.frame("variant" = all_variant_list)

```

##Standard LP cells
```{r positional patterns}
endo_mis = combined2 %>% filter(type == "mis") %>% select(c("variant", "endo_ratio", "start", "end", "position"))
endo_non = combined2 %>% filter(type == "non") %>% select(c("variant", "endo_ratio", "start", "end", "position"))
endo_syn = combined2 %>% filter(type == "syn") %>% select(c("variant", "endo_ratio", "start", "end", "position"))

endo_positional_ratios <- endo_mis %>% group_by(position) %>% summarize(missense_geomean = mean(endo_ratio), missense_min = min(endo_ratio))
endo_positional_ratios2 <- merge(endo_positional_ratios, endo_non[,c("position","endo_ratio")], by = "position", all.x = T)
endo_positional_ratios3 <- merge(endo_positional_ratios2, endo_syn[,c("position","endo_ratio")], by = "position", all.x = T)
colnames(endo_positional_ratios3) <- c("position", "missense_geomean","missense_min", "nonsense","synonymous")

endo_positional_ratios4 <- merge(endo_positional_ratios3, annots[,c("position","annot")], by = "position")

Missense_nonsense_scatterplot <- ggplot() + scale_x_log10(limits  = c(0.001,5.1)) + scale_y_log10(limits  = c(0.001,5.1)) + 
  geom_point(data = endo_positional_ratios4, aes(x = missense_geomean, y = nonsense, color = annot), alpha = 0.75) +
  geom_text_repel(data = endo_positional_ratios4, aes(x = missense_geomean, y = nonsense, label = position), color = "red", size = 2, segment.color = "orange",
                  segment.alpha = 0.5) +
  theme(panel.grid = element_blank()) + 
  labs(x = "Missense (geomean),\nstandard LP score", y = "Nonsense,\nstandard LP score") +
  NULL; Missense_nonsense_scatterplot
ggsave(file = "Revision_1/plots_r1/supplement_r1/Missense_nonsense_scatterplot.pdf", Missense_nonsense_scatterplot, height = 2, width = 3.5)

paste("Speamean rho, standard LP missense vs nonsense:",round(cor(endo_positional_ratios4$missense_geomean, endo_positional_ratios4$nonsense, method = "spearman", use = "complete"),2))

Missense_synonymous_scatterplot <- ggplot() + scale_x_log10(limits  = c(0.001,5.1)) + scale_y_log10(limits  = c(0.001,5.1)) + 
  geom_point(data = endo_positional_ratios4, aes(x = missense_geomean, y = synonymous, color = annot), alpha = 0.75) +
  geom_text_repel(data = endo_positional_ratios4, aes(x = missense_geomean, y = synonymous, label = position), color = "red", size = 2, segment.color = "orange",
                  segment.alpha = 0.5) +
  theme(panel.grid = element_blank()) + 
  labs(x = "Missense (geomean),\nstandard LP score", y = "Synonymous,\nstandard LP score") +
  NULL; Missense_synonymous_scatterplot
ggsave(file = "Revision_1/plots_r1/supplement_r1/Missense_synonymous_scatterplot.pdf", Missense_synonymous_scatterplot, height = 2, width = 3.5)

paste("Speamean rho, standard LP missense vs synonymous:",round(cor(endo_positional_ratios4$missense_geomean, endo_positional_ratios4$synonymous, method = "spearman", use = "complete"),2))

Nonsense_synonymous_scatterplot <- ggplot() + 
  scale_x_log10(limits  = c(0.001,5.1)) + scale_y_log10(limits  = c(0.001,5.1)) + 
  geom_point(data = endo_positional_ratios4, aes(x = nonsense, y = synonymous, color = annot), alpha = 0.75) +
  geom_text_repel(data = endo_positional_ratios4, aes(x = nonsense, y = synonymous, label = position), color = "red", size = 2, segment.color = "orange",
                  segment.alpha = 0.5) +
  theme(panel.grid = element_blank()) + 
  labs(x = "Nonsense,\nstandard LP score", y = "Synonymous,\nstandard LP score") +
  NULL; Nonsense_synonymous_scatterplot
ggsave(file = "Revision_1/plots_r1/supplement_r1/Nonsense_synonymous_scatterplot.pdf", Nonsense_synonymous_scatterplot, height = 2, width = 3.5)

paste("Speamean rho, standard LP nonsense vs synonymous:",round(cor(endo_positional_ratios4$nonsense, endo_positional_ratios4$synonymous, method = "spearman", use = "complete"),2))
```


```{r Standard LP heatmap}

endo_score_heatmap = combined2 %>% select(c("variant", "position", "start", "end","type", "endo_ratio")) %>% filter(!(position %in% c(0,99)) & !(end %in% c("?")))

## For values below 0.01, set it to 0.01 for the heatmap
for(x in 1:nrow(endo_score_heatmap)){
  if(!is.na(endo_score_heatmap$endo_ratio[x]) & endo_score_heatmap$endo_ratio[x] < 0.01){endo_score_heatmap$endo_ratio[x] <- 0.01}
}

endo_score_heatmap$position = as.character(endo_score_heatmap$position)
endo_score_heatmap = endo_score_heatmap %>% filter(variant != "Z0Z")

endo_score_heatmap$end <- factor(endo_score_heatmap$end, levels = c("_","P","C","M","T","S","G","A","V","I","L","F","Y","W","H","K","R","N","Q","D","E"))

endo_score_heatmap1 <- rbind(endo_score_heatmap,
                         data.frame("position" = annotation_df4$position, "endo_ratio" = 1, "variant" = NA, "start" = annotation_df4$start, "end" = "WT res", "type" = "annot"),
                         data.frame("position" = annotation_df4$position, "endo_ratio" = annotation_df4$coord, "variant" = NA, "start" = "Coord", "end" = "Coord", "type" = "annot"),
                         data.frame("position" = annotation_df4$position, "endo_ratio" = annotation_df4$helix, "variant" = NA, "start" = "Helix", "end" = "Helix", "type" = "annot"), 
                          data.frame("position" = positional_scores$position, "endo_ratio" = positional_scores$endo_avg, "variant" = NA, "r1_ratio" = NA, "r2_ratio" = NA, "r4_ratio" = NA, "r5_ratio" = NA, "r6_ratio" = NA, "start" = "Geomean (mis)", "end" = "Geomean (mis)", "type" = "annot", "rep_count" = NA))

endo_score_heatmap1$end <- factor(endo_score_heatmap1$end, levels = rev(c("Geomean (mis)","Helix","Coord","WT res","_","R","H","K","D","E","S","T","N","Q","A","V","I","L","M","F","Y","W","C","G","P")))
endo_syn$end <- factor(endo_syn$end, levels = rev(c("Geomean (mis)","Helix","Coord","WT res","_","R","H","K","D","E","S","T","N","Q","A","V","I","L","M","F","Y","W","C","G","P")))

heatmap_endo <- ggplot() + labs(x = NULL, y = NULL) +
  scale_fill_gradient2(low = "blue", mid = "white", high = "red", na.value = "grey") +
  geom_tile(data = endo_score_heatmap1 %>% filter(position != "0Z"), aes(x = position, y = end, fill = log10(endo_ratio))) +
  geom_point(data = endo_syn, aes(x = position, y = end), size = 0.25) +
  geom_text(data = endo_score_heatmap1 %>% filter(end == "WT res" & endo_ratio == 1), aes(x = position, y = end, label = start), size = 1) +
  geom_text(data = endo_score_heatmap1 %>% filter(end == "Coord" & endo_ratio == 1), aes(x = position, y = end), label = "S", color = "darkgreen", size = 1) +
  geom_text(data = endo_score_heatmap1 %>% filter(end == "Coord" & endo_ratio == 1.1), aes(x = position, y = end), label = "M", color = "darkgreen", size = 1) +
  geom_text(data = endo_score_heatmap1 %>% filter(end == "Helix" & endo_ratio == 1), aes(x = position, y = end), label = "H", color = "purple", size = 1) +
  NULL +
  labs(x= "Position", y= "Residue Change", fill = "Enrichment Score") + theme(legend.position = "bottom") + theme(text = element_text(size = 5)) +
  NULL; heatmap_endo
ggsave(file = "Plots/main/fig3b_heatmap_endo.pdf", heatmap_endo, height = 2, width = 4)
```

##Knockout LP cells
```{r STIM1 KO positional patterns}
sko_mis = combined2 %>% filter(type == "mis") %>% select(c("variant", "sko_ratio", "start", "end", "position"))
sko_non = combined2 %>% filter(type == "non") %>% select(c("variant", "sko_ratio", "start", "end", "position"))
sko_syn = combined2 %>% filter(type == "syn") %>% select(c("variant", "sko_ratio", "start", "end", "position"))

sko_positional_ratios <- sko_mis %>% group_by(position) %>% summarize(missense_geomean = mean(sko_ratio), missense_min = min(sko_ratio))
sko_positional_ratios2 <- merge(sko_positional_ratios, sko_non[,c("position","sko_ratio")], by = "position", all.x = T)
sko_positional_ratios3 <- merge(sko_positional_ratios2, sko_syn[,c("position","sko_ratio")], by = "position", all.x = T)
colnames(sko_positional_ratios3) <- c("position", "missense_geomean","missense_min", "nonsense","synonymous")

sko_positional_ratios4 <- merge(sko_positional_ratios3, annots[,c("position","annot")], by = "position")

SKO_missense_nonsense_scatterplot <- ggplot() + scale_x_log10(limits  = c(0.001,5.1)) + scale_y_log10(limits  = c(0.001,5.1)) + 
  geom_point(data = sko_positional_ratios4, aes(x = missense_geomean, y = nonsense, color = annot), alpha = 0.75) +
  geom_text_repel(data = sko_positional_ratios4, aes(x = missense_geomean, y = nonsense, label = position), color = "red", size = 2, segment.color = "orange",
                  segment.alpha = 0.5) +
  theme(panel.grid = element_blank()) + 
  labs(x = "Missense (geomean),\nknockout LP score", y = "Nonsense,\nknockout LP score") +
  NULL; SKO_missense_nonsense_scatterplot
ggsave(file = "Revision_1/plots_r1/supplement_r1/SKO_missense_nonsense_scatterplot.pdf", SKO_missense_nonsense_scatterplot, height = 2, width = 3.5)

paste("Speamean rho, knockout LP missense vs nonsense:",round(cor(sko_positional_ratios4$missense_geomean, sko_positional_ratios4$nonsense, method = "spearman", use = "complete"),2))

SKO_missense_synonymous_scatterplot <- ggplot() + scale_x_log10(limits  = c(0.001,5.1)) + scale_y_log10(limits  = c(0.001,5.1)) + 
  geom_point(data = sko_positional_ratios4, aes(x = missense_geomean, y = synonymous, color = annot), alpha = 0.75) +
  geom_text_repel(data = sko_positional_ratios4, aes(x = missense_geomean, y = synonymous, label = position), color = "red", size = 2, segment.color = "orange",
                  segment.alpha = 0.5) +
  theme(panel.grid = element_blank()) + 
  labs(x = "Missense (geomean),\nknockout LP score", y = "Synonymous,\nknockout LP score") +
  NULL; SKO_missense_synonymous_scatterplot
ggsave(file = "Revision_1/plots_r1/supplement_r1/SKO_missense_synonymous_scatterplot.pdf", SKO_missense_synonymous_scatterplot, height = 2, width = 3.5)

paste("Speamean rho, knockout LP missense vs synonymous:",round(cor(sko_positional_ratios4$missense_geomean, sko_positional_ratios4$synonymous, method = "spearman", use = "complete"),2))

SKO_nonsense_synonymous_scatterplot <- ggplot() + 
  scale_x_log10(limits  = c(0.001,5.1)) + scale_y_log10(limits  = c(0.001,5.1)) + 
  geom_point(data = sko_positional_ratios4, aes(x = nonsense, y = synonymous, color = annot), alpha = 0.75) +
  geom_text_repel(data = sko_positional_ratios4, aes(x = nonsense, y = synonymous, label = position), color = "red", size = 2, segment.color = "orange",
                  segment.alpha = 0.5) +
  theme(panel.grid = element_blank()) + 
  labs(x = "Nonsense,\nknockout LP score", y = "Synonymous,\nknockout LP score") +
  NULL; SKO_nonsense_synonymous_scatterplot
ggsave(file = "Revision_1/plots_r1/supplement_r1/SKO_nonsense_synonymous_scatterplot.pdf", SKO_nonsense_synonymous_scatterplot, height = 2, width = 3.5)

paste("Speamean rho, knockout LP nonsense vs synonymous:",round(cor(sko_positional_ratios4$nonsense, sko_positional_ratios4$synonymous, method = "spearman", use = "complete"),2))
```

```{r STIM1 KO heatmap}
sko_score_heatmap = combined2 %>% select(c("variant", "position", "start", "end","type", "sko_ratio")) %>% filter(!(position %in% c(0,99)) & !(end %in% c("?")))
sko_score_heatmap[sko_score_heatmap$sko_ratio == "NaN","sko_ratio"] <- NA

## For values below 0.01, set it to 0.01 for the heatmap
for(x in 1:nrow(sko_score_heatmap)){
  if(!is.na(sko_score_heatmap$sko_ratio[x]) & sko_score_heatmap$sko_ratio[x] < 0.01){sko_score_heatmap$sko_ratio[x] <- 0.01}
}

sko_score_heatmap$position = as.character(sko_score_heatmap$position)
sko_score_heatmap = sko_score_heatmap %>% filter(variant != "Z0Z")

sko_score_heatmap$end <- factor(sko_score_heatmap$end, levels = c("_","P","C","M","T","S","G","A","V","I","L","F","Y","W","H","K","R","N","Q","D","E"))

sko_score_heatmap1 <- rbind(sko_score_heatmap,
                         data.frame("position" = annotation_df4$position, "sko_ratio" = 1, "variant" = NA, "start" = annotation_df4$start, "end" = "WT res", "type" = "annot"),
                         data.frame("position" = annotation_df4$position, "sko_ratio" = annotation_df4$coord, "variant" = NA, "start" = "Coord", "end" = "Coord", "type" = "annot"),
                         data.frame("position" = annotation_df4$position, "sko_ratio" = annotation_df4$helix, "variant" = NA, "start" = "Helix", "end" = "Helix", "type" = "annot"), 
                          data.frame("position" = positional_scores$position, "sko_ratio" = positional_scores$sko_avg, "variant" = NA, "r1_ratio" = NA, "r2_ratio" = NA, "r4_ratio" = NA, "r5_ratio" = NA, "r6_ratio" = NA, "start" = "Geomean (mis)", "end" = "Geomean (mis)", "type" = "annot", "rep_count" = NA))

sko_score_heatmap1$end <- factor(sko_score_heatmap1$end, levels = rev(c("Geomean (mis)","Helix","Coord","WT res","_","R","H","K","D","E","S","T","N","Q","A","V","I","L","M","F","Y","W","C","G","P")))
sko_syn$end <- factor(sko_syn$end, levels = rev(c("Geomean (mis)","Helix","Coord","WT res","_","R","H","K","D","E","S","T","N","Q","A","V","I","L","M","F","Y","W","C","G","P")))

heatmap_sko <- ggplot() + labs(x = NULL, y = NULL) +
  scale_fill_gradient2(low = "blue", mid = "white", high = "red", na.value = "grey") +
  geom_tile(data = sko_score_heatmap1 %>% filter(position != "0Z"), aes(x = position, y = end, fill = log10(sko_ratio))) +
  geom_point(data = sko_syn, aes(x = position, y = end), size = 0.25) +
  geom_text(data = sko_score_heatmap1 %>% filter(end == "WT res" & sko_ratio == 1), aes(x = position, y = end, label = start), size = 1) +
  geom_text(data = sko_score_heatmap1 %>% filter(end == "Coord" & sko_ratio == 1), aes(x = position, y = end), label = "S", color = "darkgreen", size = 1) +
  geom_text(data = sko_score_heatmap1 %>% filter(end == "Coord" & sko_ratio == 1.1), aes(x = position, y = end), label = "M", color = "darkgreen", size = 1) +
  geom_text(data = sko_score_heatmap1 %>% filter(end == "Helix" & sko_ratio == 1), aes(x = position, y = end), label = "H", color = "purple", size = 1) +
  NULL +
  labs(x= "Position", y= "Residue Change", fill = "Enrichment Score") + theme(legend.position = "bottom") + theme(text = element_text(size = 5)) +
  NULL; heatmap_sko
ggsave(file = "Plots/main/fig3c_heatmap_sko.pdf", heatmap_sko, height = 2, width = 4)
```

```{r Comparing enrichment scores across cell backgrounds}
endo_cor2 = endo_freq[,c("variant","r1_ratio","r2_ratio", "r4_ratio", "r5_ratio", "r6_ratio")]
endo_cor2[,c("r1_ratio","r2_ratio", "r4_ratio", "r5_ratio", "r6_ratio")] <- log10(endo_cor2[,c("r1_ratio","r2_ratio", "r4_ratio", "r5_ratio", "r6_ratio")])
endo_cor2_melt <- melt(endo_cor2, id = "variant") %>% mutate(type = "Standard")

sko_cor2 = sko_freq[,c("variant","r1_ratio","r2_ratio", "r3_ratio","r4_ratio", "r5_ratio", "r6_ratio")]
sko_cor2[,c("r1_ratio","r2_ratio", "r3_ratio","r4_ratio", "r5_ratio", "r6_ratio")] <- log10(sko_cor2[,c("r1_ratio","r2_ratio", "r3_ratio", "r4_ratio", "r5_ratio", "r6_ratio")])
sko_cor2_melt <- melt(sko_cor2, id = "variant") %>% mutate(type = "Knockout")

endo_sko_melt <- rbind(endo_cor2_melt, sko_cor2_melt)
endo_sko_melt$variable <- as.character(endo_sko_melt$variable)

a67x <- endo_sko_melt %>% filter(variant == "A67_")

endo_sko_melt[endo_sko_melt$variable == "r1_ratio","variable"] <- "rep1"
endo_sko_melt[endo_sko_melt$variable == "r2_ratio","variable"] <- "rep2"
endo_sko_melt[endo_sko_melt$variable == "r3_ratio","variable"] <- "rep3"
endo_sko_melt[endo_sko_melt$variable == "r4_ratio","variable"] <- "rep4"
endo_sko_melt[endo_sko_melt$variable == "r5_ratio","variable"] <- "rep5"
endo_sko_melt[endo_sko_melt$variable == "r6_ratio","variable"] <- "rep6"

Endogenous_SKO_histogram <- ggplot() + theme(panel.grid.minor = element_blank(), panel.grid.major.y = element_blank()) + 
  labs(x = "Enrichment score", y = "Number of instances", fill = "Replicate") + 
  geom_histogram(data = endo_sko_melt, aes(x = value, fill = variable), color = "black", size = 0.2) +
  facet_grid(rows = vars(type), scales = "free_y") +
  NULL; Endogenous_SKO_histogram
ggsave(file = "Plots/supplement/supp3c_Endogenous_SKO_histogram.pdf", Endogenous_SKO_histogram, height = 2.5, width = 3.75)

```

##Bacteria - original
```{r Bacteria STIM1 heatmap}
bac_mis = combined2 %>% filter(type == "mis") %>% select(c("variant", "bac_ratio", "start", "end", "position"))
bac_non = combined2 %>% filter(type == "non") %>% select(c("variant", "bac_ratio", "start", "end", "position"))
bac_syn = combined2 %>% filter(type == "syn") %>% select(c("variant", "bac_ratio", "start", "end", "position"))

bac_positional_ratios <- bac_mis %>% group_by(position) %>% summarize(missense_geomean = 10^mean(log10(bac_ratio), na.rm = T), missense_min = min(bac_ratio))
bac_positional_ratios2 <- merge(bac_positional_ratios, bac_non[,c("position","bac_ratio")], by = "position", all.x = T)
bac_positional_ratios3 <- merge(bac_positional_ratios2, bac_syn[,c("position","bac_ratio")], by = "position", all.x = T)
colnames(bac_positional_ratios3) <- c("position", "missense_geomean","missense_min", "nonsense","synonymous")

Bacterial_missense_geomean_nonsense_scatterplot <- ggplot() +
  #scale_x_log10(limits  = c(0.0001,5.1)) + scale_y_log10(limits  = c(0.0001,5.1)) + 
  geom_point(data = bac_positional_ratios3, aes(x = log10(missense_geomean), y = log10(nonsense)), alpha = 0.25) +
  #geom_text_repel(data = bac_positional_ratios3, aes(x = log10(missense_geomean), y = log10(nonsense), label = position), color = "red", size = 2, segment.color = "orange") +
  scale_x_continuous(breaks = pretty_breaks(n = 3)) +  
  scale_y_continuous(breaks = pretty_breaks(n = 3)) +
  labs(x = "Missense score", y = "Nonsense\nscore") +
  NULL; Bacterial_missense_geomean_nonsense_scatterplot
ggsave(file = "Plots/main/fig4c_Bacterial_missense_geomean_nonsense_scatterplot.pdf", Bacterial_missense_geomean_nonsense_scatterplot, height = 1.4, width = 2)

paste("Speamean rho, knockout LP nonsense vs synonymous:",round(cor(bac_positional_ratios3$missense_geomean, bac_positional_ratios3$nonsense, method = "spearman", use = "complete"),2))

ggplot() + scale_x_log10(limits  = c(0.0001,5.1)) + scale_y_log10(limits  = c(0.0001,5.1)) + 
  geom_point(data = bac_positional_ratios3, aes(x = missense_min, y = nonsense)) +
  geom_text_repel(data = bac_positional_ratios3, aes(x = missense_min, y = nonsense, label = position), color = "red", size = 2, segment.color = "orange")

ggplot() + scale_x_log10(limits  = c(0.0001,5.1)) + scale_y_log10(limits  = c(0.0001,5.1)) + 
  geom_point(data = bac_positional_ratios3, aes(x = missense_geomean, y = synonymous)) +
  geom_text_repel(data = bac_positional_ratios3, aes(x = missense_geomean, y = synonymous, label = position), color = "red", size = 2, segment.color = "orange")

ggplot() + scale_x_log10(limits  = c(0.0001,5.1)) + scale_y_log10(limits  = c(0.0001,5.1)) + 
  geom_point(data = bac_positional_ratios3, aes(x = nonsense, y = synonymous)) +
  geom_text_repel(data = bac_positional_ratios3, aes(x = nonsense, y = synonymous, label = position), color = "red", size = 2, segment.color = "orange")
bac_score_heatmap = combined2 %>% select(c("variant", "position", "start", "end","type", "bac_ratio")) %>% filter(!(position %in% c(0,99)) & !(end %in% c("?")))

## For values below 0.01, set it to 0.01 for the heatmap
for(x in 1:nrow(bac_score_heatmap)){
  if(!is.na(bac_score_heatmap$bac_ratio[x]) & bac_score_heatmap$bac_ratio[x] < 0.01){bac_score_heatmap$bac_ratio[x] <- 0.01}
}

bac_score_heatmap$position = as.character(bac_score_heatmap$position)
bac_score_heatmap = bac_score_heatmap %>% filter(variant != "Z0Z")

bac_score_heatmap$end <- factor(bac_score_heatmap$end, levels = c("_","P","C","M","T","S","G","A","V","I","L","F","Y","W","H","K","R","N","Q","D","E"))

## Also bring in positional geomeans
bac_positional_ratios4 <- bac_positional_ratios3[,c("position","missense_geomean")]
bac_positional_ratios4[is.na(bac_positional_ratios4)] <- 0
bac_positional_ratios4[bac_positional_ratios4$missense_geomean < 0.1,"missense_geomean"] <- 0.1

bac_score_heatmap1 <- rbind(bac_score_heatmap,
                         data.frame("position" = annotation_df4$position, "bac_ratio" = 1, "variant" = NA, "start" = annotation_df4$start, "end" = "WT res", "type" = "annot"),
                         data.frame("position" = annotation_df4$position, "bac_ratio" = annotation_df4$coord, "variant" = NA, "start" = "Coord", "end" = "Coord", "type" = "annot"),
                         data.frame("position" = annotation_df4$position, "bac_ratio" = annotation_df4$helix, "variant" = NA, "start" = "Helix", "end" = "Helix", "type" = "annot"), 
                          data.frame("position" = bac_positional_ratios4$position, "bac_ratio" = bac_positional_ratios4$missense_geomean, "variant" = NA, "r1_ratio" = NA, "r2_ratio" = NA, "r4_ratio" = NA, "r5_ratio" = NA, "r6_ratio" = NA, "start" = "Geomean (mis)", "end" = "Geomean (mis)", "type" = "annot", "rep_count" = NA))


bac_score_heatmap1$end <- factor(bac_score_heatmap1$end, levels = rev(c("Geomean (mis)","Helix","Coord","WT res","_","R","H","K","D","E","S","T","N","Q","A","V","I","L","M","F","Y","W","C","G","P")))
bac_syn$end <- factor(bac_syn$end, levels = rev(c("Geomean (mis)","Helix","Coord","WT res","_","R","H","K","D","E","S","T","N","Q","A","V","I","L","M","F","Y","W","C","G","P")))

# bac_score_heatmap1 = left_join(variant_name, bac_score_heatmap1)

heatmap_bac <- ggplot() + labs(x = NULL, y = NULL) +
  scale_fill_gradient2(low = "blue", mid = "white", high = "red", na.value = "grey") +
  geom_tile(data = bac_score_heatmap1 %>% filter(position != "0Z"), aes(x = position, y = end, fill = log10(bac_ratio))) +
  geom_point(data = bac_syn, aes(x = position, y = end), size = 0.25) +
  geom_text(data = bac_score_heatmap1 %>% filter(end == "WT res" & bac_ratio == 1), aes(x = position, y = end, label = start), size = 1) +
  geom_text(data = bac_score_heatmap1 %>% filter(end == "Coord" & bac_ratio == 1), aes(x = position, y = end), label = "S", color = "darkgreen", size = 1) +
  geom_text(data = bac_score_heatmap1 %>% filter(end == "Coord" & bac_ratio == 1.1), aes(x = position, y = end), label = "M", color = "darkgreen", size = 1) +
  geom_text(data = bac_score_heatmap1 %>% filter(end == "Helix" & bac_ratio == 1), aes(x = position, y = end), label = "H", color = "purple", size = 1) +
  NULL +
  labs(x= "Position", y= "Residue Change", fill = "Enrichment Score") + theme(legend.position = "bottom") + theme(text = element_text(size = 5)) +
  NULL; heatmap_bac
ggsave(file = "Plots/main/fig4a_heatmap_bac.pdf", heatmap_bac, height = 2, width = 4)
```
##Validation plots
```{r validation plots}
#standard LP
endo_ssv_plot = ggplot(data = combined2, aes(x= log10(endo_ratio) , y= log10(endo_ssv_ratio)))+
  geom_point(size = 2, alpha = 0.5) +
  geom_text_repel(aes(label = variant), color = "orange", alpha = 0.75) +
  labs(x= "Standard LP DMS", y= "Standard LP Validation") +
  #stat_smooth (geom="line", alpha=0.3, size=1, span=0.5, method = lm) +
  scale_x_continuous(limits = c(-3,1)) + scale_y_continuous(limits = c(-2.2,0)) +
  theme(legend.position = "none") +
  NULL; endo_ssv_plot
ggsave("plots/supplement/supp3d_endo_ssv_plot.pdf", endo_ssv_plot, height = 2.5, width = 2.5)

#knockout LP
sko_ssv_plot = ggplot(data = combined2, aes(x= log10(sko_ratio) , y= log10(sko_ssv_ratio)))+
  geom_point(size = 2, alpha = 0.5) +
  geom_text_repel(aes(label = variant), color = "orange", alpha = 0.75) +
  labs(x= "Knockout LP DMS", y= "Knockout LP Validation") +
  theme(legend.position = "none") +
  #stat_smooth (geom="line", alpha=0.3, size=1, span=0.5, method = lm) + 
  scale_x_continuous(limits = c(-0.75,0.25)) + scale_y_continuous(limits = c(-3.3,0)) +
  NULL; sko_ssv_plot
ggsave("plots/supplement/supp3e_sko_ssv_plot.pdf", sko_ssv_plot, height = 2.5, width = 2.5)
```

##Bacteria DMS with three E.coli strains and three different media
```{r counts}
bac_raw_counts <- read.csv(file = "data/bac_stim1.csv", header = T, stringsAsFactors = F)
bac_raw_counts <-  subset(bac_raw_counts, select = -X )
bac_raw_counts = bac_raw_counts %>% select(-position)

## Figure out the total number of reads per sample
bac_readcounts <- data.frame("total_count" = colSums(bac_raw_counts[,2:ncol(bac_raw_counts)], na.rm = T))

## Plot the number of total reads we have per sample
ggplot() + scale_x_log10(limits = c(1e5,1e7)) + 
  geom_histogram(data = bac_readcounts, aes(x = total_count), bins = 20, color = "black", fill = "grey80")
## Libraries have lower total reads (presumably because of Amp-EZ); genomic DNA sequencing had generally comparable total reads

## figure out read counts that should service as limits of detection for each sample
bac_lod_readcounts <- c()
bac_lod_freqs <- c()
for(re in bac_readcounts$total_count){
  temp_subset <- (lod_frame %>% mutate(abs_diff = abs(reads - re)) %>% arrange(abs_diff))[1,]
  bac_lod_readcounts <- c(bac_lod_readcounts, temp_subset$count)
  bac_lod_freqs <- c(bac_lod_freqs, temp_subset$freq)
}
bac_lod_dataframe <- data.frame("reads" = bac_readcounts, "count" = bac_lod_readcounts, "freq" = bac_lod_freqs)

## Now figure out which data points were below the limit of detection
bac_raw_counts[is.na(bac_raw_counts)] <- 0

bac_raw_counts2 <- data.frame("variant" = bac_raw_counts$variant,
                           "lib1" = bac_raw_counts$lib1 < bac_lod_readcounts[1],
                           "lib2" = bac_raw_counts$lib2 < bac_lod_readcounts[2],
                           "r1_10b_lb" = bac_raw_counts$r1_10b_lb < bac_lod_readcounts[3],
                           "r1_10b_mm" = bac_raw_counts$r1_10b_mm < bac_lod_readcounts[4],
                           "r1_10b_ca" = bac_raw_counts$r1_10b_ca < bac_lod_readcounts[5],
                           "r1_dh5a_lb" = bac_raw_counts$r1_dh5a_lb < bac_lod_readcounts[6],
                           "r1_dh5a_mm" = bac_raw_counts$r1_dh5a_mm < bac_lod_readcounts[7],
                           "r1_dh5a_ca" = bac_raw_counts$r1_dh5a_ca < bac_lod_readcounts[8],
                           "r1_bl21_lb" = bac_raw_counts$r1_bl21_lb < bac_lod_readcounts[9],
                           "r1_bl21_mm" = bac_raw_counts$r1_bl21_mm < bac_lod_readcounts[10],
                           "r1_bl21_ca" = bac_raw_counts$r1_bl21_ca < bac_lod_readcounts[11],
                           "r2_10b_lb" = bac_raw_counts$r2_10b_lb < bac_lod_readcounts[12],
                           "r2_10b_mm" = bac_raw_counts$r2_10b_mm < bac_lod_readcounts[13],
                           "r2_10b_ca" = bac_raw_counts$r2_10b_ca < bac_lod_readcounts[14],
                           "r2_dh5a_lb" = bac_raw_counts$r2_dh5a_lb < bac_lod_readcounts[15],
                           "r2_dh5a_mm" = bac_raw_counts$r2_dh5a_mm < bac_lod_readcounts[16],
                           "r2_dh5a_ca" = bac_raw_counts$r2_dh5a_ca < bac_lod_readcounts[17],
                           "r2_bl21_lb" = bac_raw_counts$r2_bl21_lb < bac_lod_readcounts[18],
                           "r2_bl21_mm" = bac_raw_counts$r2_bl21_mm < bac_lod_readcounts[19],
                           "r2_bl21_ca" = bac_raw_counts$r2_bl21_ca < bac_lod_readcounts[20],
                           "r3_10b_lb" = bac_raw_counts$r3_10b_lb < bac_lod_readcounts[21],
                           "r3_10b_mm" = bac_raw_counts$r3_10b_mm < bac_lod_readcounts[22],
                           "r3_10b_ca" = bac_raw_counts$r3_10b_ca < bac_lod_readcounts[23],
                           "r3_dh5a_lb" = bac_raw_counts$r3_dh5a_lb < bac_lod_readcounts[24],
                           "r3_dh5a_mm" = bac_raw_counts$r3_dh5a_mm < bac_lod_readcounts[25],
                           "r3_dh5a_ca" = bac_raw_counts$r3_dh5a_ca < bac_lod_readcounts[26],
                           "r3_bl21_lb" = bac_raw_counts$r3_bl21_lb < bac_lod_readcounts[27],
                           "r3_bl21_mm" = bac_raw_counts$r3_bl21_mm < bac_lod_readcounts[28],
                           "r3_bl21_ca" = bac_raw_counts$r3_bl21_ca < bac_lod_readcounts[29]
                           )

## These were the total number of variants below the limit of detection in each bac sequencing run:
colSums(bac_raw_counts2[,c("r1_10b_lb","r1_10b_mm","r1_10b_ca","r1_dh5a_lb","r1_dh5a_mm","r1_dh5a_ca","r1_bl21_lb","r1_bl21_mm","r1_bl21_ca",
                          "r2_10b_lb","r2_10b_mm","r2_10b_ca","r2_dh5a_lb","r2_dh5a_mm","r2_dh5a_ca","r2_bl21_lb","r2_bl21_mm","r2_bl21_ca",
                         "r3_10b_lb","r3_10b_mm","r3_10b_ca","r3_dh5a_lb","r3_dh5a_mm","r3_dh5a_ca","r3_bl21_lb","r3_bl21_mm","r3_bl21_ca")])

## Now calculate bac frequencies, and for bacs below the limit of detection, give them the limit of detection frequency value
bac_freq = bac_raw_counts %>% mutate(f_lib1 = lib1/bac_readcounts$total_count[1], 
                                     f_lib2 = lib2/bac_readcounts$total_count[2], 
                                     f_r1_10b_lb = r1_10b_lb/bac_readcounts$total_count[3],
                                     f_r1_10b_mm = r1_10b_mm/bac_readcounts$total_count[4],
                                     f_r1_10b_ca = r1_10b_ca/bac_readcounts$total_count[5],
                                     f_r1_dh5a_lb = r1_dh5a_lb/bac_readcounts$total_count[6],
                                     f_r1_dh5a_mm = r1_dh5a_mm/bac_readcounts$total_count[7],
                                     f_r1_dh5a_ca = r1_dh5a_ca/bac_readcounts$total_count[8],
                                     f_r1_bl21_lb = r1_bl21_lb/bac_readcounts$total_count[9],
                                     f_r1_bl21_mm = r1_bl21_mm/bac_readcounts$total_count[10],
                                     f_r1_bl21_ca = r1_bl21_ca/bac_readcounts$total_count[11],
                                     f_r2_10b_lb = r2_10b_lb/bac_readcounts$total_count[12],
                                     f_r2_10b_mm = r2_10b_mm/bac_readcounts$total_count[13],
                                     f_r2_10b_ca = r2_10b_ca/bac_readcounts$total_count[14],
                                     f_r2_dh5a_lb = r2_dh5a_lb/bac_readcounts$total_count[15],
                                     f_r2_dh5a_mm = r2_dh5a_mm/bac_readcounts$total_count[16],
                                     f_r2_dh5a_ca = r2_dh5a_ca/bac_readcounts$total_count[17],
                                     f_r2_bl21_lb = r2_bl21_lb/bac_readcounts$total_count[18],
                                     f_r2_bl21_mm = r2_bl21_mm/bac_readcounts$total_count[19],
                                     f_r2_bl21_ca = r2_bl21_ca/bac_readcounts$total_count[20],
                                     f_r3_10b_lb = r3_10b_lb/bac_readcounts$total_count[21],
                                     f_r3_10b_mm = r3_10b_mm/bac_readcounts$total_count[22],
                                     f_r3_10b_ca = r3_10b_ca/bac_readcounts$total_count[23],
                                     f_r3_dh5a_lb = r3_dh5a_lb/bac_readcounts$total_count[24],
                                     f_r3_dh5a_mm = r3_dh5a_mm/bac_readcounts$total_count[25],
                                     f_r3_dh5a_ca = r3_dh5a_ca/bac_readcounts$total_count[26],
                                     f_r3_bl21_lb = r3_bl21_lb/bac_readcounts$total_count[27],
                                     f_r3_bl21_mm = r3_bl21_mm/bac_readcounts$total_count[28],
                                     f_r3_bl21_ca = r3_bl21_ca/bac_readcounts$total_count[29])

bac_freq[bac_raw_counts2$r1_10b_lb == T,"f_r1_10b_lb"] <- bac_lod_freqs[3]
bac_freq[bac_raw_counts2$r1_10b_mm == T,"f_r1_10b_mm"] <- bac_lod_freqs[4]
bac_freq[bac_raw_counts2$r1_10b_ca == T,"f_r1_10b_ca"] <- bac_lod_freqs[5]
bac_freq[bac_raw_counts2$r1_dh5a_lb == T,"f_r1_dh5a_lb"] <- bac_lod_freqs[6]
bac_freq[bac_raw_counts2$r1_dh5a_mm == T,"f_r1_dh5a_mm"] <- bac_lod_freqs[7]
bac_freq[bac_raw_counts2$r1_dh5a_ca == T,"f_r1_dha5_ca"] <- bac_lod_freqs[8]
bac_freq[bac_raw_counts2$r1_bl21_lb == T,"f_r1_bl21_lb"] <- bac_lod_freqs[9]
bac_freq[bac_raw_counts2$r1_bl21_mm == T,"f_r1_bl21_mm"] <- bac_lod_freqs[10]
bac_freq[bac_raw_counts2$r1_bl21_ca == T,"f_r1_bl21_ca"] <- bac_lod_freqs[11]
bac_freq[bac_raw_counts2$r2_10b_lb == T,"f_r2_10b_lb"] <- bac_lod_freqs[12]
bac_freq[bac_raw_counts2$r2_10b_mm == T,"f_r2_10b_mm"] <- bac_lod_freqs[13]
bac_freq[bac_raw_counts2$r2_10b_ca == T,"f_r2_10b_ca"] <- bac_lod_freqs[14]
bac_freq[bac_raw_counts2$r2_dh5a_lb == T,"f_r2_dh5a_lb"] <- bac_lod_freqs[15]
bac_freq[bac_raw_counts2$r2_dh5a_mm == T,"f_r2_dh5a_mm"] <- bac_lod_freqs[16]
bac_freq[bac_raw_counts2$r2_dh5a_ca == T,"f_r2_dha5_ca"] <- bac_lod_freqs[17]
bac_freq[bac_raw_counts2$r2_bl21_lb == T,"f_r2_bl21_lb"] <- bac_lod_freqs[18]
bac_freq[bac_raw_counts2$r2_bl21_mm == T,"f_r2_bl21_mm"] <- bac_lod_freqs[19]
bac_freq[bac_raw_counts2$r2_bl21_ca == T,"f_r2_bl21_ca"] <- bac_lod_freqs[20]
bac_freq[bac_raw_counts2$r3_10b_lb == T,"f_r3_10b_lb"] <- bac_lod_freqs[21]
bac_freq[bac_raw_counts2$r3_10b_mm == T,"f_r3_10b_mm"] <- bac_lod_freqs[22]
bac_freq[bac_raw_counts2$r3_10b_ca == T,"f_r3_10b_ca"] <- bac_lod_freqs[23]
bac_freq[bac_raw_counts2$r3_dh5a_lb == T,"f_r3_dh5a_lb"] <- bac_lod_freqs[24]
bac_freq[bac_raw_counts2$r3_dh5a_mm == T,"f_r3_dh5a_mm"] <- bac_lod_freqs[25]
bac_freq[bac_raw_counts2$r3_dh5a_ca == T,"f_r3_dha5_ca"] <- bac_lod_freqs[26]
bac_freq[bac_raw_counts2$r3_bl21_lb == T,"f_r3_bl21_lb"] <- bac_lod_freqs[27]
bac_freq[bac_raw_counts2$r3_bl21_mm == T,"f_r3_bl21_mm"] <- bac_lod_freqs[28]
bac_freq[bac_raw_counts2$r3_bl21_ca == T,"f_r3_bl21_ca"] <- bac_lod_freqs[29]

bac_freq[bac_raw_counts2$lib1 == T,c("f_lib1","f_lib2","f_r1_10b_lb","f_r1_10b_mm","f_r1_10b_ca","f_r1_dh5a_lb","f_r1_dh5a_mm","f_r1_dh5a_ca","f_r1_bl21_lb","f_r1_bl21_mm","f_r1_bl21_ca","f_r2_10b_lb","f_r2_10b_mm","f_r2_10b_ca","f_r2_dh5a_lb","f_r2_dh5a_mm","f_r2_dh5a_ca","f_r2_bl21_lb","f_r2_bl21_mm","f_r2_bl21_ca","f_r3_10b_lb","f_r3_10b_mm","f_r3_10b_ca","f_r3_dh5a_lb","f_r3_dh5a_mm","f_r3_dh5a_ca","f_r3_bl21_lb","f_r3_bl21_mm","f_r3_bl21_ca")] <- NA
bac_freq[bac_raw_counts2$lib2 == T,c("f_lib1","f_lib2","f_r1_10b_lb","f_r1_10b_mm","f_r1_10b_ca","f_r1_dh5a_lb","f_r1_dh5a_mm","f_r1_dh5a_ca","f_r1_bl21_lb","f_r1_bl21_mm","f_r1_bl21_ca","f_r2_10b_lb","f_r2_10b_mm","f_r2_10b_ca","f_r2_dh5a_lb","f_r2_dh5a_mm","f_r2_dh5a_ca","f_r2_bl21_lb","f_r2_bl21_mm","f_r2_bl21_ca","f_r3_10b_lb","f_r3_10b_mm","f_r3_10b_ca","f_r3_dh5a_lb","f_r3_dh5a_mm","f_r3_dh5a_ca","f_r3_bl21_lb","f_r3_bl21_mm","f_r3_bl21_ca")] <- NA


bac_freq$bac_frac_lod <- rowSums(bac_raw_counts2[,c("r1_10b_lb","r1_10b_mm","r1_10b_ca","r1_dh5a_lb","r1_dh5a_mm","r1_dh5a_ca","r1_bl21_lb","r1_bl21_mm","r1_bl21_ca","r2_10b_lb","r2_10b_mm","r2_10b_ca","r2_dh5a_lb","r2_dh5a_mm","r2_dh5a_ca","r2_bl21_lb","r2_bl21_mm","r2_bl21_ca","r3_10b_lb","r3_10b_mm","r3_10b_ca","r3_dh5a_lb","r3_dh5a_mm","r3_dh5a_ca","r3_bl21_lb","r3_bl21_mm","r3_bl21_ca")])/length(c("r1_10b_lb","r1_10b_mm","r1_10b_ca","r1_dh5a_lb","r1_dh5a_mm","r1_dh5a_ca","r1_bl21_lb","r1_bl21_mm","r1_bl21_ca","r2_10b_lb","r2_10b_mm","r2_10b_ca","r2_dh5a_lb","r2_dh5a_mm","r2_dh5a_ca","r2_bl21_lb","r2_bl21_mm","r2_bl21_ca","r3_10b_lb","r3_10b_mm","r3_10b_ca","r3_dh5a_lb","r3_dh5a_mm","r3_dh5a_ca","r3_bl21_lb","r3_bl21_mm","r3_bl21_ca"))

## Take geometric means of the variant frequencies for the two preps
bac_freq$lib_freq <- 10^rowMeans(log10(bac_freq[,c("f_lib1","f_lib2")]))

bac_freq$r1_10b_lb_ratio <- bac_freq$f_r1_10b_lb / bac_freq$lib_freq
bac_freq$r1_10b_mm_ratio <- bac_freq$f_r1_10b_mm / bac_freq$lib_freq
bac_freq$r1_10b_ca_ratio <- bac_freq$f_r1_10b_ca / bac_freq$lib_freq
bac_freq$r1_dh5a_lb_ratio <- bac_freq$f_r1_dh5a_lb / bac_freq$lib_freq
bac_freq$r1_dh5a_mm_ratio <- bac_freq$f_r1_dh5a_mm / bac_freq$lib_freq
bac_freq$r1_dh5a_ca_ratio <- bac_freq$f_r1_dh5a_ca / bac_freq$lib_freq
bac_freq$r1_bl21_lb_ratio <- bac_freq$f_r1_bl21_lb / bac_freq$lib_freq
bac_freq$r1_bl21_mm_ratio <- bac_freq$f_r1_bl21_mm / bac_freq$lib_freq
bac_freq$r1_bl21_ca_ratio <- bac_freq$f_r1_bl21_ca / bac_freq$lib_freq

bac_freq$r2_10b_lb_ratio <- bac_freq$f_r2_10b_lb / bac_freq$lib_freq
bac_freq$r2_10b_mm_ratio <- bac_freq$f_r2_10b_mm / bac_freq$lib_freq
bac_freq$r2_10b_ca_ratio <- bac_freq$f_r2_10b_ca / bac_freq$lib_freq
bac_freq$r2_dh5a_lb_ratio <- bac_freq$f_r2_dh5a_lb / bac_freq$lib_freq
bac_freq$r2_dh5a_mm_ratio <- bac_freq$f_r2_dh5a_mm / bac_freq$lib_freq
bac_freq$r2_dh5a_ca_ratio <- bac_freq$f_r2_dh5a_ca / bac_freq$lib_freq
bac_freq$r2_bl21_lb_ratio <- bac_freq$f_r2_bl21_lb / bac_freq$lib_freq
bac_freq$r2_bl21_mm_ratio <- bac_freq$f_r2_bl21_mm / bac_freq$lib_freq
bac_freq$r2_bl21_ca_ratio <- bac_freq$f_r2_bl21_ca / bac_freq$lib_freq

bac_freq$r3_10b_lb_ratio <- bac_freq$f_r3_10b_lb / bac_freq$lib_freq
bac_freq$r3_10b_mm_ratio <- bac_freq$f_r3_10b_mm / bac_freq$lib_freq
bac_freq$r3_10b_ca_ratio <- bac_freq$f_r3_10b_ca / bac_freq$lib_freq
bac_freq$r3_dh5a_lb_ratio <- bac_freq$f_r3_dh5a_lb / bac_freq$lib_freq
bac_freq$r3_dh5a_mm_ratio <- bac_freq$f_r3_dh5a_mm / bac_freq$lib_freq
bac_freq$r3_dh5a_ca_ratio <- bac_freq$f_r3_dh5a_ca / bac_freq$lib_freq
bac_freq$r3_bl21_lb_ratio <- bac_freq$f_r3_bl21_lb / bac_freq$lib_freq
bac_freq$r3_bl21_mm_ratio <- bac_freq$f_r3_bl21_mm / bac_freq$lib_freq
bac_freq$r3_bl21_ca_ratio <- bac_freq$f_r3_bl21_ca / bac_freq$lib_freq

bac_ratio_ns = bac_freq %>% select(c("variant", "r1_10b_lb_ratio", "r1_10b_mm_ratio", "r1_10b_ca_ratio", "r1_dh5a_lb_ratio", "r1_dh5a_mm_ratio", "r1_dh5a_ca_ratio", "r1_bl21_lb_ratio", "r1_bl21_mm_ratio", "r1_bl21_ca_ratio",
                                  "r2_10b_lb_ratio", "r2_10b_mm_ratio", "r2_10b_ca_ratio", "r2_dh5a_lb_ratio", "r2_dh5a_mm_ratio", "r2_dh5a_ca_ratio", "r2_bl21_lb_ratio", "r2_bl21_mm_ratio", "r2_bl21_ca_ratio",
                                  "r3_10b_lb_ratio", "r3_10b_mm_ratio", "r3_10b_ca_ratio", "r3_dh5a_lb_ratio", "r3_dh5a_mm_ratio", "r3_dh5a_ca_ratio", "r3_bl21_lb_ratio", "r3_bl21_mm_ratio", "r3_bl21_ca_ratio"))


bac_ratio_ns$lb_10b<- rowMeans(log10(bac_freq[,c("r1_10b_lb_ratio", "r2_10b_lb_ratio", "r3_10b_lb_ratio")]))
bac_ratio_ns$lb_dh5a<- rowMeans(log10(bac_freq[,c("r1_dh5a_lb_ratio", "r2_dh5a_lb_ratio", "r3_dh5a_lb_ratio")]))
bac_ratio_ns$lb_bl21<- rowMeans(log10(bac_freq[,c("r1_bl21_lb_ratio", "r2_bl21_lb_ratio", "r3_bl21_lb_ratio")]))

bac_ratio_ns$mm_10b<- rowMeans(log10(bac_freq[,c("r1_10b_mm_ratio", "r2_10b_mm_ratio", "r3_10b_mm_ratio")]))
bac_ratio_ns$mm_dh5a<- rowMeans(log10(bac_freq[,c("r1_dh5a_mm_ratio", "r2_dh5a_mm_ratio", "r3_dh5a_mm_ratio")]))
bac_ratio_ns$mm_bl21<- rowMeans(log10(bac_freq[,c("r1_bl21_mm_ratio", "r2_bl21_mm_ratio", "r3_bl21_mm_ratio")]))

bac_ratio_ns$ca_10b<- rowMeans(log10(bac_freq[,c("r1_10b_ca_ratio", "r2_10b_ca_ratio", "r3_10b_ca_ratio")]))
bac_ratio_ns$ca_dh5a<- rowMeans(log10(bac_freq[,c("r1_dh5a_ca_ratio", "r2_dh5a_ca_ratio", "r3_dh5a_ca_ratio")]))
bac_ratio_ns$ca_bl21<- rowMeans(log10(bac_freq[,c("r1_bl21_ca_ratio", "r2_bl21_ca_ratio", "r3_bl21_ca_ratio")]))

bac_ratio_ns$position <- substr(bac_ratio_ns$variant,2,3)
bac_ratio_ns$start <- substr(bac_ratio_ns$variant,1,1)
bac_ratio_ns$end <- substr(bac_ratio_ns$variant,4,4)


bac_cor_ns = bac_ratio_ns %>% select(c("lb_10b", "lb_dh5a", "lb_bl21", "mm_10b", "mm_dh5a", "mm_bl21", "ca_10b", "ca_dh5a", "ca_bl21"))
bac_cor_ns = bac_cor_ns[!is.infinite(rowSums(bac_cor_ns)),]

bac_cor_ns_plot = bac_cor_ns %>% ggpairs(aes(alpha = 0.15))
bac_cor_ns_plot

#correlation plots for bacteria strain + media
cor_10b_lb = bac_freq %>% mutate(R1 = log10(f_r1_10b_lb), R2 = log10(f_r2_10b_lb), R3 = log10(f_r3_10b_lb)) %>% select(R1, R2, R3) %>% ggpairs(lower = list(continuous = wrap("points", alpha = 0.15, size = 1)))
cor_10b_mm = bac_freq %>% mutate(R1 = log10(f_r1_10b_mm), R2 = log10(f_r2_10b_mm), R3 = log10(f_r3_10b_mm)) %>% select(R1, R2, R3) %>% ggpairs(lower = list(continuous = wrap("points", alpha = 0.15, size = 1)))
cor_10b_ca = bac_freq %>% mutate(R1 = log10(f_r1_10b_ca), R2 = log10(f_r2_10b_ca), R3 = log10(f_r3_10b_ca)) %>% select(R1, R2, R3) %>% ggpairs(lower = list(continuous = wrap("points", alpha = 0.15, size = 1)))

cor_dh5a_lb = bac_freq %>% mutate(R1 = log10(f_r1_dh5a_lb), R2 = log10(f_r2_dh5a_lb), R3 = log10(f_r3_dh5a_lb)) %>% select(R1, R2, R3) %>% ggpairs(lower = list(continuous = wrap("points", alpha = 0.15, size = 1)))
cor_dh5a_mm = bac_freq %>% mutate(R1 = log10(f_r1_dh5a_mm), R2 = log10(f_r2_dh5a_mm), R3 = log10(f_r3_dh5a_mm)) %>% select(R1, R2, R3) %>% ggpairs(lower = list(continuous = wrap("points", alpha = 0.15, size = 1)))
cor_dh5a_ca = bac_freq %>% mutate(R1 = log10(f_r1_dh5a_ca), R2 = log10(f_r2_dh5a_ca), R3 = log10(f_r3_dh5a_ca)) %>% select(R1, R2, R3) %>% ggpairs(lower = list(continuous = wrap("points", alpha = 0.15, size = 1)))

cor_bl21_lb = bac_freq %>% mutate(R1 = log10(f_r1_bl21_lb), R2 = log10(f_r2_bl21_lb), R3 = log10(f_r3_bl21_lb)) %>% select(R1, R2, R3) %>% ggpairs(lower = list(continuous = wrap("points", alpha = 0.15, size = 1)))
cor_bl21_mm = bac_freq %>% mutate(R1 = log10(f_r1_bl21_mm), R2 = log10(f_r2_bl21_mm), R3 = log10(f_r3_bl21_mm)) %>% select(R1, R2, R3) %>% ggpairs(lower = list(continuous = wrap("points", alpha = 0.15, size = 1)))
cor_bl21_ca = bac_freq %>% mutate(R1 = log10(f_r1_bl21_ca), R2 = log10(f_r2_bl21_ca), R3 = log10(f_r3_bl21_ca)) %>% select(R1, R2, R3) %>% ggpairs(lower = list(continuous = wrap("points", alpha = 0.15, size = 1)))



#revision1 - exporting correlation plot for bacteria scores as supplement
ggsave("Revision_1/plots_r1/supplement_r1/supp4d_cor_10b_lb.pdf", cor_10b_lb, height = 2.5, width = 2.5)
ggsave("Revision_1/plots_r1/supplement_r1/supp4d_cor_10b_mm.pdf", cor_10b_mm, height = 2.5, width = 2.5)
ggsave("Revision_1/plots_r1/supplement_r1/supp4d_cor_10b_ca.pdf", cor_10b_ca, height = 2.5, width = 2.5)

ggsave("Revision_1/plots_r1/supplement_r1/supp4d_cor_dh5a_lb.pdf", cor_dh5a_lb, height = 2.5, width = 2.5)
ggsave("Revision_1/plots_r1/supplement_r1/supp4d_cor_dh5a_mm.pdf", cor_dh5a_mm, height = 2.5, width = 2.5)
ggsave("Revision_1/plots_r1/supplement_r1/supp4d_cor_dh5a_ca.pdf", cor_dh5a_ca, height = 2.5, width = 2.5)

ggsave("Revision_1/plots_r1/supplement_r1/supp4d_cor_bl21_lb.pdf", cor_bl21_lb, height = 2.5, width = 2.5)
ggsave("Revision_1/plots_r1/supplement_r1/supp4d_cor_bl21_mm.pdf", cor_bl21_mm, height = 2.5, width = 2.5)
ggsave("Revision_1/plots_r1/supplement_r1/supp4d_cor_bl21_ca.pdf", cor_bl21_ca, height = 2.5, width = 2.5)

```

```{r bacteria heatmap}
bac_ratio_ns_tomelt <- bac_ratio_ns %>% select(variant, lb_10b, lb_dh5a, lb_bl21, mm_10b, mm_dh5a, mm_bl21, ca_10b, ca_dh5a, ca_bl21)

bac_ratio_ns_melted <- melt(bac_ratio_ns_tomelt)
bac_ratio_ns_melted$variable <- as.character(bac_ratio_ns_melted$variable)
bac_ratio_ns_melted$media <- NA
bac_ratio_ns_melted$bacteria <- NA
bac_ratio_ns_melted$position <- NA
bac_ratio_ns_melted$end <- NA
for(x in 1:nrow(bac_ratio_ns_melted)){
  bac_ratio_ns_melted$media[x] <- strsplit(bac_ratio_ns_melted$variable[x],"_")[[1]][1]
  bac_ratio_ns_melted$bacteria[x] <- strsplit(bac_ratio_ns_melted$variable[x],"_")[[1]][2]
  bac_ratio_ns_melted$position[x] <- gsub("[^0-9]","",bac_ratio_ns_melted$variant[x])
  bac_ratio_ns_melted$end[x] <- substr(gsub("[0-9]","",bac_ratio_ns_melted$variant[x]),2,2)
}

## Do WT normalization of the values
bac_ratio_ns_melted$wt_norm <- 0
for(x in 1:nrow(bac_ratio_ns_melted)){
  temp_wt_value <- subset(bac_ratio_ns_melted, media == bac_ratio_ns_melted$media[x] & bacteria == bac_ratio_ns_melted$bacteria[x] & variant == "Z0Z")$value
  bac_ratio_ns_melted$wt_norm[x] <- bac_ratio_ns_melted$value[x] - temp_wt_value
}

bac_ratio_ns_melted_no_wt <- bac_ratio_ns_melted %>% filter(variant != "Z0Z")

bac_ratio_ns_melted_no_wt$position <- factor(bac_ratio_ns_melted_no_wt$position, levels = c("70","78","92","75","77","97"))
bac_ratio_ns_melted_no_wt$media <- factor(bac_ratio_ns_melted_no_wt$media, levels = c("lb","ca","mm"))
bac_ratio_ns_melted_no_wt$bacteria <- factor(bac_ratio_ns_melted_no_wt$bacteria, levels = c("10b","dh5a","bl21"))

Bacterial_condition_heatmaps <- ggplot() + 
  scale_fill_gradient2(low = "blue", mid = "white", high = "red", na.value = "grey80") +
  geom_tile(data = bac_ratio_ns_melted_no_wt, aes(x = position, y = end, fill = wt_norm)) +
  facet_grid(rows = vars(media), cols = vars(bacteria)) +
  labs(x = NULL, y = "Mutant amino acid") +
  theme(axis.text.y = element_blank(), legend.position = "top") +
  NULL; Bacterial_condition_heatmaps
ggsave(file = "Plots/main/fig4e_Bacterial_condition_heatmaps.pdf", Bacterial_condition_heatmaps, height = 3.25, width = 3.5)

#original DMS scores vs new bacteria DMS score
bac_score_10b = left_join(bac_ratio_ns, combined2)
bac_old_vs_new = ggplot(data = bac_score_10b, aes(x= log10(bac_ratio), y = lb_10b))+
  geom_point(alpha = 0.5) +
  stat_smooth (geom="line", alpha=0.3, size=1, span=0.5, method = lm) + stat_poly_eq(use_label(c("R2"))) +
  labs(x= "Original Bacteria Score", y= "Bacteria DMS score")+NULL; bac_old_vs_new
ggsave("plots/supplement/supp4c_bac_old_vs_new.pdf", bac_old_vs_new, height = 5, width = 5)
```

```{r Try doing a principal component analysis after casting variants as columns and treating each condition as rows}

bac_ratio_ns_matrix_for_pca <- data.frame(t(bac_ratio_ns_tomelt %>% filter(!is.na(lb_10b))))
colnames(bac_ratio_ns_matrix_for_pca) <- bac_ratio_ns_matrix_for_pca[1,]
bac_ratio_ns_matrix_for_pca2 <- bac_ratio_ns_matrix_for_pca[-1,]
bac_ratio_ns_matrix_for_pca2[, ] <- apply(bac_ratio_ns_matrix_for_pca2[, ], 2, function(x) as.numeric(as.character(x)))
bac_ratio_ns_matrix_for_pca3 <- as.matrix(bac_ratio_ns_matrix_for_pca2)
bac_ratio_pca = prcomp(bac_ratio_ns_matrix_for_pca3)

# create data frame with scores
scores = as.data.frame(bac_ratio_pca$x)
scores$labels <- rownames(scores)

for(x in 1:nrow(scores)){
  scores$media[x] <- strsplit(scores$labels[x],"_")[[1]][1]
  scores$bacteria[x] <- strsplit(scores$labels[x],"_")[[1]][2]
}

bacteria_media_pca_scatterplot <- ggplot() + 
  xlab(paste("Principal component 1\n(",round(summary(bac_ratio_pca)$importance[2] * 100,0),"% of variance)",sep="")) + 
  ylab(paste("Principal component 2\n(",round(summary(bac_ratio_pca)$importance[5] * 100,0),"% of variance)",sep="")) + 
  #geom_text_repel(data = scores, aes(x = PC1, y = PC2, label = labels), segment.color = 'grey90', point.padding = 0.05, size = 2, segment.alpha = 0.5, color = "red") +
  geom_point(data = scores, aes(x = PC1, y = PC2, color = bacteria, shape = media), size = 2, alpha = 0.5) +
  theme(panel.grid = element_blank(), legend.position = "top") + 
  NULL; bacteria_media_pca_scatterplot
ggsave(file = "Plots/main/fig4f_Bacteria_media_pca_scatterplot.pdf", bacteria_media_pca_scatterplot, height = 2.5, width = 2)
```
```{r creating a final table with all scores}
t1 = read.csv("data/dms_scores.csv") %>% select(variant, endo_ratio, sko_ratio, bac_ratio, endo_ssv_ratio, sko_ssv_ratio)
t2 = read.csv("data/survival_scores.csv") %>% select(variant, lb_10b, lb_dh5a, lb_bl21, mm_10b, mm_dh5a, mm_bl21, ca_10b, ca_dh5a, ca_bl21)
t3 = left_join(t1, t2)
write.csv(t3, "data/final_table.csv")
```



##Analysis of DMS scores:
1. Comparisons of variants based on position, mutated residue, synonymous or nonsense variants for all three datasets
```{r positional comparison}
positional_matrix_for_pca <- data.frame((positional_scores))
rownames(positional_matrix_for_pca) <- positional_matrix_for_pca[,1]
positional_matrix_for_pca2 <- positional_matrix_for_pca[,-1]
positional_matrix_for_pca2 <- log10(positional_matrix_for_pca2)
positional_matrix_for_pca2[, ] <- apply(positional_matrix_for_pca2[, ], 2, function(x) as.numeric(as.character(x)))
positional_matrix_for_pca3 <- as.matrix(positional_matrix_for_pca2)
positional_pca = prcomp(positional_matrix_for_pca3, scale. = TRUE)

# create data frame with scores
scores = as.data.frame(positional_pca$x)
scores$position <- rownames(scores)

positional_media_pca_scatterplot <- ggplot() + 
  xlab(paste("Principal component 1\n(",round(summary(positional_pca)$importance[2] * 100,0),"% of variance)",sep="")) + 
  ylab(paste("Principal component 2\n(",round(summary(positional_pca)$importance[5] * 100,0),"% of variance)",sep="")) + 
  #geom_text_repel(data = scores, aes(x = PC1, y = PC2, label = labels), segment.color = 'grey90', point.padding = 0.05, size = 2, segment.alpha = 0.5, color = "red") +
  geom_point(data = scores, aes(x = PC1, y = PC2), size = 2, alpha = 0.5) +
  geom_text_repel(data = scores, aes(x = PC1, y = PC2, label = position), size = 2, alpha = 0.5, color = "red") +
  theme(panel.grid = element_blank(), legend.position = "top") + 
  NULL; positional_media_pca_scatterplot

positional_hclust <- melt(positional_matrix_for_pca, id = "position")

row.order <- hclust(dist(positional_matrix_for_pca2))$order
col.order <- hclust(dist(t(positional_matrix_for_pca2)))$order

double_pos_order <- rev(rownames(positional_matrix_for_pca2[row.order, col.order]))
double_label_order <- colnames(positional_matrix_for_pca2[row.order, col.order])

positional_hclust2 <- positional_hclust
positional_hclust2$position <- factor(positional_hclust2$position, levels = double_pos_order)
positional_hclust2$variable <- factor(positional_hclust2$variable, levels = double_label_order)

annots <- rbind(data.frame("position" = seq(63,75),"variable" = "annot", value = 0.98),
                data.frame("position" = seq(76,88),"variable" = "annot", value = 0.99),
                data.frame("position" = seq(89,98),"variable" = "annot", value = 1))

positional_hclust3 <- rbind(positional_hclust2, annots)

clustered_positional_heatmap <- ggplot() + 
  theme(axis.text.y = element_text(size = 7), axis.text.x = element_text(size = 7, angle = 45, hjust = 0.75), legend.position = "top") + 
  labs(x = NULL, y = NULL) + 
  scale_fill_gradient2(low = "blue", mid = "white", high = "red") +
  geom_tile(data = positional_hclust3, aes(x = position, y = variable, fill = log10(value))) +
  geom_tile(data = positional_hclust3 %>% filter(variable == "annot" & value == 0.98), aes(x = position, y = variable), fill = "red", alpha = 0.6) +
  geom_tile(data = positional_hclust3 %>% filter(variable == "annot" & value == 0.99), aes(x = position, y = variable), fill = "green", alpha = 0.6) +
  geom_tile(data = positional_hclust3 %>% filter(variable == "annot" & value == 1), aes(x = position, y = variable), fill = "blue", alpha = 0.6) +
  geom_text(data = positional_hclust3 %>% filter(variable == "sko_avg" & position %in% c(70,92,94)), aes(x = position, y = variable), label = "*") +
  NULL; clustered_positional_heatmap
ggsave(file = "plots/main/fig5b_clustered_positional_heatmap.pdf", clustered_positional_heatmap, height = 1.85, width = 4.6)

```
```{r Mutated residue comparison}
mut_residue_scores = combined2 %>% group_by(end) %>% mutate(endo_avg = 10^mean(log10(endo_ratio), na.rm = T), sko_avg = 10^mean(log10(sko_ratio), na.rm = T), bac_avg = 10^mean(log10(bac_ratio), na.rm =T))

MutRes_endo_v_ko_scatterplot <- ggplot(data = mut_residue_scores %>% distinct(end, endo_avg, sko_avg), aes(x= log10(endo_avg), y= log10(sko_avg)))+ 
  # scale_x_continuous(limits = c(-0.4,0)) + scale_y_continuous(limits =c(-0.25,0.1))+
  geom_point(size = 3, alpha = 0.5) +
  geom_text_repel(data = mut_residue_scores %>% distinct(end, endo_avg, sko_avg), aes(x= log10(endo_avg), y= log10(sko_avg), label = end), color = "red", size = 6, segment.color = "orange") + 
  stat_smooth (geom="line", alpha=0.3, size=3, span=0.5, method = lm) + stat_poly_eq(use_label(c("R2")), size = 4) +
  labs(x= "Endogenous STIM1 mean score", y= "STIM1 KO mean score") +
  theme(axis.text=element_text(size=14),
        axis.title=element_text(size=14))+
  NULL; MutRes_endo_v_ko_scatterplot

MutRes_endo_v_bac_scatterplot <- ggplot(data = mut_residue_scores %>% distinct(end, endo_avg, bac_avg), aes(x= log10(endo_avg), y= log10(bac_avg)))+ 
  # scale_x_continuous(limits = c(-0.4,0)) + scale_y_continuous(limits =c(-0.25,0.1))+
  geom_point(size = 3, alpha = 0.5) +
  geom_text_repel(data = mut_residue_scores %>% distinct(end, endo_avg, bac_avg), aes(x= log10(endo_avg), y= log10(bac_avg), label = end), color = "red", size = 3, segment.color = "orange") + 
  stat_smooth (geom="line", alpha=0.3, size=1, span=0.5, method = lm) + stat_poly_eq(use_label(c("R2"))) +
  labs(x= "Average Endogenous STIM1 score", y= "Average Bacteria score") +
  NULL; MutRes_endo_v_bac_scatterplot

MutRes_ko_v_bac_scatterplot <- ggplot(data = mut_residue_scores %>% distinct(end, bac_avg, sko_avg), aes(x= log10(sko_avg), y= log10(bac_avg)))+ 
  # scale_x_continuous(limits = c(-0.4,0)) + scale_y_continuous(limits =c(-0.25,0.1))+
  geom_point(size = 3, alpha = 0.5) +
  geom_text_repel(data = mut_residue_scores %>% distinct(end, bac_avg, sko_avg), aes(x= log10(sko_avg), y= log10(bac_avg), label = end), color = "red", size = 3, segment.color = "orange") + 
  stat_smooth (geom="line", alpha=0.3, size=1, span=0.5, method = lm) + stat_poly_eq(use_label(c("R2"))) +
  labs(x= "Average Endogenous STIM1 score", y= "Average Bacteria score") +
  NULL; MutRes_ko_v_bac_scatterplot

#synonymous variants
syn_scores = combined2 %>% filter(type == "syn")

syn_endo_v_ko_scatterplot <- ggplot(data = syn_scores, aes(x= log10(endo_ratio), y= log10(sko_ratio)))+ 
  # scale_x_continuous(limits = c(-0.4,0)) + scale_y_continuous(limits =c(-0.25,0.1))+
  geom_point(size = 3, alpha = 0.5) +
  geom_text_repel(data = syn_scores, aes(x= log10(endo_ratio), y= log10(sko_ratio), label = position), color = "red", size = 3, segment.color = "orange") + 
  stat_smooth (geom="line", alpha=0.3, size=1, span=0.5, method = lm) + stat_poly_eq(use_label(c("R2"))) +
  labs(x= "Endogenous score", y= "Knockout score") +
  NULL; syn_endo_v_ko_scatterplot

syn_endo_v_bac_scatterplot <- ggplot(data = syn_scores, aes(x= log10(endo_ratio), y= log10(bac_ratio)))+ 
  # scale_x_continuous(limits = c(-0.4,0)) + scale_y_continuous(limits =c(-0.25,0.1))+
  geom_point(size = 3, alpha = 0.5) +
  geom_text_repel(data = syn_scores, aes(x= log10(endo_ratio), y= log10(bac_ratio), label = position), color = "red", size = 3, segment.color = "orange") + 
  stat_smooth (geom="line", alpha=0.3, size=1, span=0.5, method = lm) + stat_poly_eq(use_label(c("R2"))) +
  labs(x= "Endogenous score", y= "Bacterial score") +
  NULL; syn_endo_v_bac_scatterplot

syn_ko_v_bac_scatterplot <- ggplot(data = syn_scores, aes(x= log10(sko_ratio), y= log10(bac_ratio)))+ 
  # scale_x_continuous(limits = c(-0.4,0)) + scale_y_continuous(limits =c(-0.25,0.1))+
  geom_point(size = 3, alpha = 0.5) +
  geom_text_repel(data = syn_scores, aes(x= log10(sko_ratio), y= log10(bac_ratio), label = position), color = "red", size = 3, segment.color = "orange") + 
  stat_smooth (geom="line", alpha=0.3, size=1, span=0.5, method = lm) + stat_poly_eq(use_label(c("R2"))) +
  labs(x= "Knockout score", y= "Bacterial score") +
  NULL; syn_ko_v_bac_scatterplot

#nonsense variants
non_scores = combined2 %>% filter(type == "non")

non_endo_v_ko_scatterplot <- ggplot(data = non_scores, aes(x= log10(endo_ratio), y= log10(sko_ratio)))+ 
  # scale_x_continuous(limits = c(-0.4,0)) + scale_y_continuous(limits =c(-0.25,0.1))+
  geom_point(size = 3, alpha = 0.5) +
  geom_text_repel(data = non_scores, aes(x= log10(endo_ratio), y= log10(sko_ratio), label = position), color = "red", size = 3, segment.color = "orange") + 
  stat_smooth (geom="line", alpha=0.3, size=1, span=0.5, method = lm) + stat_poly_eq(use_label(c("R2"))) +
  labs(x= "Endogenous score", y= "Knockout score") +
  NULL; non_endo_v_ko_scatterplot

non_endo_v_bac_scatterplot <- ggplot(data = non_scores, aes(x= log10(endo_ratio), y= log10(bac_ratio)))+ 
  # scale_x_continuous(limits = c(-0.4,0)) + scale_y_continuous(limits =c(-0.25,0.1))+
  geom_point(size = 3, alpha = 0.5) +
  geom_text_repel(data = non_scores, aes(x= log10(endo_ratio), y= log10(bac_ratio), label = position), color = "red", size = 3, segment.color = "orange") + 
  stat_smooth (geom="line", alpha=0.3, size=1, span=0.5, method = lm) + stat_poly_eq(use_label(c("R2"))) +
  labs(x= "Endogenous score", y= "Bacterial score") +
  NULL; non_endo_v_bac_scatterplot

non_ko_v_bac_scatterplot <- ggplot(data = non_scores, aes(x= log10(sko_ratio), y= log10(bac_ratio)))+ 
  # scale_x_continuous(limits = c(-0.4,0)) + scale_y_continuous(limits =c(-0.25,0.1))+
  geom_point(size = 3, alpha = 0.5) +
  geom_text_repel(data = non_scores, aes(x= log10(sko_ratio), y= log10(bac_ratio), label = position), color = "red", size = 3, segment.color = "orange") + 
  stat_smooth (geom="line", alpha=0.3, size=1, span=0.5, method = lm) + stat_poly_eq(use_label(c("R2"))) +
  labs(x= "Knockout score", y= "Bacterial score") +
  NULL; non_ko_v_bac_scatterplot
```
```{r Compile all of the r-squared values from above to figure out inter-dataset relationships, fig.height = 6, fig.width = 5}
method2 <- "spearman"

mut_residue_plotting <- mut_residue_scores %>% group_by(end) %>% summarize(endo_avg = unique(endo_avg), sko_avg = unique(sko_avg), bac_avg = unique(bac_avg))

dataset_correlations <- data.frame("label" = c("Endo_v_KO", "Endo_v_Bac", "KO_v_Bac"),
  "missense" = c(cor(log10(combined2$endo_ratio), log10(combined2$sko_ratio), use = "complete", method = method2)^2,
                 cor(log10(combined2$endo_ratio), log10(combined2$bac_ratio), use = "complete", method = method2)^2,
                 cor(log10(combined2$sko_ratio), log10(combined2$bac_ratio), use = "complete", method = method2)^2),
  "position" = c(cor(log10(positional_scores$endo_avg), log10(positional_scores$sko_avg), use = "complete", method = method2)^2,
                 cor(log10(positional_scores$endo_avg), log10(positional_scores$bac_avg), use = "complete", method = method2)^2,
                 cor(log10(positional_scores$sko_avg), log10(positional_scores$bac_avg), use = "complete", method = method2)^2),
  "mutated" = c(cor(log10(mut_residue_plotting$endo_avg), log10(mut_residue_plotting$sko_avg), use = "complete", method = method2)^2,
                 cor(log10(mut_residue_plotting$endo_avg), log10(mut_residue_plotting$bac_avg), use = "complete", method = method2)^2,
                 cor(log10(mut_residue_plotting$sko_avg), log10(mut_residue_plotting$bac_avg), use = "complete", method = method2)^2),
  "synonymous" = c(cor(log10(syn_scores$endo_ratio), log10(syn_scores$sko_ratio), use = "complete", method = method2)^2,
                 cor(log10(syn_scores$endo_ratio), log10(syn_scores$bac_ratio), use = "complete", method = method2)^2,
                 cor(log10(syn_scores$sko_ratio), log10(syn_scores$bac_ratio), use = "complete", method = method2)^2),
  "nonsense" = c(cor(log10(non_scores$endo_ratio), log10(non_scores$sko_ratio), use = "complete", method = method2)^2,
                 cor(log10(non_scores$endo_ratio), log10(non_scores$bac_ratio), use = "complete", method = method2)^2,
                 cor(log10(non_scores$sko_ratio), log10(non_scores$bac_ratio), use = "complete", method = method2)^2)
)

dataset_correlations_melt <- melt(dataset_correlations, id = "label")
dataset_correlations_melt$label <- factor(dataset_correlations_melt$label, levels = rev(unique(dataset_correlations_melt$label)))

Dataset_spearman_comparison_plot <- ggplot() + 
  geom_tile(data = dataset_correlations_melt, aes(x = variable, y = label, fill = value), color = "grey80") +
  geom_text(data = dataset_correlations_melt, aes(x = variable, y = label, label = round(value,2))) +
  scale_fill_gradient(low = "white", high = "red") +
  labs(x = NULL, y = NULL) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) + 
  NULL; Dataset_spearman_comparison_plot
ggsave(file = "Plots/main/fig5a_Dataset_spearman_comparison_plot.pdf", Dataset_spearman_comparison_plot, height = 2, width = 4)

## Perhaps show all the pairwise correlations as a supp figure

mis_es <- ggplot() + geom_point(data = combined2, aes(x = log10(endo_ratio), y = log10(sko_ratio)), alpha = 0.1) + labs(x = "Endogenous", y = "Knockout") + theme(panel.grid = element_blank()) + scale_x_continuous(limits = c(-2.5, 0.5)) + scale_y_continuous(limits = c(-1, 0.5))
mis_eb <- ggplot() + geom_point(data = combined2, aes(x = log10(endo_ratio), y = log10(bac_ratio)), alpha = 0.1) + labs(x = "Endogenous", y = "Bacterial") + theme(panel.grid = element_blank()) + scale_x_continuous(limits = c(-2.5, 0.5)) + scale_y_continuous(limits = c(-1.5, 0.5))
mis_sb <- ggplot() + geom_point(data = combined2, aes(x = log10(sko_ratio), y = log10(bac_ratio)), alpha = 0.1) + labs(x = "Knockout", y = "Bacterial") + theme(panel.grid = element_blank()) + scale_x_continuous(limits = c(-1, 0.5)) + scale_y_continuous(limits = c(-1.5, 0.5))
pos_es <- ggplot() + geom_point(data = positional_scores, aes(x = log10(endo_avg), y = log10(sko_avg)), alpha = 0.2) + labs(x = "Endogenous", y = "Knockout") + theme(panel.grid = element_blank()) + scale_x_continuous(limits = c(-2.5, 0.5)) + scale_y_continuous(limits = c(-1, 0.5))
pos_eb <- ggplot() + geom_point(data = positional_scores, aes(x = log10(endo_avg), y = log10(bac_avg)), alpha = 0.2) + labs(x = "Endogenous", y = "Bacterial") + theme(panel.grid = element_blank()) + scale_x_continuous(limits = c(-2.5, 0.5)) + scale_y_continuous(limits = c(-1.5, 0.5))
pos_sb <- ggplot() + geom_point(data = positional_scores, aes(x = log10(sko_avg), y = log10(bac_avg)), alpha = 0.2) + labs(x = "Knockout", y = "Bacterial") + theme(panel.grid = element_blank()) + scale_x_continuous(limits = c(-1, 0.5)) + scale_y_continuous(limits = c(-1.5, 0.5))
mut_es <- ggplot() + geom_point(data = mut_residue_plotting, aes(x = log10(endo_avg), y = log10(sko_avg)), alpha = 0.2) + labs(x = "Endogenous", y = "Knockout") + theme(panel.grid = element_blank()) + scale_x_continuous(limits = c(-2.5, 0.5)) + scale_y_continuous(limits = c(-1, 0.5))
mut_eb <- ggplot() + geom_point(data = mut_residue_plotting, aes(x = log10(endo_avg), y = log10(bac_avg)), alpha = 0.2) + labs(x = "Endogenous", y = "Bacterial") + theme(panel.grid = element_blank()) + scale_x_continuous(limits = c(-2.5, 0.5)) + scale_y_continuous(limits = c(-1.5, 0.5))
mut_sb <- ggplot() + geom_point(data = mut_residue_plotting, aes(x = log10(sko_avg), y = log10(bac_avg)), alpha = 0.2) + labs(x = "Knockout", y = "Bacterial") + theme(panel.grid = element_blank()) + scale_x_continuous(limits = c(-1, 0.5)) + scale_y_continuous(limits = c(-1.5, 0.5))
syn_es <- ggplot() + geom_point(data = syn_scores, aes(x = log10(endo_ratio), y = log10(sko_ratio)), alpha = 0.2) + labs(x = "Endogenous", y = "Knockout") + theme(panel.grid = element_blank()) + scale_x_continuous(limits = c(-2.5, 0.5)) + scale_y_continuous(limits = c(-1, 0.5))
syn_eb <- ggplot() + geom_point(data = syn_scores, aes(x = log10(endo_ratio), y = log10(bac_ratio)), alpha = 0.2) + labs(x = "Endogenous", y = "Bacterial") + theme(panel.grid = element_blank()) + scale_x_continuous(limits = c(-2.5, 0.5)) + scale_y_continuous(limits = c(-1.5, 0.5))
syn_sb <- ggplot() + geom_point(data = syn_scores, aes(x = log10(sko_ratio), y = log10(bac_ratio)), alpha = 0.2) + labs(x = "Knockout", y = "Bacterial") + theme(panel.grid = element_blank()) + scale_x_continuous(limits = c(-1, 0.5)) + scale_y_continuous(limits = c(-1.5, 0.5))
non_es <- ggplot() + geom_point(data = non_scores, aes(x = log10(endo_ratio), y = log10(sko_ratio)), alpha = 0.2) + labs(x = "Endogenous", y = "Knockout") + theme(panel.grid = element_blank()) + scale_x_continuous(limits = c(-2.5, 0.5)) + scale_y_continuous(limits = c(-1, 0.5))
non_eb <- ggplot() + geom_point(data = non_scores, aes(x = log10(endo_ratio), y = log10(bac_ratio)), alpha = 0.2) + labs(x = "Endogenous", y = "Bacterial") + theme(panel.grid = element_blank()) + scale_x_continuous(limits = c(-2.5, 0.5)) + scale_y_continuous(limits = c(-1.5, 0.5))
non_sb <- ggplot() + geom_point(data = non_scores, aes(x = log10(sko_ratio), y = log10(bac_ratio)), alpha = 0.2) + labs(x = "Knockout", y = "Bacterial") + theme(panel.grid = element_blank()) + scale_x_continuous(limits = c(-1, 0.5)) + scale_y_continuous(limits = c(-1.5, 0.5))

pairwise_scatterplots <- (mis_es | mis_eb | mis_sb) / (pos_es | pos_eb | pos_sb) / (mut_es | mut_eb | mut_sb) / (syn_es | syn_eb | syn_sb) / (non_es | non_eb | non_sb) 
pairwise_scatterplots

ggsave("Revision_1/plots_r1/supplement_r1/Supp_fig_pairwise_correlations.pdf", pairwise_scatterplots, height = 7, width = 5)


```

Synonymous distribution - to make the pink box for the rest of the figures
```{r}
syn_endo_range <- c(log10(quantile(syn_scores$endo_ratio, 0.025, na.rm = T)),log10(quantile(syn_scores$endo_ratio, 0.975, na.rm = T)))
syn_sko_range <- c(log10(quantile(syn_scores$sko_ratio, 0.025, na.rm = T)),log10(quantile(syn_scores$sko_ratio, 0.975, na.rm = T)))
```

2. EF-hand motif
```{r Compare mutational signatures for coordination residues in the EF-hand motif}

coord_res <- c(76, 78, 80, 81, 87)

coord_subset <- combined2 %>% filter(position %in% coord_res & type == "mis")
coord_subset$res <- substr(coord_subset$variant,1,3)

Coord_res_scatterplots <- ggplot(data = coord_subset, aes(x = log10(endo_ratio), y = log10(sko_ratio))) + 
  geom_point(data = coord_subset, aes(x = log10(endo_ratio), y = log10(sko_ratio)), alpha = 0.4) +
  geom_text_repel(data = coord_subset %>% filter(log10(endo_ratio) < -2 | log10(sko_ratio) < -0.75), aes(x = log10(endo_ratio), y = log10(sko_ratio), label = end), color = "blue", size = 2, segment.color = "orange") +
  geom_text(data = coord_subset %>% filter(variant == "N80D"), aes(x = log10(endo_ratio)*0.5, y = log10(sko_ratio)*1.5, label = end), color = "red", size = 2) +
  geom_point(data = coord_subset %>% filter(variant == "N80D"), aes(x = log10(endo_ratio), y = log10(sko_ratio)), alpha = 0.4, color = "red") +
  stat_smooth (geom="line", alpha=0.1, size=1, span=0.5, method = lm, color = "green") + stat_poly_eq(use_label(c("R2"))) +
  facet_grid(cols = vars(position)) +
  scale_x_continuous(breaks = pretty_breaks(n = 3)) +
  scale_y_continuous(breaks = pretty_breaks(n = 3), limits = c(-1.5,0.5)) +
  geom_rect(aes(xmin=syn_endo_range[1], xmax=syn_endo_range[2], ymin=syn_sko_range[1], ymax=syn_sko_range[2]), fill = "pink", alpha = 0.02) + 
  labs(x = "Standard score", y = "Knockout score") +
  NULL; Coord_res_scatterplots
ggsave(file = "Plots/main/fig6b_Coord_res_scatterplots.pdf", Coord_res_scatterplots, height = 1.75, width = 6)

```

```{r comparing D76 and D78 mutants}
d76 <- combined2 %>% filter(position %in% c(76))
d78 <- combined2 %>% filter(position %in% c(78))

d76_d78 <- merge(d76[,c("end","endo_ratio","sko_ratio")], d78[,c("end","endo_ratio","sko_ratio")], by = "end")
colnames(d76_d78) <- c("end","d76_endo_ratio","d76_sko_ratio","d78_endo_ratio","d78_sko_ratio")

d76_d78_LP_plot <- ggplot() + theme(panel.grid = element_blank()) + 
  scale_x_continuous(limits = c(-2.5, 0.5)) + scale_y_continuous(limits = c(-2.5, 0.25)) +
  geom_point(data = d76_d78, aes(x = log10(d76_endo_ratio), y = log10(d78_endo_ratio)), alpha = 0.25) +
  geom_text_repel(data = d76_d78, aes(x = log10(d76_endo_ratio), y = log10(d78_endo_ratio), label = end), color = "red", alpha = 0.75, size = 3, segment.alpha = 0.25) +
  labs(x = "Asp76 score", y = "Asp78 score", title = "LP cells") +
  NULL; d76_d78_LP_plot
ggsave(file = "plots/main/fig6c_d76_d78_LP_plot.pdf", d76_d78_LP_plot, height = 2.5, width = 2.5)

d76_d78_knockout_plot <- ggplot() + theme(panel.grid = element_blank()) + 
  scale_x_continuous(limits = c(-2.5, 0.5)) + scale_y_continuous(limits = c(-2.5, 0.25)) +
  geom_point(data = d76_d78, aes(x = log10(d76_sko_ratio), y = log10(d78_sko_ratio)), alpha = 0.25) +
  geom_text_repel(data = d76_d78 %>% filter(end %in% c("C","M")), aes(x = log10(d76_sko_ratio), y = log10(d78_sko_ratio), label = end), color = "red", alpha = 0.75, size = 3, segment.alpha = 0.25) +
  labs(x = "Asp76 score", y = "Asp78 score", title = "Knockout LP cells") +
  NULL; d76_d78_knockout_plot
ggsave(file = "plots/main/fig6c_d76_d78_knockout_plot.pdf", d76_d78_knockout_plot, height = 2.5, width = 2.5)

## Print out some stats
cor(log10(d76_d78$d76_endo_ratio),log10(d76_d78$d78_endo_ratio), method = "pearson")^2
cor(log10(d76_d78$d76_endo_ratio),log10(d76_d78$d78_endo_ratio), method = "spearman")^2

d76
d77 <- combined2 %>% filter(position %in% c(77))
d78
a79 <- combined2 %>% filter(position %in% c(79))
n80 <- combined2 %>% filter(position %in% c(80))
g81 <- combined2 %>% filter(position %in% c(81))
d82 <- combined2 %>% filter(position %in% c(82))
v83 <- combined2 %>% filter(position %in% c(83))
d84 <- combined2 %>% filter(position %in% c(84))
v85 <- combined2 %>% filter(position %in% c(85))
e86 <- combined2 %>% filter(position %in% c(86))
e87 <- combined2 %>% filter(position %in% c(87))

efmotif_lp <- merge(merge(merge(merge(merge(merge(merge(merge(merge(merge(merge(d76[,c("end","endo_ratio")], d77[,c("end","endo_ratio")], by = "end"), d78[,c("end","endo_ratio")], by = "end"), a79[,c("end","endo_ratio")], by = "end"), n80[,c("end","endo_ratio")], by = "end"), g81[,c("end","endo_ratio")], by = "end"), d82[,c("end","endo_ratio")], by = "end"), v83[,c("end","endo_ratio")], by = "end"), d84[,c("end","endo_ratio")], by = "end"), v85[,c("end","endo_ratio")], by = "end"), e86[,c("end","endo_ratio")], by = "end"), e87[,c("end","endo_ratio")], by = "end")

colnames(efmotif_lp) <- c("end","d76","d77","d78","a79","n80","g81","d82","v83","d84","v85","e86","e87")

efmotif_lp2 <- efmotif_lp[,-1]; rownames(efmotif_lp2) <- efmotif_lp$end

efmotif_lp3_log10 <- log10(efmotif_lp2)

efmotif_lp4_imputed <- efmotif_lp3_log10

efmotif_lp4_imputed["K","d77"] <- efmotif_lp4_imputed["R","d77"]
efmotif_lp4_imputed["N","n80"] <- 0
efmotif_lp4_imputed["D","d82"] <- 0
efmotif_lp4_imputed["D","d84"] <- 0

row.order <- hclust(dist(efmotif_lp4_imputed))$order
col.order <- hclust(dist(t(efmotif_lp4_imputed)))$order

double_end_order <- rev(rownames(efmotif_lp4_imputed[row.order, col.order]))
double_label_order <- toupper(colnames(efmotif_lp4_imputed[row.order, col.order]))

efmotif_subset <- combined2 %>% filter(position %in% seq(76,87))
efmotif_subset$label <- ""

efmotif_subset$end <- factor(efmotif_subset$end, levels = double_end_order)
efmotif_subset$label <- paste0(efmotif_subset$start,efmotif_subset$position)

double_position_order2 <- gsub("[a-z]","",double_label_order)

efmotif_subset$position <- factor(efmotif_subset$position, levels = double_position_order2)
efmotif_subset$label <- factor(efmotif_subset$label, levels = double_label_order)

efmotif_clustered_heatmap <- ggplot() + 
  theme(axis.text.y = element_text(size = 7), axis.text.x = element_text(size = 7, angle = 45, hjust = 0.75)) + 
  labs(x = NULL, y = NULL) + 
  scale_fill_gradient2(low = "blue", mid = "white", high = "red") +
  geom_tile(data = efmotif_subset, aes(x = label, y = end, fill = log10(endo_ratio))) +
  NULL; efmotif_clustered_heatmap
ggsave(file = "plots/main/fig6d_efmotif_clustered_heatmap.pdf", efmotif_clustered_heatmap, height = 2.4, width = 3.5)
```
3. Alpha1 helix and orthologs
```{r Subsetting the EF-hand into three sections}

section1 <- combined2 %>% filter(type == "mis" & position %in% seq(64,71) & !(position %in% c(70))) %>% mutate(section = "1.a1")
section2 <- combined2 %>% filter(type == "mis" & position %in% seq(76,87)) %>% mutate(section = "2.ef-hand")
section3 <- combined2 %>% filter(type == "mis" & position %in% seq(89,97) & !(position %in% c(92,94))) %>% mutate(section = "3.a2")

endo <- data.frame(type = "endo", variant = c((section1$variant),(section2$variant),(section3$variant)),
                   score = c(log10(section1$endo_ratio),log10(section2$endo_ratio),log10(section3$endo_ratio)),
                   section = c(section1$section,section2$section,section3$section)) %>% filter(!is.na(score))

endo$quantile_norm <- (endo$score - quantile(endo$score,0.05)) / (quantile(endo$score,0.95) - quantile(endo$score,0.05))

sko <- data.frame(type = "sko", variant = c((section1$variant),(section2$variant),(section3$variant)),
                   score = c(log10(section1$sko_ratio),log10(section2$sko_ratio),log10(section3$sko_ratio)),
                   section = c(section1$section,section2$section,section3$section)) %>% filter(!is.na(score))

sko$quantile_norm <- (sko$score - quantile(sko$score,0.05)) / (quantile(sko$score,0.95) - quantile(sko$score,0.05))

section123 <- rbind(endo, sko)

ggplot() + geom_beeswarm(data = section123, aes(x = section, y = quantile_norm, color = type), dodge.width = 1, size = 0.5)

endo_sko_merge <- merge(endo, sko[,c("variant","quantile_norm")], by = "variant")
endo_sko_merge$quant_norm_ratio <- endo_sko_merge$quantile_norm.y - endo_sko_merge$quantile_norm.x

endo_sko_merge <- endo_sko_merge %>% arrange(quant_norm_ratio)

ggplot() + geom_segment(data = endo_sko_merge, aes(x = 1, xend = 2, y = quantile_norm.x, yend = quantile_norm.y), alpha = 0.1, size = 2) +
  facet_grid(cols = vars(section))

Endo_sko_ratio_density_plot <- ggplot() + geom_density(data = endo_sko_merge, aes(x = quant_norm_ratio, fill = section), alpha = 0.4) +
  NULL; Endo_sko_ratio_density_plot
ggsave(file = "Plots/main/fig7b_Endo_sko_ratio_density_plot.pdf", Endo_sko_ratio_density_plot, height = 1.25, width = 3.5)

## Now also show it as a scatterplot, but this time use the actual score rather than quantile normalized

segment123_for_scatterplot <- rbind(section1, section2, section3)
segment123_for_scatterplot$section <- as.character(segment123_for_scatterplot$section)

EFdomain_subsectioned_scatterplot <- ggplot() + theme(panel.grid = element_blank()) + 
  labs(x = "Standard score", y = "Knockout score") + 
  geom_point(data = segment123_for_scatterplot[,c("endo_ratio","sko_ratio","section")], aes(x = log10(endo_ratio), y = log10(sko_ratio), color = section), alpha = 0.3, size = 0.8) +
  NULL; EFdomain_subsectioned_scatterplot
ggsave(file = "Plots/main/fig7a_EFdomain_subsectioned_scatterplot.pdf", EFdomain_subsectioned_scatterplot, height = 2, width = 4)

segment123_positional <- segment123_for_scatterplot %>% group_by(position, section) %>% summarize(endo_positional = mean(endo_positional), sko_positional = mean(sko_positional))

not_segment123_positional <- combined2 %>% filter(!(position %in% unique(segment123_for_scatterplot$position))) %>% filter(!position %in% c(70,92,94)) %>% group_by(position) %>% summarize(endo_positional = mean(endo_positional), sko_positional = mean(sko_positional))

subsectioned_positinal_labels <- c(63, 83, 84, 87)

EFdomain_position_subsectioned_scatterplot <- ggplot() + theme(panel.grid = element_blank()) + 
  labs(x = "Standard score", y = "Knockout score") + 
  geom_point(data = segment123_positional[,c("endo_positional","sko_positional","section")], aes(x = log10(endo_positional), y = log10(sko_positional), color = section), alpha = 0.8, size = 0.8) +
  geom_point(data = not_segment123_positional, aes(x = log10(endo_positional), y = log10(sko_positional)), alpha = 0.4, size = 0.8) +
  geom_text_repel(data = segment123_positional %>% filter(position %in% subsectioned_positinal_labels), aes(x = log10(endo_positional), y = log10(sko_positional), label = position)) +
  geom_text_repel(data = not_segment123_positional %>% filter(position %in% subsectioned_positinal_labels), aes(x = log10(endo_positional), y = log10(sko_positional), label = position)) +
  NULL; EFdomain_position_subsectioned_scatterplot
ggsave(file = "Plots/main/fig7c_EFdomain_position_subsectioned_scatterplot.pdf", EFdomain_position_subsectioned_scatterplot, height = 2, width = 3.5)

```

```{r Orthologs}
human <- "LSFEAVRNIHKLMDDDANGDVDVEESDEFLREDLNYHDPTVKHSTFHGEDKLISVEDLWKAWKSSEVY"
e66d <- "LSFDAVRNIHKLMDDDANGDVDVEESDEFLREDLNYHDPTVKHSTFHGEDKLISVEDLWKAWKSSEVY"
r69c <- "LSFEAVCNIHKLMDDDANGDVDVEESDEFLREDLNYHDPTVKHSTFHGEDKLISVEDLWKAWKSSEVY"
n70s <- "LSFEAVRSIHKLMDDDANGDVDVEESDEFLREDLNYHDPTVKHSTFHGEDKLISVEDLWKAWKSSEVY"
v103a <- "LSFEAVRNIHKLMDDDANGDVDVEESDEFLREDLNYHDPTAKHSTFHGEDKLISVEDLWKAWKSSEVY"
s106n <- "LSFEAVRNIHKLMDDDANGDVDVEESDEFLREDLNYHDPTVKHNTFHGEDKLISVEDLWKAWKSSEVY"
e111g <- "LSFEAVRNIHKLMDDDANGDVDVEESDEFLREDLNYHDPTVKHSTFHGGDKLISVEDLWKAWKSSEVY"
s126a <- "LSFEAVRNIHKLMDDDANGDVDVEESDEFLREDLNYHDPTVKHSTFHGEDKLISVEDLWKAWKASEVY"
e66d_n70s <- "LSFDAVRSIHKLMDDDANGDVDVEESDEFLREDLNYHDPTVKHSTFHGEDKLISVEDLWKAWKSSEVY"
e66d_s126a <- "LSFDAVRNIHKLMDDDANGDVDVEESDEFLREDLNYHDPTVKHSTFHGEDKLISVEDLWKAWKASEVY"
v68i_l74q_a123s <- "LSFEAIRNIHKQMDDDANGDVDVEESDEFLREDLNYHDPTVKHSTFHGEDKLISVEDLWKSWKSSEVY"
v68i_l92i_s100a <- "LSFEAIRNIHKLMDDDANGDVDVEESDEFIREDLNYHDSTVKHSTFHGEDKLISVEDLWKAWKSSEVY"
e66d_n70s_s106n <- "LSFDAVRSIHKLMDDDANGDVDVEESDEFLREDLNYHDPTVKHNTFHGEDKLISVEDLWKAWKSSEVY"
l74q_d82n_s126a <- "LSFEAVRNIHKQMDDDANGNVDVEESDEFLREDLNYHDPTVKHSTFHGEDKLISVEDLWKAWKASEVY"
l74q_a123s_s126a <- "LSFEAVRNIHKQMDDDANGDVDVEESDEFLREDLNYHDPAVKHSTFHGEDKLISVEDLWKSWKASEVY"

ortholog_frame <- data.frame("position" = seq(63,63+67))

for(x in 1:nchar(human)){
  ortholog_frame$human[x] <- substr(human,x,x)
  ortholog_frame$e66d[x] <- substr(e66d,x,x)
  ortholog_frame$r69c[x] <- substr(r69c,x,x)
  ortholog_frame$n70s[x] <- substr(n70s,x,x)
  ortholog_frame$v103a[x] <- substr(v103a,x,x)
  ortholog_frame$s106n[x] <- substr(s106n,x,x)
  ortholog_frame$e111g[x] <- substr(e111g,x,x)
  ortholog_frame$s126a[x] <- substr(s126a,x,x)
  ortholog_frame$e66d_n70s[x] <- substr(e66d_n70s,x,x)
  ortholog_frame$e66d_s126a[x] <- substr(e66d_s126a,x,x)
  ortholog_frame$v68i_l74q_a123s[x] <- substr(v68i_l74q_a123s,x,x)
  ortholog_frame$v68i_l92i_s100a[x] <- substr(v68i_l92i_s100a,x,x)
  ortholog_frame$e66d_n70s_s106n[x] <- substr(e66d_n70s_s106n,x,x)
  ortholog_frame$l74q_d82n_s126a[x] <- substr(l74q_d82n_s126a,x,x)
  ortholog_frame$l74q_t102a_a123s_s126a[x] <- substr(l74q_a123s_s126a,x,x)
}

ortholog_frame_melt <- melt(ortholog_frame, id = "position")
ortholog_frame_melt$variable <- toupper(ortholog_frame_melt$variable)
ortholog_frame_melt$variable <- factor(ortholog_frame_melt$variable, levels = rev(toupper(c("human","e66d","r69c","n70s","v103a","s106n","e111g","s126a","e66d_n70s","e66d_s126a","e66d_n70s_s106n","v68i_l92i_s100a","v68i_l74q_a123s","l74q_d82n_s126a","l74q_t102a_a123s_s126a"))))

ortholog_frame2 <- ortholog_frame

for(x in 3:ncol(ortholog_frame2)){
  for(y in 1:nrow(ortholog_frame2)){
   if(ortholog_frame2[y,x] == ortholog_frame2[y,"human"]){ortholog_frame2[y,x] <- ""}
  }
}

ortholog_frame_melt2 <- melt(ortholog_frame2, id = "position")

ortholog_frame_melt2$variable <- toupper(ortholog_frame_melt2$variable)
ortholog_frame_melt2$variable <- factor(ortholog_frame_melt2$variable, levels = rev(toupper(c("human","e66d","r69c","n70s","v103a","s106n","e111g","s126a","e66d_n70s","e66d_s126a","e66d_n70s_s106n","v68i_l92i_s100a","v68i_l74q_a123s","l74q_d82n_s126a","l74q_t102a_a123s_s126a"))))

orthologs_efhands_plot <- ggplot() + theme(panel.grid = element_blank()) +
  geom_text(data = ortholog_frame_melt, aes(x = position, y = variable, label = value), size = 2, alpha = 0.1) +
  geom_text(data = ortholog_frame_melt2, aes(x = position, y = variable, label = value), size = 2) +
    geom_rect(aes(xmin = 64, xmax = 74, ymin = "HUMAN", ymax = "L74Q_T102A_A123S_S126A"), alpha = 0.15, fill = "red") +
  geom_rect(aes(xmin = 76, xmax = 87, ymin = "HUMAN", ymax = "L74Q_T102A_A123S_S126A"), alpha = 0.15, fill = "green") +
  geom_rect(aes(xmin = 89, xmax = 97, ymin = "HUMAN", ymax = "L74Q_T102A_A123S_S126A"), alpha = 0.15, fill = "blue") +
    geom_rect(aes(xmin = 101, xmax = 107, ymin = "HUMAN", ymax = "L74Q_T102A_A123S_S126A"), alpha = 0.15, fill = muted("red")) +
  geom_rect(aes(xmin = 109, xmax = 117, ymin = "HUMAN", ymax = "L74Q_T102A_A123S_S126A"), alpha = 0.15, fill = muted("green")) +
  geom_rect(aes(xmin = 118, xmax = 126, ymin = "HUMAN", ymax = "L74Q_T102A_A123S_S126A"), alpha = 0.15, fill = muted("blue")) +
  labs(x = "Position in mammalian STIM1 protein", y = NULL) +
  NULL; orthologs_efhands_plot
ggsave(file = "plots/main/fig7d_orthologs_efhands_plot.pdf", orthologs_efhands_plot, height = 2.5, width = 6.25)

#bringing in variants caused by single nucleotide variation (SNV)
snv <- read.table(file = "Data/STIM1_SNV_table.tsv", sep = "\t", header = T)

snv2 <- snv %>% group_by(variant) %>% summarize(variant = variant, snv = "yes") %>% distinct()
snv3 <- merge(combined2, snv2, by = "variant")

snv3$endo_score <- log10(snv3$endo_ratio)
snv3$sko_score <- log10(snv3$sko_ratio)
snv3$bac_score <- log10(snv3$bac_ratio)

snv3_mis <- snv3 %>% filter(type == "mis")
snv3_syn <- snv3 %>% filter(type == "syn")
ortholog = read.csv("data/stim1_orthologs.csv") %>% filter(!is.na(Example))

ortholog2 <- merge(ortholog %>% filter(Num_diff == 1 & Interpretation == "ortholog" & Redundant == "no"), snv3, by = "variant", all.x = T)
ortholog3 <- merge(ortholog %>% filter(Num_diff == 1 & Interpretation == "ortholog" & Redundant == "no"), combined2, by = "variant", all.x = T) 
ortholog4 <- merge(ortholog %>% filter(Num_diff == 1 & Redundant != "yes"), combined2, by = "variant", all.x = T) 

Ortholog_variants <- ggplot() + scale_x_continuous(limits = log10(c(0.003,3))) + scale_y_continuous(limits = log10(c(0.07,5))) +
  geom_point(data = combined2, aes(x = log10(endo_ratio), y = log10(sko_ratio)), alpha = 0.05, size = 1) +
  geom_rect(aes(xmin=syn_endo_range[1], xmax=syn_endo_range[2], ymin=syn_sko_range[1], ymax=syn_sko_range[2]), fill = "pink", alpha = 0.5) + 
  #geom_point(data = ortholog4, aes(x = log10(endo_ratio), y = log10(sko_ratio)), shape = 21, fill = "pink", alpha = 0.9, size = 1.5, color = "black") +
  geom_point(data = ortholog3, aes(x = log10(endo_ratio), y = log10(sko_ratio)), shape = 21, fill = "red", alpha = 0.9, size = 1.5, color = "black") +
  geom_text_repel(data = ortholog3, aes(x = log10(endo_ratio), y = log10(sko_ratio), label = variant), shape = 21, fill = "red", alpha = 0.9, color = "black") +
  labs(x = "Standard score", y = "Knockout score") +
  theme(panel.grid = element_blank()) +
  NULL; Ortholog_variants
ggsave(file = "Plots/main/fig7e_Ortholog_variants.pdf", Ortholog_variants, height = 2.25, width = 2.5)

```

##STIM1 cEF-hand variants invovled in human disease
```{r Subset for SNVs}
Endo_snv_histogram <- ggplot() + 
  theme(panel.grid.minor = element_blank(), panel.grid.major = element_blank()) +
  labs(x = "Standard score", y = "Missense\nvariants") +
  geom_histogram(data = combined2 %>% filter(type == "mis"), aes(x = log10(endo_ratio)), bins = 20, fill = "cyan3", color = "black", size = 0.5, alpha = 0.6) +
  geom_histogram(data = snv3_mis, aes(x = log10(endo_ratio)), bins = 20, fill = "grey40", color = "black", size = 0.5, alpha = 0.6) +
  geom_histogram(data = snv3_syn, aes(x = log10(endo_ratio)), bins = 20, fill = "pink", color = "black", size = 0.5, alpha = 0.6) +
  NULL; Endo_snv_histogram
ggsave(file = "Plots/main/fig8a_Endo_snv_histogram.pdf", Endo_snv_histogram, height = 1.2, width = 2.55)

Sko_snv_histogram <- ggplot() + 
  theme(panel.grid.minor = element_blank(), panel.grid.major = element_blank()) +
  labs(x = "Knockout score", y = "Missense\nvariants") +
  geom_histogram(data = combined2 %>% filter(type == "mis"), aes(x = log10(sko_ratio)), bins = 20, fill = "cyan3", color = "black", size = 0.5, alpha = 0.6) +
  geom_histogram(data = snv3_mis, aes(x = log10(sko_ratio)), bins = 20, fill = "grey40", color = "black", size = 0.5, alpha = 0.6) +
  geom_histogram(data = snv3_syn, aes(x = log10(sko_ratio)), bins = 20, fill = "pink", color = "black", size = 0.5, alpha = 0.6) +
  NULL; Sko_snv_histogram
ggsave(file = "Plots/main/fig8a_Sko_snv_histogram.pdf", Sko_snv_histogram, height = 1.2, width = 2.50)

SNV_variants <- ggplot() + 
  theme(panel.grid = element_blank()) +
  geom_rect(aes(xmin=syn_endo_range[1], xmax=syn_endo_range[2], ymin=syn_sko_range[1], ymax=syn_sko_range[2]), fill = "pink", alpha = 0.5) +
  geom_point(data = combined2 %>% filter(!(variant %in% snv3$variant)), 
             aes(x = log10(endo_ratio), y = log10(sko_ratio)), alpha = 0.25, size = 0.5, shape = 19, color = "cyan3") +
  geom_point(data = snv3, aes(x = log10(endo_ratio), y = log10(sko_ratio)), shape = 19, alpha = 0.6, size = 0.5, color = "grey40") +
  labs(x = "Standard score", y = "Knockout score") +
  NULL; SNV_variants
ggsave(file = "Plots/main/fig8a_SNV_variants.pdf", SNV_variants, height = 2.3, width = 2.5)
```
```{r ClinVar and GnomAD}
gnomad = read.csv("data/stim1_gnomad.csv")

other_healthy = read.csv("data/stim1_curated.csv") %>% filter(RGC_MCPS == "yes" & GnomAD != "yes")

gnomad1 = gnomad %>% select(variant)
other_healthy1 = other_healthy %>% select(variant)
gnomad2 = bind_rows(gnomad1, other_healthy1)

gnomad3 <- merge(gnomad2, snv3, by = "variant")

Gnomad_variants <- ggplot() + 
  # scale_x_log10(limits = c(0.01,3)) + scale_y_log10(limits = c(0.1,5)) +
  theme(panel.grid = element_blank()) +
  geom_point(data = snv3, aes(x = log10(endo_ratio), y = log10(sko_ratio)), alpha = 0.1, size = 1) +
  geom_rect(aes(xmin=syn_endo_range[1], xmax=syn_endo_range[2], ymin=syn_sko_range[1], ymax=syn_sko_range[2]), fill = "pink", alpha = 0.3) + 
  geom_point(data = gnomad3 %>% filter(variant != "K73R"), aes(x = log10(endo_ratio), y = log10(sko_ratio)), shape = 21, fill = "pink", alpha = 0.9, size = 1.5, color = "black") +
  geom_point(data = gnomad3 %>% filter(variant == "K73R"), aes(x = log10(endo_ratio), y = log10(sko_ratio)), shape = 21, fill = "red", alpha = 0.9, size = 1.5, color = "black") +
  labs(x = "Standard score", y = "Knockout score") +
  NULL; Gnomad_variants
ggsave(file = "Plots/main/fig8b_Gnomad_variants.pdf", Gnomad_variants, height = 1.4, width = 2)

##stopped here
variants = read.csv("data/stim1_curated.csv")

clinvar <- variants %>% filter(!is.na(ClinVar) & !(ClinVar %in% c("NA"))) %>% filter(Position >= 66 & Position <= 98)
clinvar2 <- merge(clinvar, snv3, by = "variant")
clinvar2$Disease1 <- factor(clinvar2$Disease1, levels = c("Myopathy",NA))

Clinvar_variants2 <- ggplot() + #scale_x_log10(limits = c(0.01,3)) + scale_y_log10(limits = c(0.1,5)) +
  theme(panel.grid = element_blank()) +
  geom_point(data = snv3, aes(x = log10(endo_ratio), y = log10(sko_ratio)), alpha = 0.1, size = 1) +
  geom_rect(aes(xmin=syn_endo_range[1], xmax=syn_endo_range[2], ymin=syn_sko_range[1], ymax=syn_sko_range[2]), fill = "pink", alpha = 0.3) + 
  geom_point(data = clinvar2 %>% filter(Disease1 == "Myopathy"), aes(x = log10(endo_ratio), y = log10(sko_ratio), color = Disease1), alpha = 0.6, size = 1.5, fill = "purple", color = "black", shape = 21) +
  geom_point(data = clinvar2 %>% filter(is.na(Disease1)), aes(x = log10(endo_ratio), y = log10(sko_ratio), color = Disease1), alpha = 0.6, size = 1.5, fill = "orange", color = "black", shape = 21) +
  geom_text_repel(data = clinvar2 %>% filter(endo_ratio < 0.12), aes(x = log10(endo_ratio), y = log10(sko_ratio), label = variant), size = 2, segment.color = "orange", color = "red", nudge_x = -0.5) +
  labs(x = "Standard score", y = "Knockout score") +
  NULL; Clinvar_variants2
ggsave(file = "Plots/main/fig8c_Clinvar_variants2.pdf", Clinvar_variants2, height = 1.4, width = 2)

```

```{r COSMIC and cBioPortal}
cosmic = read.csv("data/stim1_cosmic.csv")

cosmic$variant <- substr(cosmic$AA.Mutation,3,12)

cosmic$res <- ""; cosmic$position <- ""; cosmic$mut <- "";
for(x in 1:nrow(cosmic)){
  cosmic$res[x] <- substr(cosmic$variant[x],1,1)
  cosmic$position[x] <- gsub("[^0-9]","",cosmic$variant[x])
  cosmic$mut[x] <- substr(gsub("[0-9]","",cosmic$variant[x]),2,2)
}
cosmic$position <- as.numeric(cosmic$position)


cosmic_table <- cosmic %>% group_by(position) %>% tally()

ggplot() + scale_y_log10() +
  geom_vline(xintercept = c(63,98)) +
  geom_point(data = cosmic_table, aes(x = position, y = n))


cosmic2 <- merge(cosmic, snv3, by = "variant")

#### Next bring in cBioPortal data

cbioportal = read.csv("data/stim1_cbioportal.csv")
#241001_STIM1_cBioPortal.tsv

cbioportal_variants <- cbioportal$Protein.Change
cbioportal_variants2 <- cbioportal_variants[nchar(cbioportal_variants) <= 5]
cbioportal_variants3 <- data.frame(table(cbioportal_variants2))
colnames(cbioportal_variants3) <- c("variant","cbioportal")

cbioportal_variants4 <- merge(cbioportal_variants3, snv3, by = "variant", all.y = T)
cbioportal_variants5 <- merge(cosmic, cbioportal_variants4, by = "variant", all.y = T)
cbioportal_variants6 <- cbioportal_variants5 %>% filter(!is.na(cbioportal) | !is.na(Count))

## Number of variants below syn threshold
cbioportal_variants6_low <- cbioportal_variants6 %>% filter(endo_score < syn_endo_range[1] | sko_score < syn_sko_range[1])
cbioportal_variants6_lowboth <- cbioportal_variants6 %>% filter(endo_score < syn_endo_range[1] & sko_score < syn_sko_range[1])

cbioportal_only <- cbioportal_variants6 %>% filter(!is.na(cbioportal) & is.na(Count))
cosmic_only <- cbioportal_variants6 %>% filter(is.na(cbioportal) & !is.na(Count))
cbioportal_cosmic <- cbioportal_variants6 %>% filter(!is.na(cbioportal) & !is.na(Count))

cBioPortal_cosmic_variants <- ggplot() + 
  # scale_x_log10(limits = c(0.01,3)) + scale_y_log10(limits = c(0.1,5)) +
  theme(panel.grid = element_blank()) +
  geom_point(data = snv3, aes(x = log10(endo_ratio), y = log10(sko_ratio)), alpha = 0.1, size = 1) +
  geom_rect(aes(xmin=syn_endo_range[1], xmax=syn_endo_range[2], ymin=syn_sko_range[1], ymax=syn_sko_range[2]), fill = "pink", alpha = 0.3) + 
  geom_point(data = cbioportal_cosmic, aes(x = log10(endo_ratio), y = log10(sko_ratio)), alpha = 0.6, size = 1.5, fill = "darkgreen", color = "black", shape = 21) +
  geom_point(data = cbioportal_only, aes(x = log10(endo_ratio), y = log10(sko_ratio)), alpha = 0.6, size = 1.5, fill = "green", color = "black", shape = 21) +
  geom_point(data = cosmic_only, aes(x = log10(endo_ratio), y = log10(sko_ratio)), alpha = 0.6, size = 1.5, fill = "yellow4", color = "black", shape = 21) +
  labs(x = "Standard score", y = "Knockout score") +
  geom_text_repel(data = cbioportal_variants6 %>% filter(log10(endo_ratio) < -0.89 & log10(sko_ratio) < -0.23), aes(x = log10(endo_ratio), y = log10(sko_ratio), label = variant, nudge_x = -1.5), size = 2, segment.color = "orange", segment.alpha = 0.5, color = "red") +
  #geom_text(data = cbioportal_variants6 %>% filter(endo_ratio < 0.13 & sko_ratio < 0.6), aes(x = endo_ratio*0.9, y = sko_ratio*0.9, label = variant), size = 2, color = "red") +
  NULL; cBioPortal_cosmic_variants
ggsave(file = "Plots/main/fig8d_cBioPortal_cosmic_variants.pdf", cBioPortal_cosmic_variants, height = 1.4, width = 2)

```
Revision 1 
##Functional assays - CaMPARI calcium influx and ER stress
```{r}
er_campari = read.csv("data/er_stress_campari.csv")

campari_dms_plot = ggplot(data = er_campari %>% distinct(variant, endo_ratio, campari, disease), aes(y= log10(endo_ratio), x= campari, color = disease))+
  geom_point(alpha = 0.5, size = 1.5)+
  theme(legend.position = "none", panel.grid = element_blank())+
  geom_text_repel(aes(label = variant), size =2) + 
  labs(y= "Standard score", x= "CaMPARI2 red:green ratio") + 
  scale_x_continuous(breaks = breaks_pretty(n = 3)) + scale_y_continuous(breaks = breaks_pretty(n = 5)) +
  NULL; campari_dms_plot

ggsave("plots/main/fig9a_campari_dms_plot.pdf", campari_dms_plot, height = 1.5, width = 1.8)

bax_bcl_er = ggplot(data = er_campari, aes(y= log10(endo_ratio), x= 1/mean_bcl2_bax_ratio, color = disease))+
  labs(y= "Standard Score", x= "BAX / BCL-2 mRNA ratio")+ theme(legend.position = "none", panel.grid = element_blank())+
  geom_point(alpha = 0.5, size = 1.5)+
  geom_text_repel(aes(label = variant), size = 2) + 
  scale_x_continuous(breaks = breaks_pretty(n = 3)) + scale_y_continuous(breaks = breaks_pretty(n = 5)) +
  NULL; bax_bcl_er

bip_er = ggplot(data = er_campari %>% filter(Target == "bip"), aes(y= log10(endo_ratio), x= mean_exp, color = disease))+
  labs(y= "Standard Score", x= "BiP mRNA Expression")+ theme(legend.position = "none", panel.grid = element_blank())+
  geom_point(alpha = 0.5, size = 1.5)+
  geom_text_repel(aes(label = variant), size = 2) + 
  scale_x_continuous(breaks = breaks_pretty(n = 3)) + scale_y_continuous(breaks = breaks_pretty(n = 5)) +
  NULL; bip_er

chop_er = ggplot(data = er_campari %>% filter(Target == "chop"), aes(y= log10(endo_ratio), x= mean_exp, color = disease))+
  labs(y= "Standard Score", x= "CHOP mRNA Expression")+ theme(legend.position = "none", panel.grid = element_blank())+
  geom_point(alpha = 0.5, size = 1.5)+
  geom_text_repel(aes(label = variant), size = 2) +
  scale_x_continuous(breaks = c(0.002,0.006)) + scale_y_continuous(breaks = breaks_pretty(n = 5)) +
  NULL; chop_er

ggsave("plots/main/fig9b_bax_bcl_er.pdf", bax_bcl_er, height = 1.5, width = 1.8)
ggsave("plots/main/fig9c_bip_er.pdf", bip_er, height = 1.5, width = 1.8)
ggsave("plots/main/fig9d_chop_er.pdf", chop_er,height = 1.5, width = 1.8)


#Revision 1 - splitting the panels in Fig 9
#ss = small-scale DMS with WT, R429C and R304W
#dms = large-scale DMS with WT, L74P, D84G
er_campari_ss = read.csv("Revision_1/data_r1/er_stress_campari_ss.csv")
er_campari_dms = read.csv("Revision_1/data_r1/er_stress_campari_dms.csv")

campari_ss_plot = ggplot(data = er_campari_ss %>% distinct(variant, endo_ratio, campari, disease), aes(y= log10(endo_ratio), x= campari, color = disease))+
  geom_point(alpha = 0.5, size = 1.5)+
  theme(legend.position = "none", panel.grid = element_blank())+
  geom_text_repel(aes(label = variant), size =2) + 
  labs(y= "Standard score", x= "CaMPARI2 red:green ratio") + 
  scale_x_continuous(breaks = breaks_pretty(n = 3)) + scale_y_continuous(breaks = breaks_pretty(n = 5)) +
  NULL; campari_ss_plot

campari_dms_plot <- 
  ggplot(data = er_campari_dms %>% distinct(variant, endo_ratio, campari, disease), aes(y= log10(endo_ratio), x= campari, color = disease))+
  geom_point(alpha = 0.5, size = 1.5)+
  theme(legend.position = "none", panel.grid = element_blank())+
  geom_text_repel(aes(label = variant), size =2) + 
  labs(y= "Standard score", x= "CaMPARI2 red:green ratio") + 
  scale_x_continuous(breaks = breaks_pretty(n = 3), limits = c(0.06, 0.09)) + scale_y_continuous(breaks = breaks_pretty(n = 5), limits = c(-0.6,-0.2)) +
  NULL; campari_dms_plot

ggsave("Revision_1/plots_r1/main_r1/fig9a_r1_campari_ss_plot.pdf", campari_ss_plot, height = 1.5, width = 1.8)
ggsave("Revision_1/plots_r1/main_r1/fig9a_r1_campari_dms_plot.pdf", campari_dms_plot, height = 1.5, width = 1.8)


campari_scatterplot_set <- rbind(er_campari_ss %>% filter(Target == "chop") %>% select(variant, endo_ratio, campari, disease) %>% mutate(shape = "1"),
                                 er_campari_dms %>% filter(Target == "chop") %>% select(variant, endo_ratio, campari, disease) %>% mutate(shape = "2"))

campari_er_plot <- ggplot() + 
  labs(x = "CaMPARI ratio", y = "Standard score") +
  theme(panel.grid.major = element_blank(), legend.position = "none") +
  geom_point(data = campari_scatterplot_set, aes(y= log10(endo_ratio), x= campari, color = disease, shape = shape)) +
  geom_text_repel(data = campari_scatterplot_set, aes(y= log10(endo_ratio), x= campari, label = variant)) +
  NULL; campari_er_plot
ggsave("Revision_1/plots_r1/main_r1/fig9d_r1_campari_er_plot.pdf", campari_er_plot, height = 2, width = 2.25)


bax_bcl_er_ss = ggplot(data = er_campari_ss, aes(y= log10(endo_ratio), x= mean_bax_bcl2_ratio, color = disease))+
  labs(y= "Standard Score", x= "BAX / BCL-2 mRNA ratio")+ theme(legend.position = "none", panel.grid = element_blank())+
  geom_point(alpha = 0.5, size = 1.5)+
  geom_text_repel(aes(label = variant), size = 2) + 
  scale_x_continuous(breaks = breaks_pretty(n = 3)) + scale_y_continuous(breaks = breaks_pretty(n = 5)) +
  NULL; bax_bcl_er_ss

bax_bcl_er_dms = ggplot(data = er_campari_dms, aes(y= log10(endo_ratio), x= mean_bax_bcl2_ratio, color = disease))+
  labs(y= "Standard Score", x= "BAX / BCL-2 mRNA ratio")+ theme(legend.position = "none", panel.grid = element_blank())+
  geom_point(alpha = 0.5, size = 1.5)+
  geom_text_repel(aes(label = variant), size = 2) + 
  scale_x_continuous(breaks = breaks_pretty(n = 3)) + scale_y_continuous(breaks = breaks_pretty(n = 5)) +
  NULL; bax_bcl_er_dms

bax_bcl_scatterplot_set <- rbind(er_campari_ss %>% filter(Target == "BAX/BCL-2") %>% select(variant, endo_ratio, mean_bax_bcl2_ratio, disease) %>% mutate(shape = "1"),
                                 er_campari_dms %>% filter(Target == "BAX/BCL-2") %>% select(variant, endo_ratio, mean_bax_bcl2_ratio, disease) %>% mutate(shape = "2"))

bax_bcl_er_plot <- ggplot() + 
  labs(x = "Bax:Bcl2 ratio", y = "Standard score") +
  theme(panel.grid.major = element_blank(), legend.position = "none") +
  geom_point(data = bax_bcl_scatterplot_set, aes(y= log10(endo_ratio), x= mean_bax_bcl2_ratio, color = disease, shape = shape)) +
  geom_text_repel(data = bax_bcl_scatterplot_set, aes(y= log10(endo_ratio), x= mean_bax_bcl2_ratio, label = variant)) +
  NULL; bax_bcl_er_plot
ggsave("Revision_1/plots_r1/main_r1/fig9d_r1_bax_bcl_er_plot.pdf", bax_bcl_er_plot, height = 2, width = 2.25)

  
  

bip_er_ss = ggplot(data = er_campari_ss %>% filter(Target == "bip"), aes(y= log10(endo_ratio), x= mean_exp, color = disease))+
  labs(y= "Standard Score", x= "BiP mRNA Expression")+ theme(legend.position = "none", panel.grid = element_blank())+
  geom_point(alpha = 0.5, size = 1.5)+
  geom_text_repel(aes(label = variant), size = 2) + 
  scale_x_continuous(breaks = breaks_pretty(n = 3)) + scale_y_continuous(breaks = breaks_pretty(n = 5)) +
  NULL; bip_er_ss

bip_er_dms = ggplot(data = er_campari_dms %>% filter(Target == "bip"), aes(y= log10(endo_ratio), x= mean_exp, color = disease))+
  labs(y= "Standard Score", x= "BiP mRNA Expression")+ theme(legend.position = "none", panel.grid = element_blank())+
  geom_point(alpha = 0.5, size = 1.5)+
  geom_text_repel(aes(label = variant), size = 2) + 
  scale_x_continuous(breaks = breaks_pretty(n = 3)) + scale_y_continuous(breaks = breaks_pretty(n = 5)) +
  NULL; bip_er_dms

bip_scatterplot_set <- rbind(er_campari_ss %>% filter(Target == "bip") %>% select(variant, endo_ratio, mean_exp, disease) %>% mutate(shape = "1"),
                                 er_campari_dms %>% filter(Target == "bip") %>% select(variant, endo_ratio, mean_exp, disease) %>% mutate(shape = "2"))

bip_er_plot <- ggplot() + 
  labs(x = "Bip Cq", y = "Standard score") +
  theme(panel.grid = element_blank(), legend.position = "none") +
  geom_point(data = bip_scatterplot_set, aes(y= log10(endo_ratio), x= mean_exp, color = disease, shape = shape)) +
  geom_text_repel(data = bip_scatterplot_set, aes(y= log10(endo_ratio), x= mean_exp, label = variant)) +
  NULL; bip_er_plot
ggsave("Revision_1/plots_r1/main_r1/fig9d_r1_bip_er_plot.pdf", bip_er_plot, height = 2, width = 2.25)


chop_er_ss = ggplot(data = er_campari_ss %>% filter(Target == "chop"), aes(y= log10(endo_ratio), x= mean_exp, color = disease))+
  labs(y= "Standard Score", x= "CHOP mRNA Expression")+ theme(legend.position = "none", panel.grid = element_blank())+
  geom_point(alpha = 0.5, size = 1.5)+
  geom_text_repel(aes(label = variant), size = 2) +
  scale_x_continuous(breaks = breaks_pretty(n = 3)) + scale_y_continuous(breaks = breaks_pretty(n = 5)) +
  NULL; chop_er_ss

chop_er_dms = ggplot(data = er_campari_dms %>% filter(Target == "chop"), aes(y= log10(endo_ratio), x= mean_exp, color = disease))+
  labs(y= "Standard Score", x= "CHOP mRNA Expression")+ theme(legend.position = "none", panel.grid = element_blank())+
  geom_point(alpha = 0.5, size = 1.5)+
  geom_text_repel(aes(label = variant), size = 2) +
  scale_x_continuous(breaks = breaks_pretty(n = 3)) + scale_y_continuous(breaks = breaks_pretty(n = 5)) +
  NULL; chop_er_dms

chop_scatterplot_set <- rbind(er_campari_ss %>% filter(Target == "chop") %>% select(variant, endo_ratio, mean_exp, disease) %>% mutate(shape = "1"),
                                 er_campari_dms %>% filter(Target == "chop") %>% select(variant, endo_ratio, mean_exp, disease) %>% mutate(shape = "2"))

chop_er_plot <- ggplot() + 
  labs(x = "CHOP Cq", y = "Standard score") +
  theme(panel.grid.major = element_blank(), legend.position = "none") +
  geom_point(data = chop_scatterplot_set, aes(y= log10(endo_ratio), x= mean_exp, color = disease, shape = shape)) +
  geom_text_repel(data = chop_scatterplot_set, aes(y= log10(endo_ratio), x= mean_exp, label = variant)) +
  NULL; chop_er_plot
ggsave("Revision_1/plots_r1/main_r1/fig9d_r1_chop_er_plot.pdf", chop_er_plot, height = 2, width = 2.25)
ggsave("Revision_1/plots_r1/main_r1/fig9b_r1_bax_bcl_er_ss.pdf", bax_bcl_er_ss, height = 1.5, width = 1.8)
ggsave("Revision_1/plots_r1/main_r1/fig9b_r1_bax_bcl_er_dms.pdf", bax_bcl_er_dms, height = 1.5, width = 1.8)
ggsave("Revision_1/plots_r1/main_r1/fig9c_r1_bip_er_ss.pdf", bip_er_ss, height = 1.5, width = 1.8)
ggsave("Revision_1/plots_r1/main_r1/fig9c_r1_bip_er_dms.pdf", bip_er_dms, height = 1.5, width = 1.8)
ggsave("Revision_1/plots_r1/main_r1/fig9d_r1_chop_er_ss.pdf", chop_er_ss, height = 1.5, width = 1.8)
ggsave("Revision_1/plots_r1/main_r1/fig9d_r1_chop_er_dms.pdf", chop_er_dms, height = 1.5, width = 1.8)

```

Raw data for low-throughput functional assays
```{r}
bip_chop_manuscript0 = read.csv("Revision_1/data_r1/chop_bip_12dec24_edit.csv")
bip_chop_manuscript <- bip_chop_manuscript0 %>% filter(Content != "NTC") %>% group_by(variant, Target, replicate) %>% summarize(Cq = 10^mean(log10(Cq)), delta_cq = 10^mean(log10(delta_cq))) %>% mutate(mean_deltacq = exp(mean(log(delta_cq), na.rm = T)))

bip_chop_manuscript$replicate = as.character(bip_chop_manuscript$replicate)
bip_chop_manuscript1 = bip_chop_manuscript %>% filter(Target != "Actin")
bip_chop_manuscript1[bip_chop_manuscript1$variant == "vif","variant"] <- "Control"
bip_chop_manuscript1$variant = factor(bip_chop_manuscript1$variant, c("Control","WT", "D84G", "L74P", "R429C", "R304W"))
#BiP

bip_chop_manuscript1$wt_norm <- NA
for(x in 1:nrow(bip_chop_manuscript1)){
  temp_rep <- bip_chop_manuscript1$replicate[x]
  temp_target <- bip_chop_manuscript1$Target[x]
  control_val <- (bip_chop_manuscript1 %>% filter(variant == "Control" & replicate == temp_rep & Target == temp_target))$delta_cq
  bip_chop_manuscript1$wt_norm[x] <- bip_chop_manuscript1$delta_cq[x] / control_val
}
bip_chop_manuscript_summary <- bip_chop_manuscript1 %>% group_by(variant, Target) %>% summarize(mean_wt_norm = exp(mean(log(wt_norm), na.rm = T)))

bip_manu_plot = ggplot()+ 
  theme(legend.position = "bottom")+ 
  labs(x= NULL, y= "Relative\nBip\nexpression")+
  geom_hline(yintercept = 1, alpha = 0.4) + 
  geom_quasirandom(data = bip_chop_manuscript1 %>% filter(Target == "bip" & variant != "Control"), aes(x= variant, y= wt_norm, color = replicate), alpha = 0.5) +
  scale_y_log10() + 
  geom_point(data = bip_chop_manuscript_summary %>% filter(Target == "bip" & variant != "Control"), aes(x= variant, y= mean_wt_norm), color = "Black", shape = 95, size = 5) +
  theme(panel.grid.major = element_blank(), axis.text.x = element_text(angle = -90, hjust = 0.5, vjust = 0.5)) + 
  #stat_summary(fun.data="mean_cl_boot", geom="errorbar",width=0.05, colour="black", alpha = 0.25)+NULL; bip_manu_plot
  NULL; bip_manu_plot
ggsave("Revision_1/plots_r1/supplement_r1/bip_manu_plot.pdf", bip_manu_plot, height = 1.75, width = 1.75)
  # geom_errorbar(data = bip_chop_manuscript %>% filter(Target == "bip"), aes(ymin= mean_deltacq-sd, ymax= mean_deltacq+sd), width=.05, alpha = 0.05)

#CHOP
chop_manu_plot = ggplot(data = bip_chop_manuscript1 %>% filter(Target == "chop" & variant != "Control"), aes(x= variant, y= wt_norm, color = replicate))+ 
  theme(legend.position = "bottom")+ labs(x= NULL, y= "Relative\nChop\nexpression")+
  geom_quasirandom(alpha = 0.5) +
  scale_y_log10() + 
  geom_hline(yintercept = 1, alpha = 0.4) + 
  geom_point(data = bip_chop_manuscript_summary %>% filter(Target == "chop" & variant != "Control"), aes(x= variant, y= mean_wt_norm), color = "black", shape = 95, size = 5)+
  theme(panel.grid.major = element_blank(), axis.text.x = element_text(angle = -90, hjust = 0.5, vjust = 0.5)) + 
  #stat_summary(fun.data="mean_cl_boot", geom="errorbar",width=0.05, colour="black", alpha = 0.25)+
  NULL; chop_manu_plot
ggsave("Revision_1/plots_r1/supplement_r1/chop_manu_plot.pdf", chop_manu_plot, height = 1.75, width = 1.75)
  
#BAX/BCL-2
bax_bcl2_manuscript0 = read.csv("Revision_1/data_r1/bcl2_bax_12dec24.csv")
bax_bcl2_manuscript <- bax_bcl2_manuscript0 %>% filter(Content != "NTC") %>% group_by(variant, Target, replicate) %>% summarize(Cq = 10^mean(log10(Cq)), delta_cq = 10^mean(log10(delta_cq))) %>% mutate(mean_deltacq = exp(mean(log(delta_cq), na.rm = T)))
bax_bcl2_manuscript = bax_bcl2_manuscript %>% filter(Target != "Actin")
bax_bcl2_manuscript[bax_bcl2_manuscript$variant == "vif","variant"] <- "Control"

bax_bcl2_ratio <- bax_bcl2_manuscript %>% filter(!is.na(replicate)) %>% group_by(variant, replicate) %>% select(variant, replicate) %>% distinct()

bax_bcl2_ratio$ratio <- NA
for(x in 1:nrow(bax_bcl2_ratio)){
  temp_rep <- bax_bcl2_ratio$replicate[x]
  temp_variant <- bax_bcl2_ratio$variant[x]
  bax_bcl2_ratio$ratio[x] <- (bax_bcl2_manuscript %>% filter(variant == temp_variant, replicate == temp_rep, Target == "bax"))$delta_cq / (bax_bcl2_manuscript %>% filter(variant == temp_variant, replicate == temp_rep, Target == "bcl2"))$delta_cq
}

bax_bcl2_ratio$wt_norm <- NA
for(x in 1:nrow(bax_bcl2_ratio)){
  temp_rep <- as.numeric(bax_bcl2_ratio$replicate[x])
  control_val <- (bax_bcl2_ratio %>% filter(variant == "Control" & replicate == temp_rep))$ratio
  bax_bcl2_ratio$wt_norm[x] <- bax_bcl2_ratio$ratio[x] / control_val
}
bax_bcl2_ratio_summary <- bax_bcl2_ratio %>% group_by(variant) %>% summarize(mean_wt_norm = exp(mean(log(wt_norm), na.rm = T)))

#bax_bcl2_ratio = bax_bcl2_manuscript %>% select(variant, ratio, replicate)
bax_bcl2_ratio$replicate = as.character(bax_bcl2_ratio$replicate)
bax_bcl2_ratio = bax_bcl2_ratio %>% filter(variant %in% c("Control","WT", "D84G", "L74P", "R429C", "R304W")) %>% group_by(variant) %>% mutate(mean = exp(mean(log(ratio), na.rm = T)))
bax_bcl2_ratio$variant = factor(bax_bcl2_ratio$variant, c("Control","WT", "D84G", "L74P", "R429C", "R304W"))

bax_bcl2_manu_plot = ggplot() +
  geom_hline(yintercept = 1, alpha = 0.4) + 
  geom_quasirandom(data = bax_bcl2_ratio %>% filter(variant != "Control"), aes(x= variant, y= wt_norm, color = replicate), alpha = 0.5) + 
  theme(legend.position = "bottom")+ labs(x= NULL, y= "Relative\nBAX/BCL2\nratio")+
  scale_y_log10() + 
  geom_point(data = bax_bcl2_ratio_summary %>% filter(variant != "Control"), aes(x= variant, y= mean_wt_norm), color = "black", shape = 95, size = 5) +
  theme(panel.grid.major = element_blank(), axis.text.x = element_text(angle = -90, hjust = 0.5, vjust = 0.5)) + 
  #stat_summary(fun.data="mean_cl_boot", geom="errorbar",width=0.05, colour="black", alpha = 0.25)+
  NULL; bax_bcl2_manu_plot
ggsave("Revision_1/plots_r1/supplement_r1/bax_bcl2_manu_plot.pdf", bax_bcl2_manu_plot, height = 1.75, width = 1.85)
```

```{r}
campari0 = read.csv("Revision_1/data_r1/campari_data2.csv")

campari <- campari0 %>% group_by(variant, replicate) %>% summarize(green_mfi = exp(mean(log(green_mfi))), red_mfi = exp(mean(log(red_mfi))), red_mfi = exp(mean(log(red_mfi))), campari = exp(mean(log(campari))), endo_ratio = exp(mean(log(endo_ratio)))) %>% mutate(redgrnratio = red_mfi / green_mfi) 

campari$wt_norm <- NA
for(x in 1:nrow(campari)){
  temp_rep <- as.numeric(campari$replicate[x])
  control_val <- (campari %>% filter(variant == "pep" & replicate == temp_rep))$redgrnratio
  campari$wt_norm[x] <- campari$redgrnratio[x] / control_val
}
campari_summary <- campari %>% group_by(variant) %>% summarize(mean_wt_norm = exp(mean(log(wt_norm), na.rm = T)))

campari$replicate = as.character(campari$replicate)

campari_plot_set <- campari %>% filter(variant %in% c("WT", "R429C", "R304W", "L74P", "D84G"))
campari_plot_set$variant = factor(campari_plot_set$variant, c("WT", "D84G", "L74P", "R429C", "R304W"))
campari_plot_set2 <- campari_summary %>% filter(variant %in% c("WT", "R429C", "R304W", "L74P", "D84G"))
campari_plot_set2$variant = factor(campari_plot_set2$variant, c("WT", "D84G", "L74P", "R429C", "R304W"))

campari_manu_plot <- ggplot() + 
  geom_hline(yintercept = 1, alpha = 0.4) + 
  theme(legend.position = "bottom", panel.grid = element_blank(), axis.text.x = element_text(angle = -90, hjust = 0.5, vjust = 0.5))+ 
  labs(x= NULL, y= "Relative\nCaMPARI\nred:green")+
  scale_y_log10() +
  geom_quasirandom(data = campari_plot_set, aes(x= variant, y = wt_norm, color = replicate), alpha = 0.5) + 
  geom_point(data = campari_plot_set2, aes(x= variant, y= mean_wt_norm), shape = 45, size = 5)+
  #stat_summary(fun.data="mean_cl_boot", geom="errorbar",width=0.05, colour="black", alpha = 0.25)+NULL; campari_manu_plot
  NULL; campari_manu_plot

ggsave("Revision_1/plots_r1/supplement_r1/campari_manu_plot.pdf", campari_manu_plot, height = 1.75, width = 1.85)

```

##FoldX vs survival scores - Supplement fig. 10a-b
```{r}
foldx = read.csv("Revision_1/data_r1/foldx.csv")
foldx_ef = left_join(combined2, foldx)
annots[annots$value == "0.98","annot"] <- "1_alpha1"
annots[annots$value == "0.99","annot"] <- "2_EFhand"
annots[annots$value == "1","annot"] <- "3_alpha2"

foldx_kam <- foldx
foldx_kam$position <- gsub("[^0-9]","",foldx_kam$variant)
foldx_kam_pos <- merge(foldx_kam %>% group_by(position) %>% summarize(pos_tot_energy = mean(total.energy)), endo_positional_ratios3, by = "position")
foldx_kam_pos2 <- merge(foldx_kam_pos, annots[,c("position","annot")], by = "position")


Positional_foldx_plot <- ggplot() + theme(panel.grid = element_blank(), legend.position = "none") + 
  scale_x_log10() + 
  geom_point(data = foldx_kam_pos2, aes(x = missense_geomean, y = pos_tot_energy, color = annot), alpha = 0.5) +
  geom_text_repel(data = foldx_kam_pos2, aes(x = missense_geomean, y = pos_tot_energy, label = position), color = "red", alpha = 0.75, size = 2) +
  labs(x = "Standard score", y = "FoldX total energy") +
  NULL; Positional_foldx_plot
ggsave(file = "Revision_1/plots_r1/supplement_r1/Positional_foldx_plot.pdf", Positional_foldx_plot, height = 2.25, width = 3.25)

foldx_ef_kam <- foldx_ef

Variant_foldx_plot <- ggplot() + theme(panel.grid = element_blank(), legend.position = "none") + 
  #scale_x_log10() + 
  #scale_y_log10(limits = c(1,100)) +
  geom_hline(yintercept = 0) +
  geom_point(data = foldx_ef_kam, aes(x = log10(endo_ratio), y = total.energy), alpha = 0.25) +
  geom_point(data = foldx_ef_kam %>% filter(position %in% c(91,92,93,94)), aes(x = log10(endo_ratio), y = total.energy), color = "purple", alpha = 0.5) +
  geom_point(data = foldx_ef_kam %>% filter(position %in% c(76,78)), aes(x = log10(endo_ratio), y = total.energy), color = "darkgreen", alpha = 0.5) +
  geom_point(data = foldx_ef_kam %>% filter(position == 70), aes(x = log10(endo_ratio), y = total.energy), color = "pink", alpha = 0.5) +
  labs(x = "Standard score", y = "FoldX total energy") +
  NULL; Variant_foldx_plot
ggsave(file = "Revision_1/plots_r1/supplement_r1/supp10_Variant_foldx_plot.pdf", Variant_foldx_plot, height = 2.25, width = 3.25)

```
##AlphaMissense vs survival scores - Supplement fig 10c-d
```{r}
am_ef = read.csv("Revision_1/data_r1/alpha_missense_ef.csv")
am_ef = am_ef[-1]
am_ef = am_ef %>% select(protein_variant, am_pathogenicity, am_class)
colnames(am_ef)[1] = "variant"
am_ef = left_join(combined2, am_ef)
am_path_var = am_ef %>% filter(variant %in% c("H72Q", "L74P", "G81D", "D84G", "L92V", "L96V"))

kam_am_ef <- am_ef

Alpha_missense_all <- ggplot() + geom_point(data = kam_am_ef %>% filter(!(is.na(am_class))), aes(x = endo_ratio, y = am_pathogenicity, color = am_class), alpha = 0.4) +
  scale_x_log10() + scale_y_log10() + 
  labs(x = "Standard score", y = "AlphaMissense score", color = "Patients and general population", shape = "AlphaMissense prediction") +
  theme(panel.grid = element_blank(), legend.position = "right") + 
  NULL; Alpha_missense_all
ggsave("Revision_1/plots_r1/supplement_r1/Alpha_missense_all.pdf", Alpha_missense_all, height = 2.5, width = 5.25)

gnomad4 <- merge(gnomad3, kam_am_ef[,c("variant","am_pathogenicity", "am_class")], by = "variant") %>% mutate(ground_truth = "healthy")
clinvar3 <- merge(clinvar2, kam_am_ef[,c("variant","am_pathogenicity", "am_class")], by = "variant") %>% filter(Disease1 == "Myopathy") %>% mutate(ground_truth = "myopathy")
  
kam_am_comparison <- rbind(gnomad4[,c("variant","am_pathogenicity", "am_class","ground_truth","endo_ratio","sko_ratio")], clinvar3[,c("variant","am_pathogenicity", "am_class","ground_truth", "endo_ratio","sko_ratio")])

am_class_counts <- data.frame(table(kam_am_ef$am_class))
am_class_counts$freq2 <- am_class_counts$Freq / sum(am_class_counts$Freq)

table(kam_am_comparison[,c("ground_truth","am_class")])

Alpha_missense_controls <- ggplot() + geom_point(data = kam_am_comparison, aes(x = endo_ratio, y = am_pathogenicity, color = ground_truth, shape = am_class)) +
  scale_x_log10() + scale_y_log10() + 
  labs(x = "Standard score", y = "AlphaMissense score", color = "Patients and general population", shape = "AlphaMissense prediction") +
  theme(panel.grid = element_blank(), legend.position = "right") + 
  NULL; Alpha_missense_controls
ggsave("Revision_1/plots_r1/supplement_r1/supp10_Alpha_missense_controls.pdf", Alpha_missense_controls, height = 2.5, width = 5.25)

```

##ESM
```{r}
esm_scores = read.csv("Revision_1/data_r1/esm_scores_stim1.csv") %>% filter(pos %in% c(63:98))
colnames(esm_scores)[2] = "esm_score"
esm_scores = left_join(combined2, esm_scores)

esm_scores_plot = ggplot(data = esm_scores %>% filter(esm_score != 0), aes(x= log10(endo_ratio), y= esm_score))+
  geom_point(alpha = 0.15, size = 1)+
  labs(x= "Standard LP score", y= "ESM scores")+
  geom_point(data = esm_scores %>% filter(variant %in% c("L74P", "D84G", "L96V", "L92V", "H72Q", "G81D", "D84E", "N80T", "D84N", "D84A")), aes(x= log10(endo_ratio), y= esm_score), color = "red", alpha = 0.5, size = 1)+NULL; esm_scores_plot
  # stat_smooth (geom="line", alpha=0.3, size=1, span=0.5, method = lm) + stat_poly_eq(use_label(c("R2")))
ggsave("Revision_1/plots_r1/supplement_r1/esm_scores_plot.pdf", esm_scores_plot, height = 2, width = 2)

esm_scores_sko_plot = ggplot(data = esm_scores %>% filter(esm_score != 0), aes(x= log10(sko_ratio), y= esm_score))+
  geom_point(alpha = 0.15, size = 1)+
  labs(x= "Knockout LP score", y= "ESM scores")+
  geom_point(data = esm_scores %>% filter(variant %in% c("L74P", "D84G", "L96V", "L92V", "H72Q", "G81D", "D84E", "N80T", "D84N", "D84A")), aes(x= log10(sko_ratio), y= esm_score), color = "red", alpha = 0.5, size = 1)+NULL; esm_scores_sko_plot
ggsave("Revision_1/plots_r1/supplement_r1/esm_scores_sko_plot.pdf", esm_scores_sko_plot, height = 2, width = 2)


ggplot(data = esm_scores %>% filter(variant %in% c("L74P", "D84G", "L96V", "L92V", "H72Q", "G81D", "D84E", "N80T", "D84N", "D84A")), aes(x= log10(endo_ratio), y= esm_score)) +
  geom_point(alpha = 0.5) +
  geom_text_repel(aes(label = variant))

#spearman correlation for ESM scores vs survival scores
esm_scores = esm_scores %>% mutate(std_score = log10(endo_ratio), knockout_score = log10(sko_ratio))
esm_scores_path = esm_scores %>% filter(variant %in% c("L74P", "D84G", "L96V", "L92V", "H72Q", "G81D", "D84E", "N80T", "D84N", "D84A"))
cor.test(esm_scores$esm_score, esm_scores$std_score, method = "spearman", data = esm_scores)
cor.test(esm_scores_path$esm_score, esm_scores_path$std_score, method = "spearman", data = esm_scores_path)
```

##Error bars for replicate scores - Supplement fig. 9
```{r}
eb_endo = endo_freq %>% select(variant, r1_ratio, r2_ratio, r4_ratio, r5_ratio, r6_ratio)
colnames(eb_endo) = c("variant", "er1", "er2", "er4" ,"er5", "er6")
for(x in 1:nrow(eb_endo)){
 eb_endo$mean[x] <- mean(unlist(eb_endo[x,c(2,3,4,5,6)]), na.rm = T)
 eb_endo$count[x] <- sum(!is.na(unlist(eb_endo[x,c(2,3,4,5,6)])))
 eb_endo$endo_se[x] <- sd(unlist(eb_endo[x,c(2,3,4,5,6)]), na.rm = T) / sqrt(eb_endo$count[x])
}
clinvar_errorbar_endo = eb_endo %>% filter(variant %in% c("L92V", "L96V", "N80T", "H72Q", "D84G", "G81D", "D84A", "D84E", "D84N", "S88G", "Y98C")) %>% select(variant, er1, er2, er4, er5, er6) %>% melt(id = 'variant')
vus_errorbar_endo = eb_endo %>% filter(variant %in% c("F91Y", "D76G", "G81C", "E66K", "E90D", "R69C", "L74P", "N80S", "A67T", "N80D", "R69H", "K73R")) %>% select(variant, er1, er2, er4, er5, er6) %>% melt(id = "variant")
evus_syn = eb_endo %>% mutate(start = substr(variant, 1,1),
                             end = substr(variant, 4,4),
                             position = substr(variant, 2,3))
evus_syn = evus_syn %>% mutate(type = case_when(start == end ~ "syn",
                                              end == "_" ~ "non",
                                              start != end & end != "_" ~ "mis"))
colnames(evus_syn)[1] = "mut"
evus_syn = evus_syn %>% filter(type == "syn") %>% mutate(variant = "syn")
evus_syn = evus_syn %>% select(variant, er1, er2, er4, er5, er6) %>% melt(id = 'variant')
# evus_syn_all = syn_scores %>% select(variant, endo_ratio)
# evus_syn_all = data.frame(evus_syn_all)
# evus_syn_all = evus_syn_all %>% select(variant, endo_ratio)
# colnames(evus_syn_all)[2] = "value"
# clinvar_errorbar_endo1 = bind_rows(clinvar_errorbar_endo, evus_syn_all)
clinvar_errorbar_endo = bind_rows(clinvar_errorbar_endo, evus_syn)
vus_errorbar_endo = bind_rows(vus_errorbar_endo, evus_syn)

eb_sko = sko_freq %>% select(variant, r1_ratio, r2_ratio, r3_ratio, r4_ratio, r5_ratio, r6_ratio)
colnames(eb_sko) = c("variant", "sr1", "sr2", "sr3", "sr4", "sr5", "sr6")
for(x in 1:nrow(eb_sko)){
 eb_sko$mean[x] <- mean(unlist(eb_sko[x,c(2,3,4,5,6)]), na.rm = T)
 eb_sko$count[x] <- sum(!is.na(unlist(eb_sko[x,c(2,3,4,5,6)])))
 eb_sko$sko_se[x] <- sd(unlist(eb_sko[x,c(2,3,4,5,6)]), na.rm = T) / sqrt(eb_sko$count[x])
}
clinvar_errorbar_sko = eb_sko %>% filter(variant %in% c("L92V", "L96V", "N80T", "H72Q", "D84G", "G81D", "D84A", "D84E", "D84N", "S88G", "Y98C")) %>% select(variant, sr1, sr2, sr3, sr4, sr5, sr6) %>% melt(id = 'variant')
vus_errorbar_sko = eb_sko %>% filter(variant %in% c("F91Y", "D76G", "G81C", "E66K", "E90D", "R69C", "L74P", "N80S", "A67T", "N80D", "R69H", "K73R")) %>% select(variant, sr1, sr2, sr3, sr4, sr5, sr6) %>% melt(id = "variant")
svus_syn = eb_sko %>% mutate(start = substr(variant, 1,1),
                             end = substr(variant, 4,4),
                             position = substr(variant, 2,3))
svus_syn = svus_syn %>% mutate(type = case_when(start == end ~ "syn",
                                              end == "_" ~ "non",
                                              start != end & end != "_" ~ "mis"))
colnames(svus_syn)[1] = "mut"
svus_syn = svus_syn %>% filter(type == "syn") %>% mutate(variant = "syn")
svus_syn = svus_syn %>% select(variant, sr1, sr2, sr3, sr4, sr5, sr6) %>% melt(id = 'variant')
# svus_syn_all = syn_scores %>% select(variant, sko_ratio)
# svus_syn_all = data.frame(svus_syn_all)
# svus_syn_all = svus_syn_all %>% select(variant, sko_ratio)
# colnames(svus_syn_all)[2] = "value"
# clinvar_errorbar_sko1 = bind_rows(clinvar_errorbar_sko, svus_syn_all)
clinvar_errorbar_sko = bind_rows(clinvar_errorbar_sko, svus_syn)
vus_errorbar_sko = bind_rows(vus_errorbar_sko, svus_syn)

combined3 = combined2 %>% select(variant, endo_ratio, sko_ratio)
combined3 = left_join(combined3, eb_endo)
combined3 = left_join(combined3, eb_sko)

write.csv(eb_endo, "eb_endo.csv")
write.csv(eb_sko, "eb_sko.csv")

clinvar_variants = clinvar2 %>% select(variant, Disease1, Disease2, ClinVar)
clinvar_errorbar = left_join(clinvar_variants, combined3)

clinvar_errorbar_endo$log10_value <- log10(clinvar_errorbar_endo$value)
clinvar_errorbar_endo <- rbind(clinvar_errorbar_endo, c("",NA, NA, NA))
clinvar_errorbar_endo$variant = factor(clinvar_errorbar_endo$variant, c("syn", "", "H72Q","N80T","G81D","D84A", "D84E", "D84G", "D84N" ,"S88G", "L92V", "L96V", "Y98C"))
clinvar_errorbar_sko$log10_value <- log10(clinvar_errorbar_sko$value)
clinvar_errorbar_sko <- rbind(clinvar_errorbar_sko, c("",NA, NA, NA))
clinvar_errorbar_sko$variant = factor(clinvar_errorbar_sko$variant, c("syn", "", "H72Q","N80T","G81D","D84A", "D84E", "D84G", "D84N" ,"S88G", "L92V", "L96V", "Y98C"))

clinvar_eb_endo_plot <- ggplot()+ 
  geom_quasirandom(data = clinvar_errorbar_endo %>% filter(variant == "syn"), aes(x= variant, y= as.numeric(log10_value), color = variable), alpha = 0.5, size = 0.25, width = 1.25) + 
  geom_quasirandom(data = clinvar_errorbar_endo %>% filter(variant != "syn"), aes(x= variant, y= as.numeric(log10_value), color = variable), alpha = 0.5, size = 0.75, width = 0.2) + 
  geom_hline(yintercept = c(syn_endo_range[1],syn_endo_range[2]), linetype = "dotted")+
  geom_point(data = clinvar_errorbar %>% filter(variant %in% c("L92V", "L96V", "N80T", "H72Q", "D84G", "G81D", "D84A", "D84E", "D84N", "S88G", "Y98C")),
             aes(x= variant, y= log10(endo_ratio)), color = "black", shape = 45, size = 6)+
  theme(panel.grid = element_blank(), axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1), legend.position = "none")+
  scale_y_continuous(limits = c(-2.2,1.1)) +
  #stat_summary(fun.data="mean_cl_boot", geom="errorbar",width=0.05, colour="black", alpha = 0.25)+
  labs(x= NULL, y= "Standard Score", color = "Replicate")+
  NULL; clinvar_eb_endo_plot
ggsave("Revision_1/plots_r1/supplement_r1/supp9_clinvar_eb_endo_plot.pdf", clinvar_eb_endo_plot ,height = 1.5, width = 3.3)

clinvar_eb_sko_plot <- ggplot()+ 
  geom_quasirandom(data = clinvar_errorbar_sko %>% filter(variant == "syn"), aes(x= variant, y= as.numeric(log10_value), color = variable), alpha = 0.5, size = 0.25, width = 1.25) + 
  geom_quasirandom(data = clinvar_errorbar_sko %>% filter(variant != "syn"), aes(x= variant, y= as.numeric(log10_value), color = variable), alpha = 0.5, size = 0.75, width = 0.2) + 
  geom_hline(yintercept = c(syn_sko_range[1],syn_sko_range[2]), linetype = "dotted")+
  geom_point(data = clinvar_errorbar %>% filter(variant %in% c("L92V", "L96V", "N80T", "H72Q", "D84G", "G81D", "D84A", "D84E", "D84N", "S88G", "Y98C")),
             aes(x= variant, y= log10(sko_ratio)), color = "black", shape = 45, size = 6)+
  theme(panel.grid = element_blank(), axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1), legend.position = "none")+
  #stat_summary(fun.data="mean_cl_boot", geom="errorbar",width=0.05, colour="black", alpha = 0.25)+
  labs(x= NULL, y= "Knockout Score", color = "Replicate")+
  scale_y_continuous(limits = c(-1,0.6)) +
  NULL; clinvar_eb_sko_plot
ggsave("Revision_1/plots_r1/supplement_r1/supp9_clinvar_eb_sko_plot.pdf", clinvar_eb_sko_plot ,height = 1.5, width = 3.3)


vus_errorbar_endo$log10_value <- log10(vus_errorbar_endo$value)
vus_errorbar_endo <- rbind(vus_errorbar_endo, c("",NA, NA, NA))
vus_errorbar_endo$variant = factor(vus_errorbar_endo$variant, c("syn", "","E66K", "A67T", "R69C", "R69H", "K73R", "L74P", "D76G", "N80D","N80S", "G81C", "E90D", "F91Y"))
vus_errorbar_sko$log10_value <- log10(vus_errorbar_sko$value)
vus_errorbar_sko <- rbind(vus_errorbar_sko, c("",NA, NA, NA))
vus_errorbar_sko$variant = factor(vus_errorbar_sko$variant, c("syn", "", "E66K", "A67T", "R69C", "R69H", "K73R", "L74P", "D76G", "N80D","N80S", "G81C", "E90D", "F91Y"))

vus_eb_endo_plot <- ggplot()+ 
  geom_quasirandom(data = vus_errorbar_endo %>% filter(variant == "syn"), aes(x= variant, y= as.numeric(log10_value), color = variable), alpha = 0.5, size = 0.25, width = 1.25) + 
  geom_quasirandom(data = vus_errorbar_endo %>% filter(variant != "syn"), aes(x= variant, y= as.numeric(log10_value), color = variable), alpha = 0.5, size = 0.75, width = 0.2) +
  geom_hline(yintercept = c(syn_endo_range[1],syn_endo_range[2]), linetype = "dotted")+
  geom_point(data = clinvar_errorbar %>% filter(variant %in% c("F91Y", "D76G", "G81C", "E66K", "E90D", "R69C", "L74P", "N80S", "A67T", "N80D", "R69H", "K73R")), aes(x= variant, y= log10(endo_ratio)), color = "black", shape = 45, size = 6)+
  theme(panel.grid = element_blank(), axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1), legend.position = "none")+
  scale_y_continuous(limits = c(-2.2,1.1)) +
  #stat_summary(fun.data="mean_cl_boot", geom="errorbar",width=0.05, colour="black", alpha = 0.25)+
  labs(x= NULL, y= "Standard Score", color = "Replicate")+
  NULL; vus_eb_endo_plot
ggsave("Revision_1/plots_r1/supplement_r1/supp9_vus_eb_endo_plot.pdf", vus_eb_endo_plot ,height = 1.5, width = 3.3)

vus_eb_sko_plot = ggplot() +
  geom_quasirandom(data = vus_errorbar_sko %>% filter(variant == "syn"), aes(x= variant, y= as.numeric(log10_value), color = variable), alpha = 0.5, size = 0.25, width = 1.25) + 
  geom_quasirandom(data = vus_errorbar_sko %>% filter(variant != "syn"), aes(x= variant, y= as.numeric(log10_value), color = variable), alpha = 0.5, size = 0.75, width = 0.2) +
  geom_hline(yintercept = c(syn_sko_range[1], syn_sko_range[2]), linetype = "dotted")+
  geom_point(data = clinvar_errorbar %>% filter(variant %in% c("F91Y", "D76G", "G81C", "E66K", "E90D", "R69C", "L74P", "N80S", "A67T", "N80D", "R69H", "K73R")),
             aes(x= variant, y= log10(sko_ratio)), color = "black", shape = 45, size = 6)+ 
  theme(panel.grid = element_blank(), axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1), legend.position = "none")+
  #stat_summary(fun.data="mean_cl_boot", geom="errorbar",width=0.05, colour="black", alpha = 0.25)+
  labs(x= NULL, y= "Knockout Score", color = "Replicate")+
  scale_y_continuous(limits = c(-1,0.6)) +
  NULL; vus_eb_sko_plot
ggsave("Revision_1/plots_r1/supplement_r1/supp9_vus_eb_sko_plot.pdf", vus_eb_sko_plot ,height = 1.5, width = 3.3)

```

##STIM1 abundance - Supplement Fig 7b
```{r}
revision_mgl = read.csv("Revision_1/data_r1/mGL_mfi.csv")
revision_mgl = revision_mgl %>% group_by(variant, rep) %>% mutate(mean_norm = exp(mean(log(norm_wt))))
revision_mgl = revision_mgl %>% group_by(variant) %>% mutate(mean = exp(mean(log(mean_norm))))
revision_mgl$variant = factor(revision_mgl$variant, c("WT", "F91D", "L92K", "R93W", "E94P"))

mgl_norm_f91d = revision_mgl %>% filter(variant == "F91D") %>% distinct(variant, norm_wt)
mgl_norm_l92k = revision_mgl %>% filter(variant == "L92K") %>% distinct(variant, norm_wt)
mgl_norm_r93w = revision_mgl %>% filter(variant == "R93W") %>% distinct(variant, norm_wt)
mgl_norm_e94p = revision_mgl %>% filter(variant == "E94P") %>% distinct(variant, norm_wt)
mgl_norm_wt = revision_mgl %>% filter(variant == "WT") %>% distinct(variant, norm_wt)


mgl_abundance_plot = ggplot(data = revision_mgl, aes(x= variant, y= norm_wt))+ theme_bw()+
  geom_quasirandom(alpha = 0.5) + scale_y_continuous(limits = c(0,2.5)) +
  geom_point(data = revision_mgl, aes(x= variant, y= mean), color = "red") +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))+
  labs(x= "Variant", y= "Normalized \nGreen MFI")+NULL; mgl_abundance_plot

ggsave("Revision_1/plots_r1/supplement_r1/supp_7b_mgl_abundance_plot.pdf", mgl_abundance_plot, height = 2, width = 2)

t.test(mgl_norm_f91d$norm_wt, mgl_norm_wt$norm_wt)
t.test(mgl_norm_l92k$norm_wt, mgl_norm_wt$norm_wt)
t.test(mgl_norm_r93w$norm_wt, mgl_norm_wt$norm_wt)
t.test(mgl_norm_e94p$norm_wt, mgl_norm_wt$norm_wt)

revision_mgl_dms = revision_mgl %>% select(variant, mean)
revision_mgl_dms = left_join(revision_mgl, combined2)
```

##Rosace - Full script can be found in Github repo as rosace_9apr25.rmd
score tables read in here 
```{r}
rosace_std = read.csv("Revision_1/rosace/rosace_standard_score.csv")
rosace_ko = read.csv("Revision_1/rosace/rosace_knockout_score.csv")

rosace_std = rosace_std %>% select(variants, wildtype, mutation, type, std_score)
rosace_ko = rosace_ko %>% select(variants, wildtype, mutation, type, ko_score)

rosace_scores = left_join(rosace_std, rosace_ko)
colnames(rosace_scores)[1] = "variant"


combined4 = left_join(combined3, rosace_scores)

dms_rosace_endo_plot = ggplot(data = combined4, aes(x= log10(endo_ratio), y= std_score))+ theme_bw()+
  geom_point(alpha = 0.25)+
  labs(x= "Standard score (DMS)", y= "Standard score \n(ROSACE)")+NULL; dms_rosace_endo_plot

dms_rosace_sko_plot = ggplot(data = combined4, aes(x= log10(sko_ratio), y= ko_score))+ theme_bw()+
  geom_point(alpha = 0.25)+
  labs(x= "Knockout score (DMS)", y= "Knockout score \n(ROSACE)")+NULL; dms_rosace_sko_plot

ggsave("Revision_1/plots_r1/supplement_r1/supp10_dms_rosace_endo_plot.pdf", dms_rosace_endo_plot, height = 2, width = 2)
ggsave("Revision_1/plots_r1/supplement_r1/supp10_dms_rosace_sko_plot.pdf", dms_rosace_sko_plot, height = 2, width = 2)

cor.test(combined4$std_score, log10(combined4$endo_ratio), method = "spearman", data = scores)
cor.test(combined4$ko_score, log10(combined4$sko_ratio), method = "spearman", data = scores)

```

##Western Blot band intensities - Supplement Fig 1d
```{r}
wb_intensity = read.csv("Revision_1/data_r1/wb_intensity.csv")
wb_intensity$rep = as.character(wb_intensity$rep)
wb_intensity = wb_intensity %>% group_by(sample, band) %>% mutate(mean = exp(mean(log(norm_int), na.rm = T)))
wb_intensity$sample = factor(wb_intensity$sample, c("Standard", "Knockout"))

wb_intensity_plot = ggplot(data = wb_intensity, aes(x= sample, y= norm_int, color = rep)) +
  geom_quasirandom(alpha = 0.5, dodge.width = 0.5) + 
  geom_point(data = wb_intensity, aes(x= sample, y= mean), color = "black", shape = 45, size = 5)+
  theme(legend.position = "bottom", axis.title.x=element_blank(), panel.grid = element_blank(), axis.text.x = element_text(angle = -90, hjust = 0, vjust = 0.5)) +
  labs(y= "Band intensity \nnormalized to actin", color = "Replicate") + 
  NULL; wb_intensity_plot

ggsave("Revision_1/plots_r1/supplement_r1/supp_1d_wb_intensity_plot.pdf", wb_intensity_plot, height = 2, width = 1.5)
```

##Cell counts - Supplement Fig 1a
```{r}
cell_ct_rev = read.csv("Revision_1/data_r1/cell_ct_rev.csv")
cell_ct_rev2 <- cell_ct_rev %>% filter(rep != 1) %>% mutate(days = hrs / 24)
cell_ct_rev2$rep <- as.factor(cell_ct_rev2$rep)
cell_ct_rev2$log2_slp_ct <- log(cell_ct_rev2$slp_ct,2)
cell_ct_rev2$log2_sko_ct <- log(cell_ct_rev2$sko_ct,2)

ggplot() + 
  geom_line(data = cell_ct_rev2, aes(x = days, y = log2_sko_ct, color = rep)) +
  geom_point(data = cell_ct_rev2, aes(x = days, y = log2_sko_ct, color = rep)) +
  geom_line(data = cell_ct_rev2, aes(x = days, y = log2_slp_ct, color = rep), linetype = 2) +
  geom_point(data = cell_ct_rev2, aes(x = days, y = log2_slp_ct, color = rep), shape = 3)

# https://stackoverflow.com/questions/31851936/exponential-curve-fitting-in-r

log2_sko_lm <- lm(cell_ct_rev2$log2_sko_ct ~ cell_ct_rev2$days)
2^log2_sko_lm$coefficients[1]
2^log2_sko_lm$coefficients[2]

log2_slp_lm <- lm(cell_ct_rev2$log2_slp_ct ~ cell_ct_rev2$days)
2^log2_slp_lm$coefficients[1]
2^log2_slp_lm$coefficients[2]

ggplot() + 
  geom_line(data = cell_ct_rev2, aes(x = days, y = log2_sko_ct, color = rep)) +
  geom_point(data = cell_ct_rev2, aes(x = days, y = log2_sko_ct, color = rep)) +
  geom_abline(intercept = log2_sko_lm$coefficients[1], slope = log2_sko_lm$coefficients[2], color = "blue", size = 3, alpha = 0.3) +
  geom_line(data = cell_ct_rev2, aes(x = days, y = log2_slp_ct, color = rep), linetype = 2) +
  geom_point(data = cell_ct_rev2, aes(x = days, y = log2_slp_ct, color = rep), shape = 3) +
  geom_abline(intercept = log2_slp_lm$coefficients[1], slope = log2_slp_lm$coefficients[2], size =3, alpha =0.3)

model_frame <- data.frame("days" = seq(0,6, 1), "sko_fit" = NA, "slp_fit" = NA)
model_frame$sko_fit[1] <- 5e5 #2^log2_sko_lm$coefficients[1]
model_frame$slp_fit[1] <- 5e5 #2^log2_slp_lm$coefficients[1]
for(x in 2:nrow(model_frame)){
  model_frame$sko_fit[x] <- model_frame$sko_fit[x-1] * 2^log2_sko_lm$coefficients[2]
  model_frame$slp_fit[x] <- model_frame$slp_fit[x-1] * 2^log2_slp_lm$coefficients[2]
}

## Try flipping the equation around
log2_sko_lm2 <- lm(cell_ct_rev2$days ~ cell_ct_rev2$log2_sko_ct)
paste("The knockout cells seem to double every ~", round(log2_sko_lm2$coefficients[2],1),"days")

log2_slp_lm2 <- lm(cell_ct_rev2$days ~ cell_ct_rev2$log2_slp_ct)
paste("The knockout cells seem to double every ~", round(log2_slp_lm2$coefficients[2],1),"days")

Cell_growth_plot <- ggplot() + scale_y_log10() +
  theme(panel.grid = element_blank(), legend.position = "none") +
  labs(x = "Days", y = "Cell number\n(Log2 transformed)") +
  geom_line(data = cell_ct_rev2, aes(x = days, y = sko_ct, linetype = rep), color = "blue") +
  geom_point(data = cell_ct_rev2, aes(x = days, y = sko_ct, shape = rep), color = "blue") +
  geom_line(data = model_frame, aes(x = days, y = sko_fit), color = "blue", size = 4, alpha = 0.2) +
  geom_line(data = cell_ct_rev2, aes(x = days, y = slp_ct, linetype = rep)) +
  geom_point(data = cell_ct_rev2, aes(x = days, y = slp_ct, shape = rep)) +
  geom_line(data = model_frame, aes(x = days, y = slp_fit), color = "black", size = 4, alpha = 0.2) +
  annotate("text", x = 0, y = 5e6, label = paste("Doubling times:"), hjust = 0, size = 3) +
  annotate("text", x = 0, y = 3.8e6, label = paste("Standard cells: ~",round(log2_slp_lm2$coefficients[2],1),"days"), hjust = 0, size = 3) +
  annotate("text", x = 0, y = 3e6, label = paste("Knockout cells: ~",round(log2_sko_lm2$coefficients[2],1),"days"), hjust = 0, size = 3, color = "blue") +
  NULL; Cell_growth_plot
ggsave(file = "Revision_1/plots_r1/supplement_r1/supp1a_Cell_growth_plot.pdf", Cell_growth_plot, height = 2, width = 3)

```

##Bacteria growth - Supplement Fig 5c
```{r}
bac_growth = read.csv("Revision_1/data_r1/rev_bac_growth.csv")
bac_growth = bac_growth %>% mutate(cell_number = A600*8*10^8)
bac_growth = bac_growth %>% group_by(strain, plasmid, time_point) %>% mutate(mean = exp(mean(log(cell_number), na.rm = T)))

bac_growth2 <- bac_growth %>% group_by(rep, strain, plasmid, time_point) %>% summarize(cell_number = mean(cell_number))
bac_growth_geomean <- bac_growth %>% group_by(strain, plasmid, time_point) %>% summarize(cell_number = 10^mean(log10(cell_number), na.rm = T))

bac_growth2$rep <- as.factor(bac_growth2$rep)
bac_growth2$log2_cells <- log(bac_growth2$cell_number,2)
bac_growth_geomean$log2_cells <- log(bac_growth_geomean$cell_number,2)

bac_growth_plot = ggplot() + 
  theme(panel.grid = element_blank()) + 
  #geom_smooth(data = bac_growth2 %>% filter(!is.na(log2_cells)), aes(x = time_point, y = log2_cells, color = plasmid)) +
  geom_line(data = bac_growth_geomean , aes(x = time_point, y = log2_cells, color = plasmid), size = 2, alpha = 0.4) +
  geom_point(data = bac_growth_geomean , aes(x = time_point, y = log2_cells, color = plasmid), size = 2) +
  geom_point(data = bac_growth2 %>% filter(!is.na(log2_cells)), aes(x = time_point, y = log2_cells, color = plasmid), alpha = 0.4) +
  labs(x= "Time point", y= "Cell number \n(Log2 transformed)", color = "Sample")+
  theme(legend.position = "bottom")+
  facet_grid(cols = vars(strain))+NULL; bac_growth_plot
ggsave("Revision_1/plots_r1/supplement_r1/bac_growth_plot.pdf", bac_growth_plot, height = 2.5, width = 4)

## Replotting with OD instead of cell number

bac_growth2 <- bac_growth %>% group_by(rep, strain, plasmid, time_point) %>% summarize(a600 = mean(A600))
bac_growth_geomean <- bac_growth %>% group_by(strain, plasmid, time_point) %>% summarize(a600 = 10^mean(log10(A600), na.rm = T))

bac_growth2$rep <- as.factor(bac_growth2$rep)
#bac_growth2$log10_a600 <- log(bac_growth2$a600,2)
#bac_growth_geomean$log10_a600 <- log(bac_growth_geomean$a600,2)

Bacterial_growth_plot <- ggplot() + 
  theme(panel.grid = element_blank()) + 
  #geom_smooth(data = bac_growth2 %>% filter(!is.na(log10_a600)), aes(x = time_point, y = log10_a600, color = plasmid)) +
  geom_line(data = bac_growth_geomean , aes(x = time_point, y = a600, color = plasmid), size = 2, alpha = 0.4) +
  geom_point(data = bac_growth_geomean , aes(x = time_point, y = a600, color = plasmid), size = 2) +
  geom_point(data = bac_growth2 %>% filter(!is.na(a600)), aes(x = time_point, y = a600, color = plasmid), alpha = 0.4) +
  facet_grid(cols = vars(strain)) +
  labs(x = "Hours after transformation", y = "OD (A600)") + 
  NULL; Bacterial_growth_plot
ggsave("Revision_1/plots_r1/supplement_r1/supp5c_Bacterial_growth_plot.pdf", Bacterial_growth_plot, height = 2, width = 5)
ggsave("Revision_1/plots_r1/supplement_r1/Bacterial_growth_plot.png", Bacterial_growth_plot, height = 2, width = 5)
```

```{r session info}
sessionInfo()
```

